---
title: "Genetic analyses"
subtitle: Accompanying 'AE_TEMPLATE'
author: '[Sander W. van der Laan, PhD](https://vanderlaanand.science) | s.w.vanderlaan[at]gmail[dot]com'
date: '`r Sys.Date()`'
output:
  html_notebook: 
    cache: yes
    code_folding: hide
    collapse: yes
    df_print: paged
    fig.align: center
    fig_caption: yes
    fig_height: 10
    fig_retina: 2
    fig_width: 12
    theme: paper
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
    highlight: tango
mainfont: Helvetica
editor_options:
  chunk_output_type: inline
bibliography: references.bib
knit: worcs::cite_all
---

# General Setup

```{r echo = FALSE}
rm(list = ls())
```

```{r LocalSystem, echo = FALSE}
source("scripts/local.system.R")
```

```{r Source functions}
source("scripts/functions.R")
```

```{r loading_packages, message=FALSE, warning=FALSE}
source("scripts/pack01.packages.R")
```

```{r Setting: Colors}
Today = format(as.Date(as.POSIXlt(Sys.time())), "%Y%m%d")
Today.Report = format(as.Date(as.POSIXlt(Sys.time())), "%A, %B %d, %Y")
source("scripts/colors.R")
```

```{r setup_notebook, include=FALSE}
# We recommend that you prepare your raw data for analysis in 'prepare_data.R',
# and end that file with either open_data(yourdata), or closed_data(yourdata).
# Then, uncomment the line below to load the original or synthetic data
# (whichever is available), to allow anyone to reproduce your code:
# load_data()

# further define some knitr-options.
knitr::opts_chunk$set(fig.width = 12, fig.height = 8, fig.path = 'Figures/', 
                      warning = TRUE, # show warnings during codebook generation
                      message = TRUE, # show messages during codebook generation
                      error = TRUE, # do not interrupt codebook generation in case of errors, 
                                    # usually better for debugging
                      echo = TRUE,  # show R code
                      eval = TRUE)

ggplot2::theme_set(ggplot2::theme_minimal())
# pander::panderOptions("table.split.table", Inf)
library("worcs")
library("rmarkdown")
```


## Background

Collaboration to study `r TRAIT_OF_INTEREST` in relation to atherosclerotic plaques characteristics.

-   `Genes.xlsx` - list of genes of interest. 
-   `Variants.xlsx` - list of variant(s) of interest. 

```{r targets, message=FALSE, warning=FALSE}
library(openxlsx)

gene_list_df <- read.xlsx(paste0(TARGET_loc, "/targets.xlsx"), sheet = "Genes")

gene_list <- unlist(gene_list_df$Gene)
gene_list

variant_list <- read.xlsx(paste0(TARGET_loc, "/targets.xlsx"), sheet = "Variants")

credset_list <- read.xlsx(paste0(TARGET_loc, "/targets.xlsx"), sheet = "CredSet")


DT::datatable(variant_list)

```


## This notebook 

In this notebook we setup the files for the SNP analysis. 

## Athero-Express Biobank Study

The [*Athero-Express Biobank Study (AE)*](http://www.atheroexpress.nl){target="_blank"} contains plaque material of
patients that underwent endarterectomyat two Dutch tertiary referral centers. Details of the study design were described
before. Briefly, blood and plaque material were obtained during endarterectomy and stored at -80 ℃. Only carotid
endarterectomy (CEA) patients were included in the present study. All patients provided informed consent and the study
was approved by the medical ethics committee.

## Athero-Express Genomics Study

### DNA isolation and genotyping

We genotyped the AE in three separate, but consecutive experiments. In short, DNA was extracted from EDTA blood or (when
no blood was available) plaque samples of 2,652 samples of 2,266 consecutive patients from the Athero-Express Biobank Study 
and genotyped in 3 batches.

For the *Athero-Express Genomics Study 1 (AEGS1)* 975 samples from 822 unique patients, included between 2002 and 2007, were 
genotyped (440,763 markers) using the Affymetrix Genome-Wide Human SNP Array 5.0 (SNP5) chip (Affymetrix Inc., Santa Clara, CA, USA) at [Eurofins Genomics](https://www.eurofinsgenomics.eu/){target="_blank"} (formerly known as AROS). After genotyping and genotype calling,
14 samples failed hybridization and 84 samples were excluded due to low genotype call rate. This resulted in 891 samples 
(589 males, 259 females, 43 unknown sex). 

For the *Athero-Express Genomics Study 2 (AEGS2)* 1020 samples of an equal number of patients included between 2002 and 2013, 
were genotyped (587,351 markers) using the Affymetrix AxiomⓇ GW CEU 1 Array (AxM) at the [Genome Analysis. Center](https://www.helmholtz-muenchen.de/no_cache/gac/index.html){target="_blank"}. After genotyping and genotype calling,
19 samples failed hybridization and 47 sample was excluded due to low genotype call rate. This resulted in 954 samples (640 males, 
313 females, 1 unknown sex).

For the *Athero-Express Genomics Study 3 (AEGS3)* 658 samples of 643 unique patients, included between 2002 and 2016, were 
genotyped (693,931 markers) using the Illumina GSA MD v1 BeadArray (GSA) at 
[Human Genomics Facility, HUGE-F](http://glimdna.org/index.html){target="_blank"}. After genotyping and genotype calling,
0 samples failed hybridization and 8 samples were excluded due to low genotype call rate. This resulted in 650 samples 
(442 males, 197 females, and 11 unknown sex). 

All experiments were carried out according to OECD standards.

### Genotyping calling

We used the genotyping calling algorithms as advised by Affymetrix (AEGS1 and AEGS2) and Illumina (AEGS3):

-   AEGS1: BRLMM-P
-   AEGS2: AxiomGT1
-   AEGS3: Illumina GenomeStudio

### Quality control after genotyping

After genotype calling, we adhered to community standard quality control and assurance (QCA) procedures of the genotype
data from AEGS1, AEGS2, and AEGS3. Samples with low average genotype calling and sex discrepancies (compared to the
clinical data available) were excluded. The data was further filtered on:

1)  individual (sample) call rate \> 97%,
2)  SNP call rate \> 97%,
3)  minor allele frequencies (MAF) \> 3%,
4)  average heterozygosity rate ± 3.0 s.d.,
5)  relatedness (pi-hat \> 0.20),
6)  Hardy--Weinberg Equilibrium (HWE p \< 1.0×10<sup>−3\<\sup\>), and
7)  Monomorphic SNPs (\< 1.0×10<sup>−6\<\sup\>).

After QCA 2,267 samples of 2,079 patients (1,547 males, 720 females) and 1,416,258 unique variants across the three 
datasets remained. These comprise 15 (12 patients) 2nd degree relatives, 36 (33 patients) 1st degree relatives,
235 (117 patients) replicate samples, 119 (59 patients) duplicate samples, and 1,862 unrelated unique patient samples. 
For AEGS1 780 samples (535 males, 245 females) and 409,023 SNPs remained, for AEGS2 900 samples (605 males, 295 females) 
and 529,193 SNPs remained, and for AEGS3 587 samples (407 malesm 180 females) and 661,053 SNPs remained. These data are 
all mapped against hg19/b37.

### Imputation

Prior to imputation the cleaned data were converted to VCF v4.2 format, index and sorted by genomic position and per 
chromosome, and sampleIDs were reformated to UPID.[arrayID].UPID_STUDYNUMBER, remaining strand flips (AEGS3) were fixed, 
and finally the validity of all VCF-files were checked. For imputation we used the 
[TOPMed Imputation server](https://imputation.biodatacatalyst.nhlbi.nih.gov/#!) r3, freeze 10, GRCh38/UCSC b38 (`apps@topmed-r3@1.0.0 (hg38)`) 
which covers 445,600,184 sites across chromosome 1-22 and X from 133,597 individuals. We used the 'mixed' population, 
and the EAGLE algorithm for phasing, and put the _R<sup>2</sup>_ to 0.001 to filter out all badly imputed variants.

### Quality control after imputation

We merge the three AEGS datasets using a collection of scripts applying `bcftools`. Next, we applied principal component
analysis (PCA) to discriminate samples of European (EUR) ancestry from non-European ancestry. To this end we used 
`plink2 --pca` and the b38 high-coverage 1000G data covering 26 populations from around the globe. Thus we end up with:

- 2,267 samples from 2,079 patients; 1,547 males, and 720 females
- 2,159 samples are NL/EUR, 88 non-EUR, and 20 non-NL

These include:

- 15 (12) 2nd degree relatives
- 36 (33) 1st degree relatives
- 235 (117) replicates
- 119 (59) duplicates
- 1,862 (1,862) unrelated

We randomly selected one of the related samples (family members). We randomly selected one of the duplicate genotyped 
samples, when they had the same UPID, but different STUDY_NUMBER and same genotyping chip. For duplicate genotyped samples, 
when they had the same UPID, but different STUDY_NUMBER and different genotyping chips, we selected the sample with the 
latest chip. For replicate genotyped samples, with the same UPID, and same STUDY_NUMBER and same genotyping chip, we 
randomly selected one. For replicate genotyped samples, with the same UPID, and same STUDY_NUMBER and different 
genotyping chips, we selected the samples with latest chip.



```
                                         NL nonEUR nonNL
  discard_duplicate                      55      0     2
  discard_duplicate_latest_chip           6     15     4
  discard_duplicate_old_chip              3     11     1
  discard_replicate                      82      0     4
  discard_replicate_old_chip              0      0     1
  family_discard                         20      0     0
  family_discard_duplicate_latest_chip    1      0     1
  family_discard_duplicate_old_chip       2      0     0
  family_discard_replicate_latest_chip    1      0     0
  family_discard_replicate_old_chip       0      3     1
  family_keep                            19      0     0
  family_keep_duplicate_latest_chip       1      0     0
  family_keep_replicate_latest_chip       1      0     1
  passed                               1826     36     0
  passed_duplicate                       55      0     1
  passed_duplicate_latest_chip            3     14     0
  passed_duplicate_old_chip               6      9     2
  passed_replicate                       78      0     2
```

After identification of non-EUR and selecting family, duplicate and replicate samples, we executed another round
of `plink2 --pca`. We finally have:

- 1,989 NL/EUR, 59 non-EUR, 6 non-NL unique patients
- 73,603,379 unique variants across chromosome 1-22, X, and 1,436 Y-variants, and 197 MT-variants.


#### Data files

The data files are available at:

`/hpc/dhl_ec/data/_ae_originals/AEGS_QC_imputation_2023/aegscombo/_topmed_r3_f10_b38`

The files are named split by chromosome and compressed with `bgzip` and indexed with `tabix`:
`aegscombo.topmed_r3_f10_b38.split_norm_af_filter.chr[1-22.X].vcf.gz`

Where:

- `split` indicates that the multi-allelic sites have been split; 
- `norm` indicates that the data has been normalized to the b38 reference and variants not present in the reference have been removed, all variantIDs have been set to `%CHROM:%POS:%REF:%ALT`, and duplicate variants have been removed;
- `af` indicates that the `AVG_CS`, `R2`, `AF`, and `MAF` have been averaged, the `NS`, `AN`, `AC`, `AC_Hom`, `AC_Het`, and `AC_Hemi` have been summed upon merge;
- `filter` indicates that the data has been filtered on having an the `DS!="."` (remember `R2>0.001`).

> NOTE #1: the `DS` field is the dosage of the alternate allele, and `R2` is the imputation quality score. The `AVG_CS` is the average call score and highly correlated to `R2` (filtering on `R2` suffices), `AF` is the allele frequency, `MAF` is the minor allele frequency, `NS` is the number of samples with data, `AN` is the number of alleles, `AC` is the allele count, `AC_Hom` is the homozygous allele count, `AC_Het` is the heterozygous allele count, and `AC_Hemi` is the hemizygous allele count.
> NOTE #2: the `AC_Hemi` is only present in the X-chromosome data.
> NOTE #3: variants for chromosome Y and MT will be added at a later stage.


# Loading data

Loading Athero-Express Biobank Study clinical and biobank data, as well as the SampleList of genetic data. We simply load the previously saved `RDS`-file and extract the clinical data from that.

```{r LoadAEDB}
cat("* get Athero-Express Biobank Study Database...\n")

AEDB.CEA <- readRDS(file = paste0(OUT_loc, "/",Today,".",TRAIT_OF_INTEREST,".AEDB.CEA.RDS"))
# AEDB.CEA <- readRDS(file = paste0(OUT_loc, "/20240109.",TRAIT_OF_INTEREST,".AEDB.CEA.RDS"))
AEDB.CEA[1:10, 1:10]
dim(AEDB.CEA)

AEDB.full <- readRDS(file = paste0(OUT_loc, "/",Today,".",TRAIT_OF_INTEREST,".AEDB.FULL.RDS"))
# AEDB.full <- readRDS(file = paste0(OUT_loc, "/20240109.",TRAIT_OF_INTEREST,".AEDB.FULL.RDS"))
AEDB.full[1:10, 1:10]
dim(AEDB.full)

cat("\n* get Athero-Express Genomics Study keys...\n")
# b37 KeyTable
AEGS123.b37.sampleList.keytable <- fread(paste0(AEGSQC_loc, "/QC/SELECTIONS/20200419.QC.AEGS123.sampleList.keytable.txt"))
# b38 KeyTable
AEGS123.sampleList.keytable <- fread(paste0(TOPMED_loc, "/OUTPUT/20240604.aegscombo_tm_r3_f10_b38_pca_key_table_qc.txt"))
# fix this selection variable - latest version - to match the rest of the script
names(AEGS123.sampleList.keytable)[names(AEGS123.sampleList.keytable) == "QC_final_2023_combined_selection_v2"] <- "QC_final_2023_combined_selection"

dim(AEGS123.sampleList.keytable)
# AEGS123.sampleList.keytable[1:10, 1:10]

```


# Athero-Express Genomics Study

## Prepare baseline

Let's combine the full Athero-Express Biobank Study with the key-table containing the AEGS data.

> NOTE: this should sum to 2,124 samples with genotypes.

```{r create AEGS}
AEGS <- merge(AEDB.full, AEGS123.sampleList.keytable, 
              by.x = "STUDY_NUMBER", 
              by.y = "STUDY_NUMBER_2023", sort = FALSE,
              all = TRUE)

dim(AEGS)
# fix some stuff
names(AEGS)[names(AEGS) == "Chip"] <- "CHIP"
names(AEGS)[names(AEGS) == "AE_AAA_bijzonderheden.x"] <- "AE_AAA_bijzonderheden"
AEGS$AE_AAA_bijzonderheden.y <- NULL
names(AEGS)[names(AEGS) == "informedconsent.x"] <- "informedconsent"
AEGS$informedconsent.y <- NULL
```

```{r }
table(AEGS$CHIP, useNA = "ifany")

AEGS$GWAS <- AEGS$CHIP
AEGS$GWAS[is.na(AEGS$GWAS)] <- "not genotyped"
AEGS$GWAS[AEGS$GWAS != "not genotyped"] <- "genotyped"

table(AEGS$CHIP, AEGS$GWAS, useNA = "ifany")
```

Also a visualization of the AEGS with AEDB overlaps.

```{r visualise AEGS overlaps, message=FALSE, warning=FALSE}
library(UpSetR)
require(ggplot2)
require(plyr)
require(gridExtra)
require(grid)

AEDB.availGWAS = list(
AEGS1 = subset(AEGS, CHIP == "AffySNP5", select = c("STUDY_NUMBER"))[,1],
AEGS2 = subset(AEGS, CHIP == "AffyAxiomCEU", select = c("STUDY_NUMBER"))[,1],
AEGS3 = subset(AEGS, CHIP == "IllGSA", select = c("STUDY_NUMBER"))[,1],
AEDB = AEGS$STUDY_NUMBER)

p1 <- UpSetR::upset(fromList(AEDB.availGWAS), 
                    sets = c("AEDB", "AEGS1", "AEGS2", "AEGS3"), 
                    main.bar.color = c(uithof_color[15], uithof_color[3], uithof_color[2], uithof_color[21]), 
                    mainbar.y.label	= "intersection sample size", 
                    sets.bar.color = c(uithof_color[15], uithof_color[2], uithof_color[3], uithof_color[21]), 
                    sets.x.label = "sample size", keep.order = TRUE)

p1

pdf(paste0(PLOT_loc, "/", Today, ".overlap.AEDB_AEGS123.UpSetR.pdf"))
  p1
dev.off()

png(paste0(PLOT_loc, "/", Today, ".overlap.AEDB_AEGS123.UpSetR.png"))
  p1
dev.off()

rm(p1)

```

#### Notes

**UPID -- unique patient identifier**
The AEDB contains `UPID`. However, an error was rectified in the assignment of UPIDs to individuals; some had erroneously received a new UPID, that is some individuals were given *two* UPIDs. This was corrected in the AEDB, but not in the AEGS. We will use the `UPID_2023_at_genotyping` to select the samples.

**Informed consent**

As genetic information is sensitive, and people perceive this as very personal, we are (perhaps overly) strict in selecting samples based on informed consent. We only select samples with a valid informed consent.

Explicitly _no_ informed consent, or this information was not available:

- `missing` -- no informed consent was available
- `no, died` -- the patient has died
- `no, doesn't want to` -- the patient has refused to give consent
- `no, unable to sign` -- the patient was unable to sign the informed consent
- `no, no reaction` -- the patient did not react to the informed consent
- `no, lost` -- the patient was lost, _i.e._ could not be reached anymore
- `no, too old` -- the patient was too old to give informed consent
- `no (never asked for IC because there was no tissue)` -- the patient was never asked for informed consent because there was no tissue
- `no, endpoint` -- the patient was at the endpoint of the study
- `nooit geincludeerd` -- the patient was never included

Informed consent was given, but tissue cannot and/or other information cannot be used:

- `yes, no tissue, no commerical business` -- the patient has given consent, but no tissue is available, and not for commercial business
- `yes, no tissue, no questionnaires, no medical info, no commercial business` -- the patient has given consent, but no tissue is available, and not for questionnaires, medical information, or commercial business
- `yes, no tissue, no questionnaires, no health treatment, no commerical business` -- the patient has given consent, but no tissue is available, and not for questionnaires, health treatment, or commercial business
- `yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business` -- the patient has given consent, but no tissue is available, and not for questionnaires, health treatment, medical information, or commercial business
- `yes, no tissue, no health treatment` -- the patient has given consent, but no tissue is available, and not for health treatment
- `yes, no tissue, no questionnaires` -- the patient has given consent, but no tissue is available, and  not for questionnaires
- `yes, no tissue, health treatment when possible` -- the patient has given consent, but no tissue is available,  and health treatment when possible
- `yes, no tissue` -- the patient has given consent, but no tissue is available
- `yes, no tissue, no questionnaires, no health treatment, no medical info` -- the patient has given consent, but no tissue is available, and not for questionnaires, health treatment, or medical information
- `yes, no tissue, no questionnaires, no health treatment, no commercial business` -- the patient has given consent, but no tissue is available,  and not for questionnaires, health treatment, or commercial business
- `wil niets invullen, wel alles gebruiken` -- the patient provided informed consent, but does not want to fill any other form, but everything can be used
- `yes, no DNA` -- the patient has given consent, but not for the use of DNA


> Please also refer to the `AEDB.CEA.baseline.html` document for the details on the informed consent selection.

```{r SpecificSelection}
AEGSselect <- subset(AEGS, 
                     informedconsent != "missing" & # we are really strict in selecting based on 'informed consent'!
                     informedconsent != "no, died" & 
                     informedconsent != "yes, no tissue, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no health treatment" & 
                     informedconsent != "yes, no tissue, no questionnaires" & 
                     informedconsent != "yes, no tissue, health treatment when possible" & 
                     informedconsent != "yes, no tissue" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" & 
                     informedconsent != "no, doesn't want to" & 
                     informedconsent != "no, unable to sign" & 
                     informedconsent != "no, no reaction" & 
                     informedconsent != "no, lost" & 
                     informedconsent != "no, too old" & 
                     informedconsent != "no (never asked for IC because there was no tissue)" & 
                     informedconsent != "no, endpoint" & 
                     informedconsent != "wil niets invullen, wel alles gebruiken" & 
                     informedconsent != "nooit geincludeerd" & 
                     informedconsent != "yes, no DNA")

# all genotyped samples
AEGSselect.EA <- subset(AEGS, !is.na(QC_final_2023_combined_selection) & 
                           QC_final_2023_combined_selection != "discard_duplicate" & 
                           QC_final_2023_combined_selection != "discard_duplicate_old_chip" &
                           QC_final_2023_combined_selection != "discard_replicate" &
                           QC_final_2023_combined_selection != "discard_replicate_old_chip" &
                           QC_final_2023_combined_selection != "failed_hybrdization" &
                           QC_final_2023_combined_selection != "family_discard" &
                           QC_final_2023_combined_selection != "inbreeding" &
                           QC_final_2023_combined_selection != "low_callrate" &
                           QC_final_2023_combined_selection != "other_issue" &
                           QC_final_2023_combined_selection != "qc_issue" &
                     # (Artery_summary == "carotid (left & right)" | Artery_summary == "other carotid arteries (common, external)") & # we only want carotids
                     informedconsent != "missing" & # we are really strict in selecting based on 'informed consent'!
                     informedconsent != "no, died" & 
                     informedconsent != "yes, no tissue, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no health treatment" & 
                     informedconsent != "yes, no tissue, no questionnaires" & 
                     informedconsent != "yes, no tissue, health treatment when possible" & 
                     informedconsent != "yes, no tissue" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" & 
                     informedconsent != "no, doesn't want to" & 
                     informedconsent != "no, unable to sign" & 
                     informedconsent != "no, no reaction" & 
                     informedconsent != "no, lost" & 
                     informedconsent != "no, too old" & 
                     informedconsent != "no (never asked for IC because there was no tissue)" & 
                     informedconsent != "no, endpoint" & 
                     informedconsent != "wil niets invullen, wel alles gebruiken" & 
                     informedconsent != "nooit geincludeerd" & 
                     informedconsent != "yes, no DNA")

# all genotyped CEA samples
AEGSselect.CEA <- subset(AEGS, !is.na(QC_final_2023_combined_selection) & 
                           QC_final_2023_combined_selection != "discard_duplicate" & 
                           QC_final_2023_combined_selection != "discard_duplicate_old_chip" &
                           QC_final_2023_combined_selection != "discard_replicate" &
                           QC_final_2023_combined_selection != "discard_replicate_old_chip" &
                           QC_final_2023_combined_selection != "failed_hybrdization" &
                           QC_final_2023_combined_selection != "family_discard" &
                           QC_final_2023_combined_selection != "inbreeding" &
                           QC_final_2023_combined_selection != "low_callrate" &
                           QC_final_2023_combined_selection != "other_issue" &
                           QC_final_2023_combined_selection != "qc_issue" &
                     (Artery_summary == "carotid (left & right)" | Artery_summary == "other carotid arteries (common, external)") & # we only want carotids
                     informedconsent != "missing" & # we are really strict in selecting based on 'informed consent'!
                     informedconsent != "no, died" & 
                     informedconsent != "yes, no tissue, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no health treatment" & 
                     informedconsent != "yes, no tissue, no questionnaires" & 
                     informedconsent != "yes, no tissue, health treatment when possible" & 
                     informedconsent != "yes, no tissue" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" & 
                     informedconsent != "no, doesn't want to" & 
                     informedconsent != "no, unable to sign" & 
                     informedconsent != "no, no reaction" & 
                     informedconsent != "no, lost" & 
                     informedconsent != "no, too old" & 
                     informedconsent != "no (never asked for IC because there was no tissue)" & 
                     informedconsent != "no, endpoint" & 
                     informedconsent != "wil niets invullen, wel alles gebruiken" & 
                     informedconsent != "nooit geincludeerd" & 
                     informedconsent != "yes, no DNA")

dim(AEGSselect)
dim(AEGSselect.EA)
dim(AEGSselect.CEA)
```

## Overview of samples per artery

```{r}
table(AEGSselect$Artery_summary, AEGSselect$GWAS, useNA = "ifany")
table(AEGSselect$Artery_summary, AEGSselect$CHIP, useNA = "ifany")
table(AEGSselect$QC_final_2023_combined_selection, AEGSselect$CHIP, useNA = "ifany")
table(AEGSselect$QC_final_2023_combined_selection, AEGSselect$SAMPLE_TYPE, useNA = "ifany")
```


## Baseline

Showing the baseline table of the Athero-Express Genomics Study.

```{r }
# Create baseline tables
# http://rstudio-pubs-static.s3.amazonaws.com/13321_da314633db924dc78986a850813a50d5.html
library(labelled)
AEGSselect$GWAS <- to_factor(AEGSselect$GWAS)
AEGSselect$CHIP <- to_factor(AEGSselect$CHIP)
AEGSselect$PCA <- to_factor(AEGSselect$PCA)
AEGSselect$SAMPLE_TYPE <- to_factor(AEGSselect$SAMPLE_TYPE)
AEGSselect$informedconsent <- to_factor(AEGSselect$informedconsent)
AEGSselect$Artery_summary <- to_factor(AEGSselect$Artery_summary)

AEGSselect.EA$QC_final_2023_combined_selection <- to_factor(AEGSselect.EA$QC_final_2023_combined_selection)
AEGSselect.EA$GWAS <- to_factor(AEGSselect.EA$GWAS)
AEGSselect.EA$CHIP <- to_factor(AEGSselect.EA$CHIP)
AEGSselect.EA$SUPERPOP <- to_factor(AEGSselect.EA$SUPERPOP)
AEGSselect.EA$POP <- to_factor(AEGSselect.EA$POP)
AEGSselect.EA$PCA <- to_factor(AEGSselect.EA$PCA)
AEGSselect.EA$SAMPLE_TYPE <- to_factor(AEGSselect.EA$SAMPLE_TYPE)
AEGSselect.EA$informedconsent <- to_factor(AEGSselect.EA$informedconsent)
AEGSselect.EA$Artery_summary <- to_factor(AEGSselect.EA$Artery_summary)

AEGSselect.CEA$QC_final_2023_combined_selection <- to_factor(AEGSselect.CEA$QC_final_2023_combined_selection)
AEGSselect.CEA$GWAS <- to_factor(AEGSselect.CEA$GWAS)
AEGSselect.CEA$CHIP <- to_factor(AEGSselect.CEA$CHIP)
AEGSselect.CEA$SUPERPOP <- to_factor(AEGSselect.CEA$SUPERPOP)
AEGSselect.CEA$POP <- to_factor(AEGSselect.CEA$POP)
AEGSselect.CEA$PCA <- to_factor(AEGSselect.CEA$PCA)
AEGSselect.CEA$SAMPLE_TYPE <- to_factor(AEGSselect.CEA$SAMPLE_TYPE)
AEGSselect.CEA$informedconsent <- to_factor(AEGSselect.CEA$informedconsent)
AEGSselect.CEA$Artery_summary <- to_factor(AEGSselect.CEA$Artery_summary)
```


```{r Baseline SampleSelect: prepare}
cat("===========================================================================================\n")
cat("CREATE BASELINE TABLE\n")

# Baseline table variables
basetable_vars = c("Hospital", 
                   "Age", "Gender", 
                   "TC_final", "LDL_final", "HDL_final", "TG_final", 
                   "systolic", "diastoli", "GFR_MDRD", "BMI", 
                   "KDOQI", "BMI_WHO", 
                   "SmokerCurrent", "eCigarettes", "ePackYearsSmoking",
                   "DiabetesStatus", "Hypertension.selfreport", "Hypertension.selfreportdrug", "Hypertension.composite", 
                   "Hypertension.drugs", "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD", 
                   "Stroke_Dx", "sympt", "Symptoms.5G", "restenos",
                   "EP_composite", "EP_composite_time",
                   "macmean0", "smcmean0", "Macrophages.bin", "SMC.bin", "neutrophils", "Mast_cells_plaque", "vessel_density_averaged",
                   "IPH.bin", 
                   "Calc.bin", "Collagen.bin", 
                   "Fat.bin_10", "Fat.bin_40", "OverallPlaquePhenotype", "Plaque_Vulnerability_Index",
                   "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm",
                   "QC_final_2023_combined_selection", "GWAS", "CHIP", "PCA", "SUPERPOP", "POP", 
                   "Artery_summary",
                   "PCSK9_plasma", "PCSK9_plasma_rankNorm")

basetable_bin = c("Gender", 
                  "KDOQI", "BMI_WHO", 
                  "SmokerCurrent", 
                  "DiabetesStatus", "Hypertension.selfreport", "Hypertension.selfreportdrug", "Hypertension.composite", 
                  "Hypertension.drugs", "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD", 
                  "Stroke_Dx", "sympt", "Symptoms.5G", "restenos",
                  "EP_composite", "Macrophages.bin", "SMC.bin",
                  "IPH.bin", 
                  "Calc.bin", "Collagen.bin", 
                  "Fat.bin_10", "Fat.bin_40", "OverallPlaquePhenotype", "Plaque_Vulnerability_Index",
                  "QC_final_2023_combined_selection", "GWAS", "CHIP", "PCA", "SUPERPOP", "POP", 
                   "Artery_summary")

basetable_bin

basetable_con = basetable_vars[!basetable_vars %in% basetable_bin]
basetable_con

```

### AEGS vs. AEDB

All Athero-Express Genomics Study data (n = 2,462) with valid informed consent, compared to the _remaining_, _not_ genotyped Athero-Express Biobank Study (n = 1,514).

```{r Baseline SampleSelect: Visualize, whole}
cat("\n===========================================================================================\n")
cat("DISPLAY BASELINE TABLE\n")

AEGSselect.tableOne = print(CreateTableOne(vars = basetable_vars, 
                                         # factorVars = basetable_bin,
                                         strata = "GWAS",
                                         data = AEGSselect, includeNA = TRUE), 
                          nonnormal = c(), missing = TRUE,
                          quote = FALSE, noSpaces = FALSE, showAllLevels = TRUE, explain = TRUE, 
                          format = "pf", 
                          contDigits = 3)[,1:6]
```

### AEGS -- all endarterectomies

Baseline of the valid genotyped data of all endarterectomies with valid informed consent, n = 1,945.

```{r Baseline SampleSelect: Visualize, EA}
AEGSselect.EA.tableOne = print(CreateTableOne(vars = basetable_vars, 
                                         # factorVars = basetable_bin,
                                         strata = "Gender",
                                         data = AEGSselect.EA, includeNA = TRUE), 
                          nonnormal = c(), missing = TRUE,
                          quote = FALSE, noSpaces = FALSE, showAllLevels = TRUE, explain = TRUE, 
                          format = "pf", 
                          contDigits = 3)[,1:6]
```

### AEGS -- CEA-only

Baseline of the valid genotyped data of all carotid endarterectomies (CEA) with valid informed consent, n = 1,846.

```{r Baseline SampleSelect: Visualize, CEA}
AEGSselect.CEA.tableOne = print(CreateTableOne(vars = basetable_vars, 
                                         # factorVars = basetable_bin,
                                         strata = "Gender",
                                         data = AEGSselect.CEA, includeNA = TRUE), 
                          nonnormal = c(), missing = TRUE,
                          quote = FALSE, noSpaces = FALSE, showAllLevels = TRUE, explain = TRUE, 
                          format = "pf", 
                          contDigits = 3)[,1:6]
```


### Baseline writing

Let's save the baseline characteristics of the Athero-Express Genomics Study.

```{r Baseline SampleSelection: write AEGS}
# Write basetable
require(openxlsx)

write.xlsx(file = paste0(BASELINE_loc, "/",Today,".",PROJECTNAME,".AEGS.BaselineTable.xlsx"), 
           format(as.data.frame(AEGSselect.tableOne), digits = 5, scientific = FALSE),
           rowNames = TRUE, 
           colNames = TRUE, 
           sheetName = "AEGS_Base_AEDB", overwrite = TRUE)

write.xlsx(file = paste0(BASELINE_loc, "/",Today,".",PROJECTNAME,".AEGS.EA.BaselineTable.xlsx"), 
           format(as.data.frame(AEGSselect.EA.tableOne), digits = 5, scientific = FALSE),
           rowNames = TRUE, 
           colNames = TRUE, 
           sheetName = "AEGS_Base_EA_sex", overwrite = TRUE)

write.xlsx(file = paste0(BASELINE_loc, "/",Today,".",PROJECTNAME,".AEGS.CEA.BaselineTable.xlsx"), 
           format(as.data.frame(AEGSselect.CEA.tableOne), digits = 5, scientific = FALSE),
           rowNames = TRUE, 
           colNames = TRUE, 
           sheetName = "AEGS_Base_CEA_sex", overwrite = TRUE)
```

# SampleLists

## Combined SampleList

We are ready to make a sampleList for use with the imputed data.

We fix some things in `AEGS`.
```{r}
names(AEGS)[names(AEGS) == "UPID_2023"] <- "UPID_2023_at_genotyping"
names(AEGS)[names(AEGS) == "STUDY_TYPE_2023"] <- "STUDY_TYPE"
names(AEGS)[names(AEGS) == "STUDY_2023"] <- "STUDY"
```



We change the `STUDY_TYPE` to include information on the artery operated, this is recorded in `STUDY_ARTERY`. 

```{r}
AEGS$STUDY_ARTERY <- AEGS$STUDY_TYPE
table(AEGS$STUDY_ARTERY, useNA = "ifany")
```


The categories are as follows:

- `carotid (left & right)` and `other carotid arteries (common, external)` are combined into `CEA`
- `femoral/iliac (left, right or both sides)` and `other arteries (renal, popliteal, vertebral)` are combined into `FEA`
- `carotid bypass and injury (left, right or both sides)` and `femoral bypass, angioseal and injury (left, right or both sides)` are combined into `Other`
- `aneurysmata (carotid & femoral)` and `aorta` are combined into `AAA`


```{r}
AEGS$STUDY_ARTERY[AEGS$Artery_summary == "femoral/iliac (left, right or both sides)" | AEGS$Artery_summary == "other arteries (renal, popliteal, vertebral)"] <- "FEA"
AEGS$STUDY_ARTERY[AEGS$Artery_summary == "carotid (left & right)" | AEGS$Artery_summary == "other carotid arteries (common, external)"] <- "CEA"
AEGS$STUDY_ARTERY[AEGS$Artery_summary == "carotid bypass and injury (left, right or both sides)" | AEGS$Artery_summary == "femoral bypass, angioseal and injury (left, right or both sides)"] <- "Other"
AEGS$STUDY_ARTERY[AEGS$Artery_summary == "aneurysmata (carotid & femoral)" | AEGS$Artery_summary == "aorta"] <- "AAA"
```

```{r}
table(AEGS$STUDY_ARTERY, useNA = "ifany")
```


Here we select the samples that are genotyped.

```{r Prep SampleSelect}
require(openxlsx)

AEGS_GWAS <- subset(AEGS,
               GWAS == "genotyped",
               select = c("ID_1", "ID_2", # ID_2 is the order of samples!
                          "#FID", "IID", # FID is the family ID, IID is the individual ID
                          "Gender_2023", # gender in PLINK coding -- SEX can also be used in PLINK
                          "UPID_2023_at_genotyping", 
                          "UPID", "STUDY_NUMBER", 
                          "QC_final_2023_combined_selection", 
                          "STUDY_TYPE", "STUDY_ARTERY",
                          "SAMPLE_TYPE", "DNA_protocol", "FILENAME",
                          "GWAS", "CHIP", "PCA", "SUPERPOP", "POP",
                          "PC1", "PC2", "PC3", "PC4", "PC5",
                          "PC6", "PC7", "PC8", "PC9", "PC10",
                          "SEX", "Age", "ORyear", "Artery_summary", "informedconsent",
                          "Calc.bin", "Collagen.bin", 
                          "Fat.bin_10", "Fat.bin_40", "IPH.bin", 
                          "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm",
                          "Plaque_Vulnerability_Index",
                          "PCSK9_plasma", "PCSK9_plasma_rankNorm")) # Select some phenotype of interest
dim(AEGS_GWAS)

# Fix things
attach(AEGS_GWAS)

AEGS_GWAS[,"Calcification"] <- NA
AEGS_GWAS$Calcification[Calc.bin == "no/minor"] <- "control"
AEGS_GWAS$Calcification[Calc.bin == "moderate/heavy"] <- "case"

AEGS_GWAS[,"Collagen"] <- NA
AEGS_GWAS$Collagen[Collagen.bin == "no/minor"] <- "control"
AEGS_GWAS$Collagen[Collagen.bin == "moderate/heavy"] <- "case"

AEGS_GWAS[,"Fat10"] <- NA
AEGS_GWAS$Fat10[Fat.bin_10 == " <10%"] <- "control"
AEGS_GWAS$Fat10[Fat.bin_10 == " >10%"] <- "case"

AEGS_GWAS[,"Fat40"] <- NA
AEGS_GWAS$Fat40[Fat.bin_40 == "<40%"] <- "control"
AEGS_GWAS$Fat40[Fat.bin_40 == ">40%"] <- "case"

AEGS_GWAS[,"IPH"] <- NA
AEGS_GWAS$IPH[IPH.bin == "no"] <- "control"
AEGS_GWAS$IPH[IPH.bin == "yes"] <- "case"

AEGS_GWAS[,"PVI"] <- NA
AEGS_GWAS$PVI[Plaque_Vulnerability_Index == "0"] <- "PVI_cat0"
AEGS_GWAS$PVI[Plaque_Vulnerability_Index == "1"] <- "PVI_cat1"
AEGS_GWAS$PVI[Plaque_Vulnerability_Index == "2"] <- "PVI_cat2"
AEGS_GWAS$PVI[Plaque_Vulnerability_Index == "3"] <- "PVI_cat3"
AEGS_GWAS$PVI[Plaque_Vulnerability_Index == "4"] <- "PVI_cat4"
AEGS_GWAS$PVI[Plaque_Vulnerability_Index == "5"] <- "PVI_cat5"

AEGS_GWAS$Plaque_Vulnerability_Index <- AEGS_GWAS$PVI
AEGS_GWAS$PVI <- NULL

detach(AEGS_GWAS)

# Making selection variable
attach(AEGS_GWAS)
AEGS_GWAS[,"SELECTION"] <- "not_selected"
AEGS_GWAS$SELECTION[(!is.na(QC_final_2023_combined_selection) & 
                           QC_final_2023_combined_selection != "discard_duplicate" & 
                           QC_final_2023_combined_selection != "discard_duplicate_old_chip" &
                           QC_final_2023_combined_selection != "discard_replicate" &
                           QC_final_2023_combined_selection != "discard_replicate_old_chip" &
                           QC_final_2023_combined_selection != "failed_hybrdization" &
                           QC_final_2023_combined_selection != "family_discard" &
                           QC_final_2023_combined_selection != "inbreeding" &
                           QC_final_2023_combined_selection != "low_callrate" &
                           QC_final_2023_combined_selection != "other_issue" &
                           QC_final_2023_combined_selection != "qc_issue" &
                     # (Artery_summary == "carotid (left & right)" | Artery_summary == "other carotid arteries (common, external)") #& # we only want carotids
                     informedconsent != "missing" & # we are really strict in selecting based on 'informed consent'!
                     informedconsent != "no, died" &
                     informedconsent != "yes, no tissue, no commerical business" &
                     informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" &
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" &
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" &
                     informedconsent != "yes, no tissue, no health treatment" &
                     informedconsent != "yes, no tissue, no questionnaires" &
                     informedconsent != "yes, no tissue, health treatment when possible" &
                     informedconsent != "yes, no tissue" &
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info" &
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" &
                     informedconsent != "no, doesn't want to" &
                     informedconsent != "no, unable to sign" &
                     informedconsent != "no, no reaction" &
                     informedconsent != "no, lost" &
                     informedconsent != "no, too old" &
                     informedconsent != "no (never asked for IC because there was no tissue)" &
                     informedconsent != "no, endpoint" &
                     informedconsent != "wil niets invullen, wel alles gebruiken" &
                     informedconsent != "nooit geincludeerd" &
                     informedconsent != "yes, no DNA"
                ) 
               & (PCA=="NL")] <- "selected"

detach(AEGS_GWAS)
```


```{r}
table(AEGS_GWAS$SELECTION, AEGS_GWAS$QC_final_2023_combined_selection)
table(AEGS_GWAS$SELECTION, AEGS_GWAS$STUDY_TYPE)
table(AEGS_GWAS$SELECTION, AEGS_GWAS$STUDY_ARTERY)
table(AEGS_GWAS$SELECTION, AEGS_GWAS$PCA)
table(AEGS_GWAS$SELECTION, AEGS_GWAS$POP)
```


```{r Save SampleSelect}
# Check https://www.well.ox.ac.uk/~gav/snptest/#multinomial_tests for details on multiple categories testing in SNPTEST

AEGS123_sample.list <- AEGS_GWAS[order(AEGS_GWAS$ID_2),]

AEGS123_sample.list$missing <- 0

sample_file_aegs <- dplyr::select(subset(AEGS123_sample.list, 
                                         !is.na(ID_2)),
                                  ID_1, ID_2, missing, # ID_2 is the order of samples - that way we always know what the order should be
                                  UPID_2023_at_genotyping, UPID, STUDY_NUMBER, 
                                  QC_final_2023_combined_selection, SELECTION,
                                  STUDY_TYPE, STUDY_ARTERY, Artery_summary, 
                                  SAMPLE_TYPE, DNA_protocol, FILENAME,
                                  GWAS, CHIP, PCA, SUPERPOP, POP,
                                  PC1, PC2, PC3, PC4, PC5, PC6, PC7, PC8, PC9, PC10,
                                  SEX, Age, ORyear, 
                                  Calcification, Collagen, 
                                  Fat10, Fat40, IPH, 
                                  SMC_rankNorm, MAC_rankNorm, Neutrophils_rankNorm, MastCells_rankNorm, VesselDensity_rankNorm,
                                  Plaque_Vulnerability_Index,
                                  PCSK9_plasma, PCSK9_plasma_rankNorm)  %>%
  mutate_if(is.numeric, as.character) %>%
  mutate(SAMPLE_TYPE = gsub(' ', '_', SAMPLE_TYPE)) %>%
  mutate(Artery_summary = gsub(' ', '_', Artery_summary)) %>%
  add_row(.before = 1, 
          ID_1 = "0", ID_2 = "0", missing = "0", 
          UPID_2023_at_genotyping = "D", UPID = "D", STUDY_NUMBER = "C", Artery_summary = "D",
          QC_final_2023_combined_selection = "D", SELECTION = "D",
          STUDY_TYPE = "D", STUDY_ARTERY = "D",
          SAMPLE_TYPE = "D", DNA_protocol = "D", FILENAME = "D", 
          GWAS = "D", CHIP = "D", SUPERPOP = "D", POP = "D", PCA = "D", 
          PC1 = "C", PC2 = "C", PC3 = "C", PC4 = "C", PC5 = "C", PC6 = "C", PC7 = "C", PC8 = "C", PC9 = "C", PC10 = "C",
          SEX = "D", Age = "C", ORyear = "C", 
          Calcification = "B", Collagen = "B", 
          Fat10 = "B", Fat40 = "B", IPH = "B", 
          SMC_rankNorm = "P", MAC_rankNorm = "P", Neutrophils_rankNorm = "P", MastCells_rankNorm = "P", VesselDensity_rankNorm = "P",
          Plaque_Vulnerability_Index = "D",
          PCSK9_plasma = "P", PCSK9_plasma_rankNorm = "P") %>% ## identifiers: index for these is 1, and all base variables have 0 as identifier
  print()
dim(sample_file_aegs)

fwrite(sample_file_aegs,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.sample"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

# require(DT)
# DT::datatable(sample_file_aegs, caption = "AEGS: final sample list of genotyped AE patients after quality control.", rownames = FALSE)

```

### PLINK phenotype file

We create the same file, but for use with PLINK.

```{r Save SampleSelect PLINK}
# Check https://www.well.ox.ac.uk/~gav/snptest/#multinomial_tests for details on multiple categories testing in SNPTEST

AEGS123_sample.list <- AEGS_GWAS[order(AEGS_GWAS$ID_2),]

sample_file_aegs_plink <- dplyr::select(subset(AEGS123_sample.list, 
                                         !is.na(ID_2)),
                                  `#FID`, IID, # FID is the family ID, IID is the individual ID
                                  Gender_2023, # gender in PLINK coding -- SEX can also be used in PLINK
                                  ID_1, ID_2, # ID_2 is the order of samples - that way we always know what the order should be
                                  UPID_2023_at_genotyping, UPID, STUDY_NUMBER, 
                                  QC_final_2023_combined_selection, SELECTION,
                                  STUDY_TYPE, STUDY_ARTERY, Artery_summary, 
                                  SAMPLE_TYPE, DNA_protocol, FILENAME,
                                  GWAS, CHIP, PCA, SUPERPOP, POP,
                                  PC1, PC2, PC3, PC4, PC5, PC6, PC7, PC8, PC9, PC10,
                                  SEX, Age, ORyear, 
                                  Calcification, Collagen, 
                                  Fat10, Fat40, IPH, 
                                  SMC_rankNorm, MAC_rankNorm, Neutrophils_rankNorm, MastCells_rankNorm, VesselDensity_rankNorm,
                                  Plaque_Vulnerability_Index,
                                  PCSK9_plasma, PCSK9_plasma_rankNorm)  %>%
  mutate_if(is.numeric, as.character) %>%
  mutate(SAMPLE_TYPE = gsub(' ', '_', SAMPLE_TYPE)) %>%
  mutate(Artery_summary = gsub(' ', '_', Artery_summary)) %>% ## identifiers: index for these is 1, and all base variables have 0 as identifier
  print()
dim(sample_file_aegs_plink)

fwrite(sample_file_aegs_plink,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.pheno"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

# require(DT)
# DT::datatable(sample_file_aegs, caption = "AEGS: final sample list of genotyped AE patients after quality control.", rownames = FALSE)

```


### Replication Van der Laan 2018 _et al._ 

Here replicate the [Van der Laan 2018 _et al._](https://www.ahajournals.org/doi/10.1161/CIRCGEN.118.002115) paper which included only `CHIP == AffySNP5 & CHIP == AffyAxiomCEU`. However, we do this on the samples that passed QC in 2023.

```{r }
# Check https://www.well.ox.ac.uk/~gav/snptest/#multinomial_tests for details on multiple categories testing in SNPTEST

AEGS123_sample.list <- AEGS_GWAS[order(AEGS_GWAS$ID_2),]

AEGS123_sample.list$missing <- 0

temp <- subset(AEGS123_sample.list, !is.na(ID_2))
```


```{r }
attach(temp)
temp[,"SELECTION"] <- "not_selected"
temp$SELECTION[(!is.na(QC_final_2023_combined_selection) & 
                           QC_final_2023_combined_selection != "discard_duplicate" & 
                           QC_final_2023_combined_selection != "discard_duplicate_old_chip" &
                           QC_final_2023_combined_selection != "discard_replicate" &
                           QC_final_2023_combined_selection != "discard_replicate_old_chip" &
                           QC_final_2023_combined_selection != "failed_hybrdization" &
                           QC_final_2023_combined_selection != "family_discard" &
                           QC_final_2023_combined_selection != "inbreeding" &
                           QC_final_2023_combined_selection != "low_callrate" &
                           QC_final_2023_combined_selection != "other_issue" &
                           QC_final_2023_combined_selection != "qc_issue" &
                     (STUDY_ARTERY == "CEA") # we only want carotids
                ) & (PCA=="NL")] <- "selected"
detach(temp)

sample_file_aegs_2018 <- dplyr::select(temp,
                                  ID_1, ID_2, missing, # ID_2 is the order of samples - that way we always know what the order should be
                                  UPID_2023_at_genotyping, UPID, STUDY_NUMBER, 
                                  QC_final_2023_combined_selection, SELECTION,
                                  STUDY_TYPE, STUDY_ARTERY, Artery_summary,
                                  SAMPLE_TYPE, DNA_protocol, FILENAME,
                                  GWAS, CHIP, PCA, SUPERPOP, POP,
                                  PC1, PC2, PC3, PC4, PC5, PC6, PC7, PC8, PC9, PC10,
                                  SEX, Age, ORyear, 
                                  Calcification, Collagen, 
                                  Fat10, Fat40, IPH, 
                                  SMC_rankNorm, MAC_rankNorm, Neutrophils_rankNorm, MastCells_rankNorm, VesselDensity_rankNorm,
                                  Plaque_Vulnerability_Index,
                                  PCSK9_plasma, PCSK9_plasma_rankNorm)  %>%
  mutate_if(is.numeric, as.character) %>%
  mutate(SAMPLE_TYPE = gsub(' ', '_', SAMPLE_TYPE)) %>%
  mutate(Artery_summary = gsub(' ', '_', Artery_summary)) %>%
  add_row(.before = 1, 
          ID_1 = "0", ID_2 = "0", missing = "0", 
          UPID_2023_at_genotyping = "D", UPID = "D", STUDY_NUMBER = "C",
          QC_final_2023_combined_selection = "D", SELECTION = "D",
          STUDY_TYPE = "D", STUDY_ARTERY = "D", Artery_summary = "D",
          SAMPLE_TYPE = "D", DNA_protocol = "D", FILENAME = "D", 
          GWAS = "D", CHIP = "D", SUPERPOP = "D", POP = "D", PCA = "D", 
          PC1 = "C", PC2 = "C", PC3 = "C", PC4 = "C", PC5 = "C", PC6 = "C", PC7 = "C", PC8 = "C", PC9 = "C", PC10 = "C",
          SEX = "D", Age = "C", ORyear = "C", 
          Calcification = "B", Collagen = "B", 
          Fat10 = "B", Fat40 = "B", IPH = "B", 
          SMC_rankNorm = "P", MAC_rankNorm = "P", Neutrophils_rankNorm = "P", MastCells_rankNorm = "P", VesselDensity_rankNorm = "P",
          Plaque_Vulnerability_Index = "D",
          PCSK9_plasma = "P", PCSK9_plasma_rankNorm = "P") %>% ## identifiers: index for these is 1, and all base variables have 0 as identifier
  print()
dim(sample_file_aegs_2018)

fwrite(sample_file_aegs_2018,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.2018.sample"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

rm(temp)

```


## Females only

This is the selection for females only.

```{r TEMP SampleSelect FEMALES}
# Making selection variable
AEGS123_sample.list.females <- AEGS123_sample.list
attach(AEGS123_sample.list.females)
AEGS123_sample.list.females[,"SELECTION_F"] <- "not_selected"
AEGS123_sample.list.females$SELECTION_F[(SELECTION=="selected" & SEX=="F")] <- "selected"
AEGS123_sample.list.females$SELECTION <- AEGS123_sample.list.females$SELECTION_F 
AEGS123_sample.list.females$SELECTION_F <- NULL
detach(AEGS123_sample.list.females)

table(AEGS123_sample.list.females$SELECTION, AEGS123_sample.list.females$QC_final_2023_combined_selection)
table(AEGS123_sample.list.females$SELECTION, AEGS123_sample.list.females$STUDY_TYPE)
table(AEGS123_sample.list.females$SELECTION, AEGS123_sample.list.females$PCA)
```


```{r Save SampleSelect FEMALES}
AEGS123_sample.list.females <- AEGS123_sample.list.females[order(AEGS123_sample.list.females$ID_2),]

sample_file_aegsF <- dplyr::select(subset(AEGS123_sample.list.females, 
                                         !is.na(ID_2)),
                                  ID_1, ID_2, missing, # ID_2 is the order of samples - that way we always know what the order should be
                                  UPID_2023_at_genotyping, UPID, STUDY_NUMBER, 
                                  QC_final_2023_combined_selection, SELECTION,
                                  STUDY_TYPE, STUDY_ARTERY, Artery_summary,
                                  SAMPLE_TYPE, DNA_protocol, FILENAME,
                                  GWAS, CHIP, PCA, SUPERPOP, POP,
                                  PC1, PC2, PC3, PC4, PC5, PC6, PC7, PC8, PC9, PC10,
                                  SEX, Age, ORyear, 
                                  Calcification, Collagen, 
                                  Fat10, Fat40, IPH, 
                                  SMC_rankNorm, MAC_rankNorm, Neutrophils_rankNorm, MastCells_rankNorm, VesselDensity_rankNorm,
                                  Plaque_Vulnerability_Index,
                                  PCSK9_plasma, PCSK9_plasma_rankNorm)  %>%
  mutate_if(is.numeric, as.character) %>%
  mutate(SAMPLE_TYPE = gsub(' ', '_', SAMPLE_TYPE)) %>%
  mutate(Artery_summary = gsub(' ', '_', Artery_summary)) %>%
  add_row(.before = 1, 
          ID_1 = "0", ID_2 = "0", missing = "0", 
          UPID_2023_at_genotyping = "D", UPID = "D", STUDY_NUMBER = "C",
          QC_final_2023_combined_selection = "D", SELECTION = "D",
          STUDY_TYPE = "D", STUDY_ARTERY = "D", Artery_summary = "D",
          SAMPLE_TYPE = "D", DNA_protocol = "D", FILENAME = "D", 
          GWAS = "D", CHIP = "D", SUPERPOP = "D", POP = "D", PCA = "D", 
          PC1 = "C", PC2 = "C", PC3 = "C", PC4 = "C", PC5 = "C", PC6 = "C", PC7 = "C", PC8 = "C", PC9 = "C", PC10 = "C",
          SEX = "D", Age = "C", ORyear = "C", 
          Calcification = "B", Collagen = "B", 
          Fat10 = "B", Fat40 = "B", IPH = "B", 
          SMC_rankNorm = "P", MAC_rankNorm = "P", Neutrophils_rankNorm = "P", MastCells_rankNorm = "P", VesselDensity_rankNorm = "P",
          Plaque_Vulnerability_Index = "D",
          PCSK9_plasma = "P", PCSK9_plasma_rankNorm = "P") %>% ## identifiers: index for these is 1, and all base variables have 0 as identifier
  print()
dim(sample_file_aegsF)

fwrite(sample_file_aegsF,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.females.sample"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

# require(DT)
# DT::datatable(sample_file_aegsF, caption = "AEGS: final sample list of genotyped AE patients after quality control.", rownames = FALSE)

```

## Males only

This is the selection for males only.

```{r TEMP SampleSelect MALES}
# Making selection variable
AEGS123_sample.list.males <- AEGS123_sample.list
attach(AEGS123_sample.list.males)
AEGS123_sample.list.males[,"SELECTION_M"] <- "not_selected"
AEGS123_sample.list.males$SELECTION_M[(SELECTION=="selected" & SEX=="M")] <- "selected"
AEGS123_sample.list.males$SELECTION <- AEGS123_sample.list.males$SELECTION_M 
AEGS123_sample.list.males$SELECTION_M <- NULL
detach(AEGS123_sample.list.males)

table(AEGS123_sample.list.males$SELECTION, AEGS123_sample.list.males$QC_final_2023_combined_selection)
table(AEGS123_sample.list.males$SELECTION, AEGS123_sample.list.males$STUDY_TYPE)
table(AEGS123_sample.list.males$SELECTION, AEGS123_sample.list.males$PCA)
```


```{r Save SampleSelect MALES}
AEGS123_sample.list.males <- AEGS123_sample.list.males[order(AEGS123_sample.list.males$ID_2),]

sample_file_aegsM <- dplyr::select(subset(AEGS123_sample.list.males, 
                                         !is.na(ID_2)),
                                  ID_1, ID_2, missing, # ID_2 is the order of samples - that way we always know what the order should be
                                  UPID_2023_at_genotyping, UPID, STUDY_NUMBER, 
                                  QC_final_2023_combined_selection, SELECTION,
                                  STUDY_TYPE, STUDY_ARTERY, Artery_summary,
                                  SAMPLE_TYPE, DNA_protocol, FILENAME,
                                  GWAS, CHIP, PCA, SUPERPOP, POP,
                                  PC1, PC2, PC3, PC4, PC5, PC6, PC7, PC8, PC9, PC10,
                                  SEX, Age, ORyear, 
                                  Calcification, Collagen, 
                                  Fat10, Fat40, IPH, 
                                  SMC_rankNorm, MAC_rankNorm, Neutrophils_rankNorm, MastCells_rankNorm, VesselDensity_rankNorm,
                                  Plaque_Vulnerability_Index,
                                  PCSK9_plasma, PCSK9_plasma_rankNorm)  %>%
  mutate_if(is.numeric, as.character) %>%
  mutate(SAMPLE_TYPE = gsub(' ', '_', SAMPLE_TYPE)) %>%
  mutate(Artery_summary = gsub(' ', '_', Artery_summary)) %>%
  add_row(.before = 1, 
          ID_1 = "0", ID_2 = "0", missing = "0", 
          UPID_2023_at_genotyping = "D", UPID = "D", STUDY_NUMBER = "C",
          QC_final_2023_combined_selection = "D", SELECTION = "D",
          STUDY_TYPE = "D", STUDY_ARTERY = "D", Artery_summary = "D",
          SAMPLE_TYPE = "D", DNA_protocol = "D", FILENAME = "D", 
          GWAS = "D", CHIP = "D", SUPERPOP = "D", POP = "D", PCA = "D", 
          PC1 = "C", PC2 = "C", PC3 = "C", PC4 = "C", PC5 = "C", PC6 = "C", PC7 = "C", PC8 = "C", PC9 = "C", PC10 = "C",
          SEX = "D", Age = "C", ORyear = "C", 
          Calcification = "B", Collagen = "B", 
          Fat10 = "B", Fat40 = "B", IPH = "B", 
          SMC_rankNorm = "P", MAC_rankNorm = "P", Neutrophils_rankNorm = "P", MastCells_rankNorm = "P", VesselDensity_rankNorm = "P",
          Plaque_Vulnerability_Index = "D",
          PCSK9_plasma = "P", PCSK9_plasma_rankNorm = "P")  %>% ## identifiers: index for these is 1, and all base variables have 0 as identifier
  print()
dim(sample_file_aegsM)

fwrite(sample_file_aegsM,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.males.sample"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

# require(DT)
# DT::datatable(sample_file_aegsM, caption = "AEGS: final sample list of genotyped AE patients after quality control.", rownames = FALSE)
```

# GWASToolKit preparation

## VariantLists: top SNPs

Here we create a `variantlist.txt` file used by **GWASToolKit** for analysis.

```{r create variantList}
variant_list

temp <- subset(variant_list, select = c("VariantID", "Chr", "BP"))

fwrite(temp,
       file = paste0(SNP_loc, "/variantlist.txt"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = FALSE,
       showProgress = TRUE, verbose = TRUE)
rm(temp)
```

## VariantLists: Credible Set

Here we create a `crediblesetlist.txt` file used by **GWASToolKit** for analysis. These are given the variants in the 95% credible set of causal variants.

```{r create credset_list}
credset_list

temp <- subset(credset_list, select = c("VariantID", "Chr", "BP"))

fwrite(temp,
       file = paste0(SNP_loc, "/crediblesetlist.txt"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = FALSE,
       showProgress = TRUE, verbose = TRUE)
rm(temp)
```

## Gene list

We export the list of genes for analysis in **GWASToolKit**.

This code is just an example to filter the list from genes that are not in the data.

- _COL3A_ ==> not found
- _COL2A_ ==> not found

```{r filter target genes}
gene_list_rm <- c("COL3A", "COL2A") 

temp = gene_list[!gene_list %in% gene_list_rm]

gene_list_qc <- c(temp)

# gene_list_qc <- gene_list
# 
# for debug
# gene_list_qc_replace <- c("MRTFA")
```

Here we export this list, 1 gene per line. 

```{r create genelist}
variant_list

# temp <- subset(gene_list_df, select = c("Gene"))
temp <- as.data.frame(gene_list_qc)
fwrite(temp,
       file = paste0(SNP_loc, "/genelist.txt"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = FALSE,
       showProgress = TRUE, verbose = TRUE)
rm(temp)
```

## Covariates

Here we create a `covariates.txt` file used by **GWASToolKit** for analysis.

```{r create covariatesList}
library(tidyverse)
# for 'overall' analyses
c("Age SEX ORyear STUDY_ARTERY PC1 PC2") %>% write_lines(paste0(SNP_loc, "/covariates.txt"))

# for sex-specific analyses
c("Age ORyear STUDY_ARTERY PC1 PC2") %>% write_lines(paste0(SNP_loc, "/covariates.sex.txt"))
```

## Phenotypes

Here we create a `phenotypes.txt` file used by **GWASToolKit** for analysis.

```{r create phenotypesList}
library(tidyverse)
# some plaque phenotypes
c("Calcification", "Collagen", "Fat10", "Fat40", "IPH", "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm") %>% write_lines(paste0(SNP_loc, "/phenotypes.txt"))

# the plaque vulnerability index
c("Plaque_Vulnerability_Index") %>% write_lines(paste0(SNP_loc, "/phenotypes.pvi.txt"))

# an example phenotype for a GWAS
c("PCSK9_plasma_rankNorm, PCSK9_plasma") %>% write_lines(paste0(SNP_loc, "/phenotypes.gwas.txt"))

```


# Session information

------------------------------------------------------------------------------------------------------------------------

    Version:      v1.3.5
    Last update:  2024-06-11
    Written by:   Sander W. van der Laan (s.w.vanderlaan[at]gmail[dot]com).
    Description:  Script to prepare a SNP analysis for the Athero-Express Genomics Study.
    Minimum requirements: R version 3.4.3 (2017-06-30) -- 'Single Candle', Mac OS X El Capitan

    Changes log
    * v1.3.5 Small fixes to the (example) outputs.
    * v1.3.4 Added correct reference to KeyTable for b38 data. Added 2_SNP_analyses_b37.Rmd for b37 analyses.
    * v1.3.3 Fixed an issue where the UPID used at the time of genotyping and QC was not properly included. Fixed an issue where the `STUDY_TYPE` and the `STUDY_ARTERY` were not properly included. Added an export to `PLINK`-format. Added a proper selection on informed consent.
    * v1.3.2 Fixed errors where the variantlist and crediblesetlist were malformed, added 'chr' to chromosomes. Fixed an error where the VCF-files were references incorrectly in the `.conf` file. 
    * v1.3.1 Fixed an error where .sample was malformed.
    * v1.3.0 Major update to imputation and SNP analysis. Data are imputed to TOPMed b38. 
    * v1.2.2 Textual fixes.
    * v1.2.1 Fixed issue with baseline writing.
    * v1.2.0 Update to the study database. 
    * v1.1.0 Major update to WORCS system. 
    * v1.0.6 Small bug fixes.
    * v1.0.5 Added png for overlap-figure.
    * v1.0.5 Removed obsolete references to objects.
    * v1.0.4 Fixed a mistake in the chr X sample-file creation. Now the order matches the chr X data.
    * v1.0.3 Fixed weight of files (limit of 10Mb per file for templates). Renamed entire repo.
    * v1.0.2 Added sex-specific .sample-files. Added GWASToolKit input-files.
    * v1.0.0 Initial version. Add 'plaque vulnerability index', Fixed baseline table, added codes, and results. Created sample-files.

------------------------------------------------------------------------------------------------------------------------

```{r eval = TRUE}
sessionInfo()
```

# Saving environment

```{r Saving}
save.image(paste0(PROJECT_loc, "/",Today,".",PROJECTNAME,".SNP_analyses.RData"))
```

+-----------------------------------------------------------------------------------------------------------------------------------------+
| <sup>© 1979-2024 Sander W. van der Laan | s.w.vanderlaan[at]gmail[dot]com | [vanderlaanand.science](https://vanderlaanand.science).</sup> |
+-----------------------------------------------------------------------------------------------------------------------------------------+
