---
title: "Genetic analyses"
subtitle: Accompanying 'icrpa_2024_neuroplaq'
author: '[Sander W. van der Laan, PhD](https://vanderlaanand.science) | s.w.vanderlaan@gmail.com'
date: '`r Sys.Date()`'
output:
  html_notebook: 
    cache: yes
    code_folding: hide
    collapse: yes
    df_print: paged
    fig.align: center
    fig_caption: yes
    fig_height: 10
    fig_retina: 2
    fig_width: 12
    theme: paper
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
    highlight: tango
mainfont: Helvetica
editor_options:
  chunk_output_type: inline
bibliography: references.bib
knit: worcs::cite_all
---

# General Setup

```{r echo = FALSE}
rm(list = ls())
```

```{r LocalSystem, echo = FALSE}
source("scripts/local.system.R")
```

```{r Source functions}
source("scripts/functions.R")
```

```{r loading_packages, message=FALSE, warning=FALSE}
source("scripts/pack01.packages.R")

```

```{r Setting: Colors}

Today = format(as.Date(as.POSIXlt(Sys.time())), "%Y%m%d")
Today.Report = format(as.Date(as.POSIXlt(Sys.time())), "%A, %B %d, %Y")

source("scripts/colors.R")

```

```{r setup_notebook, include=FALSE}
# We recommend that you prepare your raw data for analysis in 'prepare_data.R',
# and end that file with either open_data(yourdata), or closed_data(yourdata).
# Then, uncomment the line below to load the original or synthetic data
# (whichever is available), to allow anyone to reproduce your code:
# load_data()

# further define some knitr-options.
knitr::opts_chunk$set(fig.width = 12, fig.height = 8, fig.path = 'Figures/', 
                      warning = TRUE, # show warnings during codebook generation
                      message = TRUE, # show messages during codebook generation
                      error = TRUE, # do not interrupt codebook generation in case of errors, 
                                    # usually better for debugging
                      echo = TRUE,  # show R code
                      eval = TRUE)

ggplot2::theme_set(ggplot2::theme_minimal())
# pander::panderOptions("table.split.table", Inf)
library("worcs")
library("rmarkdown")

```


## Background

Collaboration to study `r TRAIT_OF_INTEREST` in relation to atherosclerotic plaques characteristics.

-   `Genes.xlsx` - list of genes of interest. 
-   `Variants.xlsx` - list of variant(s) of interest. 

```{r targets, message=FALSE, warning=FALSE}
library(openxlsx)

gene_list_df <- read.xlsx(paste0(TARGET_loc, "/targets.xlsx"), sheet = "Genes")

gene_list <- unlist(gene_list_df$Gene)
gene_list

variant_list <- read.xlsx(paste0(TARGET_loc, "/targets.xlsx"), sheet = "Variants")

DT::datatable(variant_list)

```


## This notebook 

In this notebook we setup the files for the SNP analysis. 

## Athero-Express Biobank Study

The [*Athero-Express Biobank Study (AE)*](http://www.atheroexpress.nl){target="_blank"} contains plaque material of
patients that underwent endarterectomyat two Dutch tertiary referral centers. Details of the study design were described
before. Briefly, blood and plaque material were obtained during endarterectomy and stored at -80 ℃. Only carotid
endarterectomy (CEA) patients were included in the present study. All patients provided informed consent and the study
was approved by the medical ethics committee.

## Athero-Express Genomics Study

### DNA isolation and genotyping

We genotyped the AE in three separate, but consecutive experiments. In short, DNA was extracted from EDTA blood or (when
no blood was available) plaque samples of 1,858 consecutive patients from the Athero-Express Biobank Study and genotyped
in 3 batches.

For the *Athero-Express Genomics Study 1 (AEGS1)* 891 patients (602 males, 262 females, 27 unknown sex), included
between 2002 and 2007, were genotyped (440,763 markers) using the Affymetrix Genome-Wide Human SNP Array 5.0 (SNP5) chip
(Affymetrix Inc., Santa Clara, CA, USA) at [Eurofins Genomics](https://www.eurofinsgenomics.eu/){target="_blank"}
(formerly known as AROS).

For the *Athero-Express Genomics Study 2 (AEGS2)* 954 patients (640 makes, 313 females, 1 unknown sex), included between
2002 and 2013, were genotyped (587,351 markers) using the Affymetrix AxiomⓇ GW CEU 1 Array (AxM) at the [Genome Analysis
Center](https://www.helmholtz-muenchen.de/no_cache/gac/index.html){target="_blank"}.

For the *Athero-Express Genomics Study 3 (AEGS3)* 658 patients (448 males, 203 females, 5 unknown sex), included between
2002 and 2016, were genotyped (693,931 markers) using the Illumina GSA MD v1 BeadArray (GSA) at [Human Genomics
Facility, HUGE-F](http://glimdna.org/index.html){target="_blank"}.

All experiments were carried out according to OECD standards.

### Genotyping calling

We used the genotyping calling algorithms as advised by Affymetrix (AEGS1 and AEGS2) and Illumina (AEGS3):

-   AEGS1: BRLMM-P
-   AEGS2: AxiomGT1
-   AEGS3: Illumina GenomeStudio

### Quality control after genotyping

After genotype calling, we adhered to community standard quality control and assurance (QCA) procedures of the genotype
data from AEGS1, AEGS2, and AEGS3. Samples with low average genotype calling and sex discrepancies (compared to the
clinical data available) were excluded. The data was further filtered on:

1)  individual (sample) call rate \> 97%,
2)  SNP call rate \> 97%,
3)  minor allele frequencies (MAF) \> 3%,
4)  average heterozygosity rate ± 3.0 s.d.,
5)  relatedness (pi-hat \> 0.20),
6)  Hardy--Weinberg Equilibrium (HWE p \< 1.0×10<sup>−3\<\sup\>), and
7)  Monomorphic SNPs (\< 1.0×10<sup>−6\<\sup\>).

After QCA 2,493 samples remained, 108 of non-European descent/ancestry, and 156 related pairs. These comprise 890
samples and 407,712 SNPs in AEGS1, 869 samples and 534,508 SNPs in AEGS2, and 649954 samples and 534,508 SNPs in AEGS3
remained.

### Imputation

Before phasing using SHAPEIT2, data was lifted to genome build b37 using the liftOver tool from UCSC
(<https://genome.ucsc.edu/cgi-bin/hgLiftOver>). Finally, data was imputed with 1000G phase 3, version 5 and HRC release
1.1 as a reference using the [Michigan Imputation Server](https://imputationserver.sph.umich.edu/){target="_blank"}.
These results were further integrated using QCTOOL v2, where HRC imputed variants are given precendence over 1000G phase
3 imputed variants.

### Quality control after imputation

We compared quality of the three AEGS datasets, and listed some variables of interest.

-   sample type (EDTA blood or plaque)
-   genotyping chip used
-   reason for filtering

We checked the studytype (AE or not), and *identity-by-descent (IBD)* within and between datasets to aid in sample
mixups, duplicate sample use, and relatedness. In addition, during genotyping quality control samples were identified
that deviated from *Hardy-Weinberg Equilibrium (HWE)*, had discordance in sex-coding and genotype sex, and deviated from
the *principal component analysis (PCA)* plot.

We will load the Athero-Express Biobank Study data, and all the samples that were send for genotyping and the final
QC'ed sampleList.

# Loading data

Loading Athero-Express Biobank Study clinical and biobank data, as well as the SampleList of genetic data. We simply load the previously saved `RDS`-file and extract the clinical data from that.

```{r LoadAEDB}
cat("* get Athero-Express Biobank Study Database...\n")

AEDB.CEA <- readRDS(file = paste0(OUT_loc, "/",Today,".",TRAIT_OF_INTEREST,".AEDB.CEA.RDS"))
# AEDB.CEA <- readRDS(file = paste0(OUT_loc, "/20240109.",TRAIT_OF_INTEREST,".AEDB.CEA.RDS"))
AEDB.CEA[1:10, 1:10]
dim(AEDB.CEA)

AEDB.full <- readRDS(file = paste0(OUT_loc, "/",Today,".",TRAIT_OF_INTEREST,".AEDB.FULL.RDS"))
# AEDB.full <- readRDS(file = paste0(OUT_loc, "/20240109.",TRAIT_OF_INTEREST,".AEDB.FULL.RDS"))
AEDB.full[1:10, 1:10]
dim(AEDB.full)


cat("\n* get Athero-Express Genomics Study keys...\n")
AEGS123.sampleList.keytable <- fread(paste0(AEGSQC_loc, "/QC/SELECTIONS/20200419.QC.AEGS123.sampleList.keytable.txt"))

dim(AEGS123.sampleList.keytable)
# AEGS123.sampleList.keytable[1:10, 1:10]


```


# Athero-Express Genomics Study

## Prepare baseline

Let's combine the full Athero-Express Biobank Study with the key-table containing the AEGS data.

> NOTE: this should sum to 2,124 samples with genotypes.

```{r create AEGS}
AEGS <- merge(AEDB.full, AEGS123.sampleList.keytable, by.x = "STUDY_NUMBER", by.y = "STUDY_NUMBER", sort = FALSE,
                  all = TRUE)

dim(AEGS)

AEGS$UPID.y <- NULL
names(AEGS)[names(AEGS) == "UPID.x"] <- "UPID"
AEGS$Age.y <- NULL
names(AEGS)[names(AEGS) == "Age.x"] <- "Age"

table(AEGS$CHIP, useNA = "ifany")

AEGS$GWAS <- AEGS$CHIP
AEGS$GWAS[is.na(AEGS$GWAS)] <- "not genotyped"
AEGS$GWAS[AEGS$GWAS != "not genotyped"] <- "genotyped"

table(AEGS$CHIP, AEGS$GWAS, useNA = "ifany")
```

Also a visualisation of the AEGS with AEDB overlaps.

```{r visualise AEGS overlaps, message=FALSE, warning=FALSE}
library(UpSetR)
require(ggplot2)
require(plyr)
require(gridExtra)
require(grid)

AEDB.availGWAS = list(
AEGS1 = subset(AEGS, CHIP == "AffySNP5", select = c("STUDY_NUMBER"))[,1],
AEGS2 = subset(AEGS, CHIP == "AffyAxiomCEU", select = c("STUDY_NUMBER"))[,1],
AEGS3 = subset(AEGS, CHIP == "IllGSA", select = c("STUDY_NUMBER"))[,1],
AEDB = AEGS$STUDY_NUMBER)

p1 <- UpSetR::upset(fromList(AEDB.availGWAS), 
                    sets = c("AEDB", "AEGS1", "AEGS2", "AEGS3"), 
                    main.bar.color = c(uithof_color[15], uithof_color[3], uithof_color[2], uithof_color[21]), 
                    mainbar.y.label	= "intersection sample size", 
                    sets.bar.color = c(uithof_color[15], uithof_color[2], uithof_color[3], uithof_color[21]), 
                    sets.x.label = "sample size", keep.order = TRUE)

p1

pdf(paste0(PLOT_loc, "/", Today, ".overlap.AEDB_AEGS123.UpSetR.pdf"))
  p1
dev.off()

png(paste0(PLOT_loc, "/", Today, ".overlap.AEDB_AEGS123.UpSetR.png"))
  p1
dev.off()

rm(p1)

```
> Please refer to the `AEDB.CEA.baseline.html` document for the details on the informed consent selection.

```{r SpecificSelection}
table(AEGS$Artery_summary, AEGS$QC2018_FILTER)
table(AEGS$informedconsent, AEGS$QC2018_FILTER)
AEGSselect <- subset(AEGS, 
                     informedconsent != "missing" & # we are really strict in selecting based on 'informed consent'!
                     informedconsent != "no, died" & 
                     informedconsent != "yes, no tissue, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no health treatment" & 
                     informedconsent != "yes, no tissue, no questionnaires" & 
                     informedconsent != "yes, no tissue, health treatment when possible" & 
                     informedconsent != "yes, no tissue" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" & 
                     informedconsent != "no, doesn't want to" & 
                     informedconsent != "no, unable to sign" & 
                     informedconsent != "no, no reaction" & 
                     informedconsent != "no, lost" & 
                     informedconsent != "no, too old" & 
                     informedconsent != "no (never asked for IC because there was no tissue)" & 
                     informedconsent != "no, endpoint" & 
                     informedconsent != "wil niets invullen, wel alles gebruiken" & 
                     informedconsent != "nooit geincludeerd" & 
                     informedconsent != "yes, no DNA")

AEGSselect.CEA <- subset(AEGS, !is.na(QC2018_FILTER) & QC2018_FILTER != "issue" & QC2018_FILTER != "family_discard" &
                     (Artery_summary == "carotid (left & right)" | Artery_summary == "other carotid arteries (common, external)") & # we only want carotids
                     informedconsent != "missing" & # we are really strict in selecting based on 'informed consent'!
                     informedconsent != "no, died" & 
                     informedconsent != "yes, no tissue, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" & 
                     informedconsent != "yes, no tissue, no health treatment" & 
                     informedconsent != "yes, no tissue, no questionnaires" & 
                     informedconsent != "yes, no tissue, health treatment when possible" & 
                     informedconsent != "yes, no tissue" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info" & 
                     informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" & 
                     informedconsent != "no, doesn't want to" & 
                     informedconsent != "no, unable to sign" & 
                     informedconsent != "no, no reaction" & 
                     informedconsent != "no, lost" & 
                     informedconsent != "no, too old" & 
                     informedconsent != "no (never asked for IC because there was no tissue)" & 
                     informedconsent != "no, endpoint" & 
                     informedconsent != "wil niets invullen, wel alles gebruiken" & 
                     informedconsent != "nooit geincludeerd" & 
                     informedconsent != "yes, no DNA")

dim(AEGSselect)

table(AEGSselect$Artery_summary, AEGSselect$QC2018_FILTER)
table(AEGSselect$Artery_summary, AEGSselect$CHIP)
table(AEGSselect$QC2018_FILTER, AEGSselect$CHIP)
table(AEGSselect$QC2018_FILTER, AEGSselect$SAMPLE_TYPE)

AEDB.temp <- subset(AEGSselect,  select = c("STUDY_NUMBER", "UPID", "Age", "Gender", "Hospital", "Artery_summary", "QC2018_FILTER", "CHIP", "SAMPLE_TYPE"))
require(labelled)
AEDB.temp$Gender <- to_factor(AEDB.temp$Gender)
AEDB.temp$Hospital <- to_factor(AEDB.temp$Hospital)
AEDB.temp$Artery_summary <- to_factor(AEDB.temp$Artery_summary)
AEDB.temp$QC2018_FILTER <- to_factor(AEDB.temp$QC2018_FILTER)
AEDB.temp$CHIP <- to_factor(AEDB.temp$CHIP)
AEDB.temp$SAMPLE_TYPE <- to_factor(AEDB.temp$SAMPLE_TYPE)

DT::datatable(AEDB.temp[1:10,], caption = "Excerpt of the whole AEDB.", rownames = FALSE)

rm(AEDB.temp)


```

## Athero-Express Genomics Study Baseline Characteristics

Showing the baseline table of the Athero-Express Genomics Study.

```{r Baseline SampleSelect: prepare}
# Create baseline tables
# http://rstudio-pubs-static.s3.amazonaws.com/13321_da314633db924dc78986a850813a50d5.html
AEGSselect$GWAS <- to_factor(AEGSselect$GWAS)
AEGSselect$CHIP <- to_factor(AEGSselect$CHIP)
AEGSselect$PCA <- to_factor(AEGSselect$PCA)
AEGSselect$SAMPLE_TYPE <- to_factor(AEGSselect$SAMPLE_TYPE)
AEGSselect$informedconsent <- to_factor(AEGSselect$informedconsent)
AEGSselect$Artery_summary <- to_factor(AEGSselect$Artery_summary)

AEGSselect.CEA$GWAS <- to_factor(AEGSselect.CEA$GWAS)
AEGSselect.CEA$CHIP <- to_factor(AEGSselect.CEA$CHIP)
AEGSselect.CEA$PCA <- to_factor(AEGSselect.CEA$PCA)
AEGSselect.CEA$SAMPLE_TYPE <- to_factor(AEGSselect.CEA$SAMPLE_TYPE)
AEGSselect.CEA$informedconsent <- to_factor(AEGSselect.CEA$informedconsent)
AEGSselect.CEA$Artery_summary <- to_factor(AEGSselect.CEA$Artery_summary)

cat("===========================================================================================\n")
cat("CREATE BASELINE TABLE\n")

# Baseline table variables
basetable_vars = c("Hospital", 
                   "Age", "Gender", 
                   "TC_final", "LDL_final", "HDL_final", "TG_final", 
                   "systolic", "diastoli", "GFR_MDRD", "BMI", 
                   "KDOQI", "BMI_WHO", 
                   "SmokerCurrent", "eCigarettes", "ePackYearsSmoking",
                   "DiabetesStatus", "Hypertension.selfreport", "Hypertension.selfreportdrug", "Hypertension.composite", 
                   "Hypertension.drugs", "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD", 
                   "Stroke_Dx", "sympt", "Symptoms.5G", "restenos",
                   "EP_composite", "EP_composite_time",
                   "macmean0", "smcmean0", "Macrophages.bin", "SMC.bin", "neutrophils", "Mast_cells_plaque", "vessel_density_averaged",
                   "IPH.bin", 
                   "Calc.bin", "Collagen.bin", 
                   "Fat.bin_10", "Fat.bin_40", "OverallPlaquePhenotype", "Plaque_Vulnerability_Index",
                   "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm",
                   "GWAS", "CHIP", "PCA", "Artery_summary",
                   "PCSK9_plasma", "PCSK9_plasma_rankNorm")

basetable_bin = c("Gender", 
                  "KDOQI", "BMI_WHO", 
                  "SmokerCurrent", 
                  "DiabetesStatus", "Hypertension.selfreport", "Hypertension.selfreportdrug", "Hypertension.composite", 
                  "Hypertension.drugs", "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD", 
                  "Stroke_Dx", "sympt", "Symptoms.5G", "restenos",
                  "EP_composite", "Macrophages.bin", "SMC.bin",
                  "IPH.bin", 
                  "Calc.bin", "Collagen.bin", 
                  "Fat.bin_10", "Fat.bin_40", "OverallPlaquePhenotype", "Plaque_Vulnerability_Index",
                  "GWAS", "CHIP", "PCA", "Artery_summary")

basetable_bin

basetable_con = basetable_vars[!basetable_vars %in% basetable_bin]
basetable_con

```

All Athero-Express Genomics Study data (n = 2,011), compared to the *remaining*, \_un_genotyped Athero-Express Biobank
Study.

```{r Baseline SampleSelect: Visualize, whole}
cat("\n===========================================================================================\n")
cat("DISPLAY BASELINE TABLE\n")

AEGSselect.tableOne = print(CreateTableOne(vars = basetable_vars, 
                                         # factorVars = basetable_bin,
                                         strata = "GWAS",
                                         data = AEGSselect, includeNA = TRUE), 
                          nonnormal = c(), missing = TRUE,
                          quote = FALSE, noSpaces = FALSE, showAllLevels = TRUE, explain = TRUE, 
                          format = "pf", 
                          contDigits = 3)[,1:6]
```


Baseline of the valid, CEA and genotyped data.

```{r Baseline SampleSelect: Visualize, CEA}
AEGSselect.CEA.tableOne = print(CreateTableOne(vars = basetable_vars, 
                                         # factorVars = basetable_bin,
                                         strata = "Gender",
                                         data = AEGSselect.CEA, includeNA = TRUE), 
                          nonnormal = c(), missing = TRUE,
                          quote = FALSE, noSpaces = FALSE, showAllLevels = TRUE, explain = TRUE, 
                          format = "pf", 
                          contDigits = 3)[,1:6]
```

### Baseline writing

Let's save the baseline characteristics of the Athero-Express Genomics Study.

```{r Baseline SampleSelection: write AEGS}
# Write basetable
require(openxlsx)

write.xlsx(file = paste0(BASELINE_loc, "/",Today,".",PROJECTNAME,".AEGS.BaselineTable.xlsx"), 
           format(as.data.frame(AEGSselect.tableOne), digits = 5, scientific = FALSE),
           rowNames = TRUE, 
           colNames = TRUE, 
           sheetName = "AEGS_Base_AEDB", overwrite = TRUE)

write.xlsx(file = paste0(BASELINE_loc, "/",Today,".",PROJECTNAME,".AEGS.CEA.BaselineTable.xlsx"), 
           format(as.data.frame(AEGSselect.CEA.tableOne), digits = 5, scientific = FALSE),
           rowNames = TRUE, 
           colNames = TRUE, 
           sheetName = "AEGS_Base_CEA_sex", overwrite = TRUE)

```

# SampleLists

## Autosomal data.

We are ready to make a sampleList for use with the imputed data.

```{r Prep SampleSelect}
require(openxlsx)

temp <- subset(AEGS,
               GWAS == "genotyped",
               select = c("ID_1", "ID_2", "UPID", "STUDY_NUMBER", # ID_2 is the order of samples!
                          "QC2018_FINAL", "QC2018_FILTER", "OriginalOrder_postMichImp_QC",
                          "AEGS_type", "CHIP", "STUDY_TYPE", "SAMPLE_TYPE", "PCA",
                          "PC1", "PC2", "PC3", "PC4", "PC5",
                          "PC6", "PC7", "PC8", "PC9", "PC10",
                          "Sex", "Age", "ORyear", 
                          "Calc.bin", "Collagen.bin", 
                          "Fat.bin_10", "Fat.bin_40", "IPH.bin", 
                          "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm",
                          "Plaque_Vulnerability_Index",
                          "PCSK9_plasma", "PCSK9_plasma_rankNorm")) # Select some phenotype of interest
dim(temp)

# Fix things
attach(temp)

temp[,"Calcification"] <- NA
temp$Calcification[Calc.bin == "no/minor"] <- "control"
temp$Calcification[Calc.bin == "moderate/heavy"] <- "case"

temp[,"Collagen"] <- NA
temp$Collagen[Collagen.bin == "no/minor"] <- "control"
temp$Collagen[Collagen.bin == "moderate/heavy"] <- "case"

temp[,"Fat10"] <- NA
temp$Fat10[Fat.bin_10 == "<10%"] <- "control"
temp$Fat10[Fat.bin_10 == ">10%"] <- "case"

temp[,"Fat40"] <- NA
temp$Fat40[Fat.bin_40 == "<40%"] <- "control"
temp$Fat40[Fat.bin_40 == ">40%"] <- "case"

temp[,"IPH"] <- NA
temp$IPH[IPH.bin == "no"] <- "control"
temp$IPH[IPH.bin == "yes"] <- "case"

temp[,"PVI"] <- NA
temp$PVI[Plaque_Vulnerability_Index == "0"] <- "PVI_cat0"
temp$PVI[Plaque_Vulnerability_Index == "1"] <- "PVI_cat1"
temp$PVI[Plaque_Vulnerability_Index == "2"] <- "PVI_cat2"
temp$PVI[Plaque_Vulnerability_Index == "3"] <- "PVI_cat3"
temp$PVI[Plaque_Vulnerability_Index == "4"] <- "PVI_cat4"
temp$PVI[Plaque_Vulnerability_Index == "5"] <- "PVI_cat5"

temp$Plaque_Vulnerability_Index <- temp$PVI
temp$PVI <- NULL

detach(temp)

# Making selection variable
attach(temp)
temp[,"SELECTION"] <- "not_selected"
temp$SELECTION[(QC2018_FILTER=="passed" | QC2018_FILTER=="family_keep") & (STUDY_TYPE=="CEA" & PCA=="EUR")] <- "selected"
detach(temp)
table(temp$SELECTION, temp$QC2018_FILTER)
table(temp$SELECTION, temp$STUDY_TYPE)
table(temp$SELECTION, temp$PCA)
table(temp$Plaque_Vulnerability_Index)

```


```{r Save SampleSelect}
# Check https://www.well.ox.ac.uk/~gav/snptest/#multinomial_tests for details on multiple categories testing in SNPTEST

AEGS123_sample.list <- temp[order(temp$OriginalOrder_postMichImp_QC),]

AEGS123_sample.list$missing <- 0

sample_file_aegs <- dplyr::select(AEGS123_sample.list,
                                  ID_1, ID_2, missing, # ID_2 is the order of samples - that way we always know what the order should be
                                  UPID, STUDY_NUMBER, 
                                  QC2018_FINAL, QC2018_FILTER, SELECTION,
                                  AEGS_type, CHIP, STUDY_TYPE, SAMPLE_TYPE, PCA,
                                  PC1, PC2, PC3, PC4, PC5, PC6, PC7, PC8, PC9, PC10,
                                  Sex, Age, ORyear, 
                                  Calcification, Collagen, 
                                  Fat10, Fat40, IPH, 
                                  SMC_rankNorm, MAC_rankNorm, Neutrophils_rankNorm, MastCells_rankNorm, VesselDensity_rankNorm,
                                  Plaque_Vulnerability_Index,
                                  PCSK9_plasma, PCSK9_plasma_rankNorm)  %>%
  mutate_if(is.numeric, as.character) %>%
  mutate(SAMPLE_TYPE = gsub(' ', '_', SAMPLE_TYPE)) %>%
  add_row(.before = 1, 
          ID_1 = "0", ID_2 = "0", missing = "0", 
          UPID = "D", STUDY_NUMBER = "C",
          QC2018_FINAL = "D", QC2018_FILTER = "D", SELECTION = "D",
          AEGS_type = "D", CHIP = "D", STUDY_TYPE = "D", SAMPLE_TYPE = "D", PCA = "D",
          PC1 = "C", PC2 = "C", PC3 = "C", PC4 = "C", PC5 = "C", PC6 = "C", PC7 = "C", PC8 = "C", PC9 = "C", PC10 = "C",
          Sex = "D", Age = "C", ORyear = "C", 
          Calcification = "B", Collagen = "B", 
          Fat10 = "B", Fat40 = "B", IPH = "B", 
          SMC_rankNorm = "P", MAC_rankNorm = "P", Neutrophils_rankNorm = "P", MastCells_rankNorm = "P", VesselDensity_rankNorm = "P",
          Plaque_Vulnerability_Index = "D",
          PCSK9_plasma = "P", PCSK9_plasma_rankNorm = "P") %>% ## identifiers: index for these is 1, and all base variables have 0 as identifier
  print()
dim(sample_file_aegs)

fwrite(sample_file_aegs,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.sample"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

require(DT)
DT::datatable(sample_file_aegs, caption = "AEGS: final sample list of genotyped AE patients after quality control.", rownames = FALSE)

rm(temp, temp2, temp3)
```


### Females only

This is the selection for females only.

```{r TEMP SampleSelect FEMALES}
require(openxlsx)

temp <- subset(AEGS,
               GWAS == "genotyped",
               select = c("ID_1", "ID_2", "UPID", "STUDY_NUMBER", # ID_2 is the order of samples!
                          "QC2018_FINAL", "QC2018_FILTER", "OriginalOrder_postMichImp_QC",
                          "AEGS_type", "CHIP", "STUDY_TYPE", "SAMPLE_TYPE", "PCA",
                          "PC1", "PC2", "PC3", "PC4", "PC5",
                          "PC6", "PC7", "PC8", "PC9", "PC10",
                          "Sex", "Age", "ORyear", 
                          "BMI",
                          "Calc.bin", "Collagen.bin", 
                          "Fat.bin_10", "Fat.bin_40", "IPH.bin", 
                          "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm", 
                          "OverallPlaquePhenotype", "Plaque_Vulnerability_Index",
                          "PCSK9_plasma", "PCSK9_plasma_rankNorm")) # Select some phenotype of interest
dim(temp)

# Fix things
attach(temp)

temp[,"Calcification"] <- NA
temp$Calcification[Calc.bin == "no/minor"] <- "control"
temp$Calcification[Calc.bin == "moderate/heavy"] <- "case"

temp[,"Collagen"] <- NA
temp$Collagen[Collagen.bin == "no/minor"] <- "control"
temp$Collagen[Collagen.bin == "moderate/heavy"] <- "case"

temp[,"Fat10"] <- NA
temp$Fat10[Fat.bin_10 == "<10%"] <- "control"
temp$Fat10[Fat.bin_10 == ">10%"] <- "case"

temp[,"Fat40"] <- NA
temp$Fat40[Fat.bin_40 == "<40%"] <- "control"
temp$Fat40[Fat.bin_40 == ">40%"] <- "case"

temp[,"IPH"] <- NA
temp$IPH[IPH.bin == "no"] <- "control"
temp$IPH[IPH.bin == "yes"] <- "case"

temp[,"PVI"] <- NA
temp$PVI[Plaque_Vulnerability_Index == "0"] <- "PVI_cat0"
temp$PVI[Plaque_Vulnerability_Index == "1"] <- "PVI_cat1"
temp$PVI[Plaque_Vulnerability_Index == "2"] <- "PVI_cat2"
temp$PVI[Plaque_Vulnerability_Index == "3"] <- "PVI_cat3"
temp$PVI[Plaque_Vulnerability_Index == "4"] <- "PVI_cat4"
temp$PVI[Plaque_Vulnerability_Index == "5"] <- "PVI_cat5"

temp$Plaque_Vulnerability_Index <- temp$PVI
temp$PVI <- NULL

detach(temp)

# Making selection variable
attach(temp)
temp[,"SELECTION"] <- "not_selected"
temp$SELECTION[(QC2018_FILTER=="passed" | QC2018_FILTER=="family_keep") & (STUDY_TYPE=="CEA" & PCA=="EUR") & Sex=="F"] <- "selected"
detach(temp)
table(temp$SELECTION, temp$QC2018_FILTER)
table(temp$SELECTION, temp$STUDY_TYPE)
table(temp$SELECTION, temp$PCA)
```


```{r Save SampleSelect FEMALES}
AEGS123_sample.list <- temp[order(temp$OriginalOrder_postMichImp_QC),]

AEGS123_sample.list$missing <- 0

sample_file_aegsF <- dplyr::select(AEGS123_sample.list,
                                  ID_1, ID_2, missing, # ID_2 is the order of samples - that way we always know what the order should be
                                  UPID, STUDY_NUMBER, 
                                  QC2018_FINAL, QC2018_FILTER, SELECTION,
                                  AEGS_type, CHIP, STUDY_TYPE, SAMPLE_TYPE, PCA,
                                  PC1, PC2, PC3, PC4, PC5, PC6, PC7, PC8, PC9, PC10,
                                  Sex, Age, ORyear, 
                                  Calcification, Collagen, 
                                  Fat10, Fat40, IPH, 
                                  SMC_rankNorm, MAC_rankNorm, Neutrophils_rankNorm, MastCells_rankNorm, VesselDensity_rankNorm,
                                  Plaque_Vulnerability_Index,
                                  PCSK9_plasma, PCSK9_plasma_rankNorm)  %>%
  mutate_if(is.numeric, as.character) %>%
  mutate(SAMPLE_TYPE = gsub(' ', '_', SAMPLE_TYPE)) %>%
  add_row(.before = 1, 
          ID_1 = "0", ID_2 = "0", missing = "0", 
          UPID = "D", STUDY_NUMBER = "C",
          QC2018_FINAL = "D", QC2018_FILTER = "D", SELECTION = "D",
          AEGS_type = "D", CHIP = "D", STUDY_TYPE = "D", SAMPLE_TYPE = "D", PCA = "D",
          PC1 = "C", PC2 = "C", PC3 = "C", PC4 = "C", PC5 = "C", PC6 = "C", PC7 = "C", PC8 = "C", PC9 = "C", PC10 = "C",
          Sex = "D", Age = "C", ORyear = "C", 
          Calcification = "B", Collagen = "B", 
          Fat10 = "B", Fat40 = "B", IPH = "B", 
          SMC_rankNorm = "P", MAC_rankNorm = "P", Neutrophils_rankNorm = "P", MastCells_rankNorm = "P", VesselDensity_rankNorm = "P",
          Plaque_Vulnerability_Index = "D",
          PCSK9_plasma = "P", PCSK9_plasma_rankNorm = "P") %>% ## identifiers: index for these is 1, and all base variables have 0 as identifier
  print()
dim(sample_file_aegsF)

fwrite(sample_file_aegsF,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.females.sample"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

require(DT)
DT::datatable(sample_file_aegsF, caption = "AEGS: final sample list of genotyped AE patients after quality control.", rownames = FALSE)

rm(temp)
```

### Males only

This is the selection for males only.

```{r TEMP SampleSelect MALES}
require(openxlsx)

temp <- subset(AEGS,
               GWAS == "genotyped",
               select = c("ID_1", "ID_2", "UPID", "STUDY_NUMBER", # ID_2 is the order of samples!
                          "QC2018_FINAL", "QC2018_FILTER", "OriginalOrder_postMichImp_QC",
                          "AEGS_type", "CHIP", "STUDY_TYPE", "SAMPLE_TYPE", "PCA",
                          "PC1", "PC2", "PC3", "PC4", "PC5",
                          "PC6", "PC7", "PC8", "PC9", "PC10",
                          "Sex", "Age", "ORyear", 
                          "BMI",
                          "Calc.bin", "Collagen.bin", 
                          "Fat.bin_10", "Fat.bin_40", "IPH.bin", 
                          "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm", 
                          "OverallPlaquePhenotype", "Plaque_Vulnerability_Index",
                          "PCSK9_plasma", "PCSK9_plasma_rankNorm")) # Select some phenotype of interest
dim(temp)

# Fix things
attach(temp)

temp[,"Calcification"] <- NA
temp$Calcification[Calc.bin == "no/minor"] <- "control"
temp$Calcification[Calc.bin == "moderate/heavy"] <- "case"

temp[,"Collagen"] <- NA
temp$Collagen[Collagen.bin == "no/minor"] <- "control"
temp$Collagen[Collagen.bin == "moderate/heavy"] <- "case"

temp[,"Fat10"] <- NA
temp$Fat10[Fat.bin_10 == "<10%"] <- "control"
temp$Fat10[Fat.bin_10 == ">10%"] <- "case"

temp[,"Fat40"] <- NA
temp$Fat40[Fat.bin_40 == "<40%"] <- "control"
temp$Fat40[Fat.bin_40 == ">40%"] <- "case"

temp[,"IPH"] <- NA
temp$IPH[IPH.bin == "no"] <- "control"
temp$IPH[IPH.bin == "yes"] <- "case"

temp[,"PVI"] <- NA
temp$PVI[Plaque_Vulnerability_Index == "0"] <- "PVI_cat0"
temp$PVI[Plaque_Vulnerability_Index == "1"] <- "PVI_cat1"
temp$PVI[Plaque_Vulnerability_Index == "2"] <- "PVI_cat2"
temp$PVI[Plaque_Vulnerability_Index == "3"] <- "PVI_cat3"
temp$PVI[Plaque_Vulnerability_Index == "4"] <- "PVI_cat4"
temp$PVI[Plaque_Vulnerability_Index == "5"] <- "PVI_cat5"

temp$Plaque_Vulnerability_Index <- temp$PVI
temp$PVI <- NULL

detach(temp)

# Making selection variable
attach(temp)
temp[,"SELECTION"] <- "not_selected"
temp$SELECTION[(QC2018_FILTER=="passed" | QC2018_FILTER=="family_keep") & (STUDY_TYPE=="CEA" & PCA=="EUR") & Sex=="M"] <- "selected"
detach(temp)
table(temp$SELECTION, temp$QC2018_FILTER)
table(temp$SELECTION, temp$STUDY_TYPE)
table(temp$SELECTION, temp$PCA)
```


```{r Save SampleSelect MALES}
AEGS123_sample.list <- temp[order(temp$OriginalOrder_postMichImp_QC),]

AEGS123_sample.list$missing <- 0

sample_file_aegsM <- dplyr::select(AEGS123_sample.list,
                                  ID_1, ID_2, missing, # ID_2 is the order of samples - that way we always know what the order should be
                                  UPID, STUDY_NUMBER, 
                                  QC2018_FINAL, QC2018_FILTER, SELECTION,
                                  AEGS_type, CHIP, STUDY_TYPE, SAMPLE_TYPE, PCA,
                                  PC1, PC2, PC3, PC4, PC5, PC6, PC7, PC8, PC9, PC10,
                                  Sex, Age, ORyear, 
                                  Calcification, Collagen, 
                                  Fat10, Fat40, IPH, 
                                  SMC_rankNorm, MAC_rankNorm, Neutrophils_rankNorm, MastCells_rankNorm, VesselDensity_rankNorm,
                                  Plaque_Vulnerability_Index,
                                  PCSK9_plasma, PCSK9_plasma_rankNorm)  %>%
  mutate_if(is.numeric, as.character) %>%
  mutate(SAMPLE_TYPE = gsub(' ', '_', SAMPLE_TYPE)) %>%
  add_row(.before = 1, 
          ID_1 = "0", ID_2 = "0", missing = "0", 
          UPID = "D", STUDY_NUMBER = "C",
          QC2018_FINAL = "D", QC2018_FILTER = "D", SELECTION = "D",
          AEGS_type = "D", CHIP = "D", STUDY_TYPE = "D", SAMPLE_TYPE = "D", PCA = "D",
          PC1 = "C", PC2 = "C", PC3 = "C", PC4 = "C", PC5 = "C", PC6 = "C", PC7 = "C", PC8 = "C", PC9 = "C", PC10 = "C",
          Sex = "D", Age = "C", ORyear = "C", 
          Calcification = "B", Collagen = "B", 
          Fat10 = "B", Fat40 = "B", IPH = "B", 
          SMC_rankNorm = "P", MAC_rankNorm = "P", Neutrophils_rankNorm = "P", MastCells_rankNorm = "P", VesselDensity_rankNorm = "P",
          Plaque_Vulnerability_Index = "D",
          PCSK9_plasma = "P", PCSK9_plasma_rankNorm = "P") %>% ## identifiers: index for these is 1, and all base variables have 0 as identifier
  print()
dim(sample_file_aegsM)


fwrite(sample_file_aegsM,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.males.sample"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

require(DT)
DT::datatable(sample_file_aegsM, caption = "AEGS: final sample list of genotyped AE patients after quality control.", rownames = FALSE)

rm(temp)
```

## X-chromosome data

The X-chromosome data is taken from previously imputed data based on 1000G phase 3 (version 5) and GoNL5. For some reason, imputing on the Michigan Imputation Server was not successful (*ACTION point*).

Here we load in the sample files for the three datasets of the X chromosomal data. We should:

- filter out samples that did not pass quality control, ending up with 2,124 sample
- re-order the data to fit the other autosomal data.


```{r SampleList X}

AEGS123_chrX <- fread(paste0(MICHIMP_loc, "/_chr23_1kg_gonl5/aegs.raw.1kg_gonl5.chr23.mappings.txt"))
names(AEGS123_chrX)[names(AEGS123_chrX) == "ID_1"] <- "SampleID_postImpChrX"

AEGS123_AllChr <- merge(AEGS123_chrX, sample_file_aegs, by.x = "SampleID_postMichImp", by.y = "ID_1", 
                        all.x = TRUE, 
                        sort = FALSE)

names(AEGS123_AllChr)[names(AEGS123_AllChr) == "ID_2.y"] <- "ID_2"
names(AEGS123_AllChr)[names(AEGS123_AllChr) == "STUDY_TYPE.y"] <- "STUDY_TYPE"
names(AEGS123_AllChr)[names(AEGS123_AllChr) == "SampleID_postMichImp"] <- "ID_1"
AEGS123_AllChr$missing.x <- NULL
AEGS123_AllChr$missing.y <- NULL
AEGS123_AllChr$STUDY_TYPE.x <- NULL
AEGS123_AllChr$ID_2.x <- NULL

dim(AEGS123_AllChr)
str(AEGS123_AllChr)
```

This seems fine, let's filter; we can use this file to filter the genetic data. And we create another file to re-order the data.

```{r SampleList X: filter}

AEGS123_AllChrQC <- subset(AEGS123_AllChr,
               !is.na(QC2018_FILTER),
               select = c("ID_1", "ID_2", "UPID", "STUDY_NUMBER", "SampleID_postImpChrX",
                          "QC2018_FINAL", "QC2018_FILTER", "SELECTION",
                          "AEGS_type", "CHIP", "STUDY_TYPE", "SAMPLE_TYPE",
                          "PC1", "PC2", "PC3", "PC4", "PC5",
                          "PC6", "PC7", "PC8", "PC9", "PC10",
                          "Sex", "Age", "ORyear", 
                          "Calcification", "Collagen", 
                          "Fat10", "Fat40", "IPH", 
                          "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm",
                          "Plaque_Vulnerability_Index",
                          "PCSK9_plasma", "PCSK9_plasma_rankNorm"))

AEGS123_AllChrQC_reorder <-AEGS123_AllChrQC[order(AEGS123_AllChrQC$ID_2),] # remember: ID_2 is the order of samples

AEGS123_AllChrQC_filtered <- subset(AEGS123_AllChr,
               is.na(QC2018_FILTER),
               select = c("ID_1", "ID_2", "UPID", "STUDY_NUMBER", "SampleID_postImpChrX",
                          "QC2018_FINAL", "QC2018_FILTER", "SELECTION",
                          "AEGS_type", "CHIP", "STUDY_TYPE", "SAMPLE_TYPE",
                          "PC1", "PC2", "PC3", "PC4", "PC5",
                          "PC6", "PC7", "PC8", "PC9", "PC10",
                          "Sex", "Age", "ORyear", 
                          "Calcification", "Collagen", 
                          "Fat10", "Fat40", "IPH", 
                          "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm",
                          "Plaque_Vulnerability_Index",
                          "PCSK9_plasma", "PCSK9_plasma_rankNorm"))

fwrite(AEGS123_AllChrQC_reorder,
       file = paste0(SNP_loc, "/",Today,".",PROJECTNAME,".AEGS123.chrX.sample"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = TRUE,
       showProgress = TRUE, verbose = TRUE)

require(DT)
DT::datatable(AEGS123_AllChrQC, caption = "AEGS: final sample list of genotyped AE patients after quality control (chromosome X).", rownames = FALSE)


```

# GWASToolKit preparation

## VariantLists: top SNPs

Here we create a `variantlist.txt` file used by **GWASToolKit** for analysis.These are the variants with LD r<sup>2</sup> => 0.6 around each of the 15 top SNPs associated with CAC.

```{r create variantList}
variant_list

temp <- subset(variant_list, select = c("VariantID", "Chr", "BP"))

fwrite(temp,
       file = paste0(SNP_loc, "/variantlist.txt"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = FALSE,
       showProgress = TRUE, verbose = TRUE)
rm(temp)
```
<!-- ## VariantLists: Credible Set -->

<!-- Here we create a `crediblesetlist.txt` file used by **GWASToolKit** for analysis.These are the variants in the 95% credible set of causal variants for the 15 top SNPs associated with CAC. -->

<!-- ```{r create credset_list} -->
<!-- credset_list -->

<!-- temp <- subset(credset_list, select = c("VariantID", "Chr", "BP")) -->

<!-- fwrite(temp, -->
<!--        file = paste0(SNP_loc, "/crediblesetlist.txt"), -->
<!--        na = "NA", sep = "\t", quote = FALSE, -->
<!--        row.names = FALSE, col.names = FALSE, -->
<!--        showProgress = TRUE, verbose = TRUE) -->
<!-- rm(temp) -->
<!-- ``` -->

## Gene list

We export the list of genes for analysis in **GWASToolKit**.

This code is to filter the list from genes that are not in the data.

- _FBIN1_ ==> not found
- _C1S1_ ==> not found
- _FBIN1_ ==> not found
 _C1S1_ ==> not found
 _SERPIND2_ ==> not found
 _C1RA_ ==> not found
 _SEPINA1A_ ==> not found
 _TF_ ==> not found
 _SERPINA1B_ ==> not found
 _SERPINA1D_ ==> not found
 _PTXDC2_ ==> not found
 _KNG2_ ==> not found
 _SERPINA3N_ ==> not found
 _LBP_ ==> not found
 _RBP4_ ==> not found


```{r filter target genes}
gene_list_rm <- c("FBIN1", 
                  "C1S1", 
                  "SERPIND2", 
                  "C1RA", 
                  "SEPINA1A", 
                  "TF",
                  "SERPINA1B",
                  "SERPINA1D",
                  "PTXDC2",
                  "KNG2", 
                  "SERPINA3N",
                  "LBP",
                  "RBP4") 

temp = gene_list[!gene_list %in% gene_list_rm]

gene_list_qc <- c(temp)

# gene_list_qc <- gene_list
# 
# for debug
# gene_list_qc_replace <- c("MRTFA")
```

Here we export this list, 1 gene per line. 

```{r create genelist}
variant_list

# temp <- subset(gene_list_df, select = c("Gene"))
temp <- as.data.frame(gene_list_qc)
fwrite(temp,
       file = paste0(SNP_loc, "/genelist.txt"),
       na = "NA", sep = "\t", quote = FALSE,
       row.names = FALSE, col.names = FALSE,
       showProgress = TRUE, verbose = TRUE)
rm(temp)
```

## Covariates

Here we create a `covariates.txt` file used by **GWASToolKit** for analysis.

```{r create covariatesList}
library(tidyverse)
# for 'overall' analyses
c("Age Sex PC1 PC2 ORyear") %>% write_lines(paste0(SNP_loc, "/covariates.txt"))

# for sex-specific analyses
c("Age PC1 PC2 ORyear") %>% write_lines(paste0(SNP_loc, "/covariates.sex.txt"))
```

## Phenotypes

Here we create a `phenotypes.txt` file used by **GWASToolKit** for analysis.

```{r create phenotypesList}
library(tidyverse)
c("Calcification", "Collagen", "Fat10", "Fat40", "IPH", "SMC_rankNorm", "MAC_rankNorm", "Neutrophils_rankNorm", "MastCells_rankNorm", "VesselDensity_rankNorm") %>% write_lines(paste0(SNP_loc, "/phenotypes.txt"))

c("Plaque_Vulnerability_Index") %>% write_lines(paste0(SNP_loc, "/phenotypes.pvi.txt"))


```


# Session information

------------------------------------------------------------------------------------------------------------------------

    Version:      v1.2.2
    Last update:  2024-01-09
    Written by:   Sander W. van der Laan (s.w.vanderlaan-2[at]umcutrecht.nl).
    Description:  Script to get some Athero-Express Biobank Study baseline characteristics.
    Minimum requirements: R version 3.4.3 (2017-06-30) -- 'Single Candle', Mac OS X El Capitan

    Changes log
    * v1.2.2 Textual fixes.
    * v1.2.1 Fixed issue with baseline writing.
    * v1.2.0 Update to the study database. 
    * v1.1.0 Major update to WORCS system. 
    * v1.0.6 Small bug fixes.
    * v1.0.5 Added png for overlap-figure.
    * v1.0.5 Removed obsolete references to objects.
    * v1.0.4 Fixed a mistake in the chr X sample-file creation. Now the order matches the chr X data.
    * v1.0.3 Fixed weight of files (limit of 10Mb per file for templates). Renamed entire repo.
    * v1.0.2 Added sex-specific .sample-files. Added GWASToolKit input-files.
    * v1.0.0 Initial version. Add 'plaque vulnerability index', Fixed baseline table, added codes, and results. Created sample-files.

------------------------------------------------------------------------------------------------------------------------

```{r eval = TRUE}
sessionInfo()
```

# Saving environment

```{r Saving}
save.image(paste0(PROJECT_loc, "/",Today,".",PROJECTNAME,".SNP_analyses.RData"))
```

+-----------------------------------------------------------------------------------------------------------------------------------------+
| <sup>© 1979-2024 Sander W. van der Laan | s.w.vanderlaan[at]gmail.com | [vanderlaanand.science](https://vanderlaanand.science).</sup> |
+-----------------------------------------------------------------------------------------------------------------------------------------+
