---
title: "Mapping targets to single cells in plaques."
subtitle: Accompanying 'AE_TEMPLATE'
author: '[Sander W. van der Laan, PhD](https://vanderlaanand.science) | s.w.vanderlaan[at]gmail[dot]com'
date: '`r Sys.Date()`'
output:
  html_notebook: 
    cache: yes
    code_folding: hide
    collapse: yes
    df_print: paged
    fig.align: center
    fig_caption: yes
    fig_height: 10
    fig_retina: 2
    fig_width: 12
    theme: paper
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
    highlight: tango
mainfont: Helvetica
editor_options:
  chunk_output_type: inline
bibliography: references.bib
knit: worcs::cite_all
---

# General Setup
```{r echo = FALSE}
rm(list = ls())
```

```{r LocalSystem, echo = FALSE}
source("scripts/local.system.R")
```

```{r Source functions}
source("scripts/functions.R")
```

```{r loading_packages, message=FALSE, warning=FALSE}
source("scripts/pack03.packages.R")
```

```{r Setting: Colors}
Today = format(as.Date(as.POSIXlt(Sys.time())), "%Y%m%d")
Today.Report = format(as.Date(as.POSIXlt(Sys.time())), "%A, %B %d, %Y")
source("scripts/colors.R")
```

```{r setup_notebook, include=FALSE}
# We recommend that you prepare your raw data for analysis in 'prepare_data.R',
# and end that file with either open_data(yourdata), or closed_data(yourdata).
# Then, uncomment the line below to load the original or synthetic data
# (whichever is available), to allow anyone to reproduce your code:
# load_data()

# further define some knitr-options.
knitr::opts_chunk$set(fig.width = 12, fig.height = 8, fig.path = 'Figures/', 
                      warning = TRUE, # show warnings during codebook generation
                      message = TRUE, # show messages during codebook generation
                      error = TRUE, # do not interrupt codebook generation in case of errors, 
                                    # usually better for debugging
                      echo = TRUE,  # show R code
                      eval = TRUE)

ggplot2::theme_set(ggplot2::theme_minimal())
# pander::panderOptions("table.split.table", Inf)
library("worcs")
library("rmarkdown")

```

# ERA-CVD 'druggable-MI-targets'

<!-- ![ERA-CVD logo]("Users/swvanderlaan/iCloud/Genomics/Projects/#Druggable-MI-Genes/Administration/ERA-CVD\ Logo_CMYK.jpg") -->

For the ERA-CVD 'druggable-MI-targets' project (grantnumber: 01KL1802) we performed two related RNA sequencing (RNAseq) experiments:

1)  conventional ('bulk') RNAseq using RNA extracted from carotid plaque samples, n ± 700. As of `r Today.Report` all samples have been selected and
RNA has been extracted; quality control (QC) was performed and we have a dataset of 635 samples.

2)  single-cell RNAseq (scRNAseq) of at least n = 40 samples (20 females, 20 males). As of `r Today.Report` data is available of 40 samples (3 females, 15 males), we are extending sampling to get more female samples.

Plaque samples are derived from carotid endarterectomies as part of the [Athero-Express Biobank Study](http:www/atheroexpress.nl) which is an ongoing study in the UMC Utrecht.

# Background

Here we map the `r TRAIT_OF_INTEREST` to single-cells from the plaques.

## Targets

Here we obtain data from the `r TRAIT_OF_INTEREST` in plaques.

```{r targets, message=FALSE, warning=FALSE}
library(openxlsx)

gene_list_df <- read.xlsx(paste0(PROJECT_loc, "/targets/targets.xlsx"), sheet = "Genes")

gene_list <- unlist(gene_list_df$Gene)
gene_list

```


# Load data

First we will load the data:

-   scRNAseq experimental data and rename the cell types.
-   Athero-Express clinical data.

Here we load the latest dataset from our Athero-Express single-cell RNA experiment.

```{r LoadData}

# load(paste0(AESCRNA_loc, "/20210811.46.patients.KP.RData"))
# scRNAseqData <- seuset
# rm(seuset)
# 
# saveRDS(scRNAseqData, paste0(AESCRNA_loc, "/20210811.46.patients.KP.RDS"))

scRNAseqData <- readRDS(paste0(AESCRNA_loc, "/20210811.46.patients.KP.RDS"))

scRNAseqData

```

The naming/classification is based on a combination conventional markers. We do not claim to know the exact identity of each cell, rather we refer to cells as 'KIT+ Mast cells"-like cells. Likewise we refer to the cell clusters as 'communities' of cells that exhibit similar properties, *i.e.* similar defining markers (*e.g. KIT*).

We will rename the cell types to human readable names.

```{r Change cell cummunity names}
### change names for clarity
backup.scRNAseqData = scRNAseqData
# get the old names to change to new names
UMAPPlot(scRNAseqData, label = FALSE, pt.size = 1.25, label.size = 4, group.by = "ident")

```

```{r}
unique(scRNAseqData@active.ident)
```

```{r}
celltypes <- c("CD68+CD4+ Monocytes" = "CD68+CD4+ Mono", 
               "CD68+IL18+TLR4+TREM2+ Resident macrophages" = "CD68+IL18+TLR4+TREM2+ MRes", 
               "CD68+CD1C+ Dendritic Cells" = "CD68+CD1C+ DC",
               "CD68+CASP1+IL1B+SELL+ Inflammatory macrophages" = "CD68+CASP1+IL1B+SELL MInf",
               "CD68+ABCA1+OLR1+TREM2+ Foam Cells" = "CD68+ABCA1+OLR1+TREM2+ FC",
               
               # T-cells
               "CD3+ T Cells I" = "CD3+ TC I",
               "CD3+ T Cells II" = "CD3+ TC II", 
               "CD3+ T Cells III" = "CD3+ TC III", 
               "CD3+ T Cells IV" = "CD3+ TC IV", 
               "CD3+ T Cells V" = "CD3+ TC V", 
               "CD3+ T Cells VI" = "CD3+ TC VI", 
               "FOXP3+ T Cells" = "FOXP3+ TC",
               
               # Endothelial cells
               "CD34+ Endothelial Cells I" = "CD34+ EC I", 
               "CD34+ Endothelial Cells II" = "CD34+ EC II", 
               
               # SMC
               "ACTA2+ Smooth Muscle Cells" = "ACTA2+ SMC", 
               
               # NK Cells
               "CD3+CD56+ NK Cells I" = "CD3+CD56+ NK I",
               "CD3+CD56+ NK Cells II" = "CD3+CD56+ NK II",
               # Mast
               "CD68+KIT+ Mast Cells" = "CD68+KIT+ MC",
               
               "CD79A+ Class-switched Memory B Cells" = "CD79A+ BCmem", 
               "CD79+ Plasma B Cells" = "CD79+ BCplasma")

scRNAseqData <- Seurat::RenameIdents(object = scRNAseqData, 
                                       celltypes)
```

```{r Change cell cummunity names - new plot}
UMAPPlot(scRNAseqData, label = TRUE, pt.size = 1.25, label.size = 4, group.by = "ident",
         repel = TRUE)

```

## Clinical data

Loading the Athero-Express clinical data.

```{r LoadAEDB}
AEDB.CEA <- readRDS(file = paste0(OUT_loc, "/", Today,".",PROJECTNAME,".AEDB.CEA.RDS"))
# AEDB.CEA <- readRDS(file = paste0(OUT_loc, "/20240531.",PROJECTNAME,".AEDB.CEA.RDS"))

```


```{r }
# Baseline table variables
basetable_vars = c("Hospital", "ORyear", "Artery_summary",
                   "Age", "Gender", 
                   # "TC_finalCU", "LDL_finalCU", "HDL_finalCU", "TG_finalCU", 
                   "TC_final", "LDL_final", "HDL_final", "TG_final", 
                   # "hsCRP_plasma",
                   "systolic", "diastoli", "GFR_MDRD", "BMI", 
                   "KDOQI", "BMI_WHO",
                   "SmokerStatus", "AlcoholUse",
                   "DiabetesStatus", 
                   "Hypertension.selfreport", "Hypertension.selfreportdrug", "Hypertension.composite", "Hypertension.drugs", 
                   "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD", 
                   "Stroke_Dx", "sympt", "Symptoms.5G", "AsymptSympt", "AsymptSympt2G",
                   "Symptoms.Update2G", "Symptoms.Update3G", "indexsymptoms_latest_4g",
                   "restenos", "stenose", 
                   "CAD_history", "PAOD", "Peripheral.interv", 
                   "EP_composite", "EP_composite_time", "EP_major", "EP_major_time",
                   "MAC_rankNorm", "SMC_rankNorm", "Macrophages.bin", "SMC.bin",
                   "Neutrophils_rankNorm", "MastCells_rankNorm",
                   "IPH.bin", "VesselDensity_rankNorm",
                   "Calc.bin", "Collagen.bin", 
                   "Fat.bin_10", "Fat.bin_40", "OverallPlaquePhenotype", "Plaque_Vulnerability_Index",
                   "PCSK9_plasma", "PCSK9_plasma_rankNorm")

basetable_bin = c("Gender",  "Artery_summary",
                  "KDOQI", "BMI_WHO",
                  "SmokerStatus", "AlcoholUse",
                  "DiabetesStatus", 
                  "Hypertension.selfreport", "Hypertension.selfreportdrug", "Hypertension.composite", "Hypertension.drugs", 
                  "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD", 
                  "Stroke_Dx", "sympt", "Symptoms.5G", "AsymptSympt", "AsymptSympt2G",
                  "Symptoms.Update2G", "Symptoms.Update3G", "indexsymptoms_latest_4g",
                  "restenos", "stenose",
                  "CAD_history", "PAOD", "Peripheral.interv", 
                  "EP_major", "EP_composite", "Macrophages.bin", "SMC.bin",
                  "IPH.bin", 
                  "Calc.bin", "Collagen.bin", 
                  "Fat.bin_10", "Fat.bin_40", "OverallPlaquePhenotype", "Plaque_Vulnerability_Index")
# basetable_bin

basetable_con = basetable_vars[!basetable_vars %in% basetable_bin]
# basetable_con
```

## AESCRNA: baseline characteristics

### Preparation

```{r Baseline: creation}
metadata <- scRNAseqData@meta.data %>% as_tibble() %>% separate(orig.ident, c("Patient", NA))
scRNAseqDataMeta <- metadata %>% distinct(Patient, .keep_all = TRUE)

scRNAseqDataMetaAE <- merge(scRNAseqDataMeta, AEDB.CEA, by.x = "Patient", by.y = "STUDY_NUMBER", sort = FALSE, all.x = TRUE)
dim(scRNAseqDataMetaAE)

# Replace missing data 
# Ref: https://cran.r-project.org/web/packages/naniar/vignettes/replace-with-na.html
require(naniar)

na_strings <- c("NA", "N A", "N / A", "N/A", "N/ A", 
                "Not Available", "Not available", 
                "missing", 
                "-999", "-99", 
                "No data available/missing", "No data available/Missing")
# Then you write ~.x %in% na_strings - which reads as “does this value occur in the list of NA strings”.

scRNAseqDataMetaAE %>%
  replace_with_na_all(condition = ~.x %in% na_strings)
```

```{r }
cat("====================================================================================================")
cat("SELECTION THE SHIZZLE")

cat("- sanity checking PRIOR to selection")
library(data.table)
require(labelled)
ae.gender <- to_factor(scRNAseqDataMetaAE$Gender)
ae.hospital <- to_factor(scRNAseqDataMetaAE$Hospital)
table(ae.gender, ae.hospital, dnn = c("Sex", "Hospital"), useNA = "ifany")

ae.artery <- to_factor(scRNAseqDataMetaAE$Artery_summary)
table(ae.artery, ae.gender, dnn = c("Sex", "Artery"), useNA = "ifany")

ae.ic <- to_factor(scRNAseqDataMetaAE$informedconsent)
table(ae.ic, ae.gender, useNA = "ifany")

rm(ae.gender, ae.hospital, ae.artery, ae.ic)


scRNAseqDataMetaAE.all <- subset(scRNAseqDataMetaAE,
                                 (Artery_summary == "carotid (left & right)" | Artery_summary == "other carotid arteries (common, external)" ) & # we only want carotids
                                   informedconsent != "missing" & # we are really strict in selecting based on 'informed consent'!
                                   informedconsent != "no, died" &
                                   informedconsent != "yes, no tissue, no commerical business" &
                                   informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" &
                                   informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" &
                                   informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" &
                                   informedconsent != "yes, no tissue, no health treatment" &
                                   informedconsent != "yes, no tissue, no questionnaires" &
                                   informedconsent != "yes, no tissue, health treatment when possible" &
                                   informedconsent != "yes, no tissue" &
                                   informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info" &
                                   informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" &
                                   informedconsent != "no, doesn't want to" &
                                   informedconsent != "no, unable to sign" &
                                   informedconsent != "no, no reaction" &
                                   informedconsent != "no, lost" &
                                   informedconsent != "no, too old" &
                                   informedconsent != "yes, no medical info, health treatment when possible" & 
                                   informedconsent != "no (never asked for IC because there was no tissue)" &
                                   informedconsent != "no, endpoint" &
                                   informedconsent != "nooit geincludeerd" & 
                                   informedconsent != "yes, no health treatment, no commercial business" & # IMPORTANT: since we are sharing with a commercial party
                                   informedconsent != "yes, no tissue, no commerical business" & 
                                   informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" & 
                                   informedconsent != "yes, no questionnaires, no health treatment, no commercial business" & 
                                   informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" & 
                                   informedconsent != "yes, no health treatment, no medical info, no commercial business" & 
                                   informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" & 
                                   informedconsent != "yes, no commerical business" & 
                                   informedconsent != "yes, health treatment when possible, no commercial business" & 
                                   informedconsent != "yes, no medical info, no commercial business" & 
                                   informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" & 
                                   informedconsent != "yes, no questionnaires, no commercial business" & 
                                   informedconsent != "yes, no questionnaires, health treatment when possible, no commercial business" & 
                                   informedconsent != "second informed concents: yes, no commercial business")
# scRNAseqDataMetaAE.all[1:10, 1:10]
dim(scRNAseqDataMetaAE.all)
# DT::datatable(scRNAseqDataMetaAE.all)

```

### Baseline

Showing the baseline table for the scRNAseq data in 39 CEA patients with
informed consent.

```{r Baseline: Visualize}
cat("===========================================================================================")
cat("CREATE BASELINE TABLE")

# Create baseline tables
# http://rstudio-pubs-static.s3.amazonaws.com/13321_da314633db924dc78986a850813a50d5.html
scRNAseqDataMetaAE.all.tableOne = print(CreateTableOne(vars = basetable_vars, 
                                                  # factorVars = basetable_bin,
                                                  # strata = "Gender",
                                                  data = scRNAseqDataMetaAE.all, includeNA = TRUE), 
                                   nonnormal = c(), 
                                   quote = FALSE, showAllLevels = TRUE,
                                   format = "p", 
                                   contDigits = 3)[,1:2]

```

Writing the baseline table to Excel format.

```{r }
# Write basetable
require(openxlsx)
# write.xlsx(file = paste0(BASELINE_loc, "/",Today,".",PROJECTNAME,".AESCRNA.32pts.after_qc.IC_commercial.BaselineTable.xlsx"), 
#            format(as.data.frame(scRNAseqDataMetaAE.all.tableOne), digits = 5, scientific = FALSE),
#            rowNames = TRUE, colNames = TRUE, 
#            sheetName = "AESCRNA", overwrite = TRUE)
write.xlsx(file = paste0(BASELINE_loc, "/",Today,".",PROJECTNAME,".AESCRNA.CEA.39pts.after_qc.IC_academic.BaselineTable.xlsx"),
           format(as.data.frame(scRNAseqDataMetaAE.all.tableOne), digits = 5, scientific = FALSE),
           rowNames = TRUE, colNames = TRUE,
           sheetName = "AESCRNA_CEA", overwrite = TRUE)

```


# Subset scRNAseq data

List of samples to be included based on informed consent (see above).

```{r}
samples_of_interest <- unlist(scRNAseqDataMetaAE.all$Patient)
samples_of_interest
```

Just make sure to keep the rows (`patientid`.`plateid`.`cellid`) in the data, so we will create a new column for that.

```{r}
scRNAseqData$cell.ident <- rownames(scRNAseqData[[]])
```

Subsetting the samples of interest based on `samples_of_interest`.

```{r}
scRNAseqDataCEA39 <- subset(scRNAseqData, subset = Patient %in% samples_of_interest)
```


Selecting the clinical data of interest.

```{r}
variables_of_interest <- c("Hospital", "ORyear", "Artery_summary",
                           "Age", "Gender",
                           "TC_final", "LDL_final", "HDL_final", "TG_final",
                           "systolic", "diastoli", "GFR_MDRD", "BMI",
                           "KDOQI", "BMI_WHO",
                           "SmokerStatus", "AlcoholUse",
                           "DiabetesStatus",
                           "Hypertension.selfreport", "Hypertension.selfreportdrug", "Hypertension.composite", "Hypertension.drugs",
                           "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD",
                           "Stroke_Dx",
                           "sympt", "Symptoms.5G", "AsymptSympt", "AsymptSympt2G",
                           "Symptoms.Update2G", "Symptoms.Update3G", "indexsymptoms_latest_4g",
                           "restenos", "stenose",
                           "CAD_history", "PAOD", "Peripheral.interv",
                           "EP_composite", "EP_composite_time", "EP_major", "EP_major_time")

temp <- subset(scRNAseqDataMetaAE.all, select = c("Patient", variables_of_interest))
```

Now we are merging in the clinical data of interest.

```{r}
# Step 1: Extract original metadata from Seurat object
original_metadata_df <- scRNAseqDataCEA39@meta.data

# Step 2: Merge with dataframe (temp) with select variables based on 'Patient' column
merged_metadata_df <- merge(original_metadata_df, temp, by = "Patient", all.x = TRUE, sort = FALSE)

# Step 3: Exclude already existing columns to avoid duplication
new_metadata_columns <- setdiff(colnames(temp), colnames(original_metadata_df))
new_meta_info_df <- merged_metadata_df[, new_metadata_columns, drop = FALSE]

# Step 4: Add the new metadata to the Seurat object
scRNAseqDataCEA39 <- AddMetaData(scRNAseqDataCEA39, metadata = new_meta_info_df)

# Step 5: Rename `Patient` to `STUDY_NUMBER` to match the clinical data
scRNAseqDataCEA39@meta.data <- dplyr::rename(scRNAseqDataCEA39@meta.data, "STUDY_NUMBER" = "Patient")

# Verify that the new metadata was added correctly
head(scRNAseqDataCEA39@meta.data)

```

## Saving new dataset

```{r}
temp2 <- as_tibble(subset(scRNAseqDataCEA39@meta.data, select = c("STUDY_NUMBER", "orig.ident", "nCount_RNA", "nFeature_RNA",
                                                                 "Plate", "Batch", "C.H", "Type", "percent.mt",
                                                                 "nCount_SCT", "nFeature_SCT", "seurat_clusters")))

# fwrite(temp2,
#        file = paste0(OUT_loc, "/", Today, ".AESCRNA.CEA.39pts.samplelist.after_qc.IC_commercial.csv"),
#        sep = ",", row.names = FALSE, col.names = TRUE,
#        showProgress = TRUE)
# rm(temp2)
# 
# temp <- dplyr::rename(temp, "STUDY_NUMBER" = "Patient")
# fwrite(temp,
#        file = paste0(OUT_loc, "/", Today, ".AESCRNA.CEA.39pts.clinicaldata.after_qc.IC_commercial.csv"),
#        sep = ",", row.names = FALSE, col.names = TRUE,
#        showProgress = TRUE)
# rm(temp)
# 
# saveRDS(scRNAseqDataCEA39, file = paste0(OUT_loc, "/", Today, ".AESCRNA.CEA.39pts.Seurat.after_qc.IC_commercial.RDS"))

fwrite(temp2,
       file = paste0(OUT_loc, "/", Today, ".AESCRNA.CEA.39pts.samplelist.after_qc.IC_academic.csv"),
       sep = ",", row.names = FALSE, col.names = TRUE,
       showProgress = TRUE)
rm(temp2)

temp <- dplyr::rename(temp, "STUDY_NUMBER" = "Patient")
fwrite(temp,
       file = paste0(OUT_loc, "/", Today, ".AESCRNA.CEA.39pts.clinicaldata.after_qc.IC_academic.csv"),
       sep = ",", row.names = FALSE, col.names = TRUE,
       showProgress = TRUE)
rm(temp)

saveRDS(scRNAseqDataCEA39, file = paste0(OUT_loc, "/", Today, ".AESCRNA.CEA.39pts.Seurat.after_qc.IC_academic.RDS"))

```


# AESCRNA

## Quality control

Here review the number of cells per sample, plate, and patients. And plot the
ratio's per sample and study number.

```{r QualityControl}
## check stuff
cat("\nHow many cells per type ...?")
sort(table(scRNAseqDataCEA39@meta.data$SCT_snn_res.1.25))

# cat("\n\nHow many cells per plate ...?")
# sort(table(scRNAseqDataCEA39@meta.data$ID))

# cat("\n\nHow many cells per type per plate ...?")
# table(scRNAseqDataCEA39@meta.data$SCT_snn_res.0.8, scRNAseqDataCEA39@meta.data$ID)

cat("\n\nHow many cells per patient ...?")
sort(table(scRNAseqDataCEA39@meta.data$STUDY_NUMBER))

cat("\n\nVisualizing these ratio's per study number and sample ...?")
UMAPPlot(scRNAseqDataCEA39, label = TRUE, pt.size = 1.25, label.size = 4, group.by = "ident",
         repel = TRUE)
ggsave(paste0(PLOT_loc, "/", Today, ".UMAP.png"), plot = last_plot())
ggsave(paste0(PLOT_loc, "/", Today, ".UMAP.ps"), plot = last_plot())


# barplot(prop.table(x = table(scRNAseqDataCEA39@active.ident, scRNAseqDataCEA39@meta.data$Patient)), 
#         cex.axis = 1.0, cex.names = 0.5, las = 1,
#         col = uithof_color, xlab = "study number", legend.text = FALSE, args.legend = list(x = "bottom"))
# dev.copy(pdf, paste0(QC_loc, "/", Today, ".cell_ratios_per_sample.pdf"))
# dev.off()

# barplot(prop.table(x = table(scRNAseqDataCEA39@active.ident, scRNAseqDataCEA39@meta.data$ID)), 
#         cex.axis = 1.0, cex.names = 0.5, las = 2,
#         col = uithof_color, xlab = "sample ID", legend.text = FALSE, args.legend = list(x = "bottom"))
# dev.copy(pdf, paste0(QC_loc, "/", Today, ".cell_ratios_per_sample_per_plate.pdf"))
# dev.off()

```

## Visualisations

### Feature plots of known cellular markers

Let's project known cellular markers.

```{r Visualisation: tSNE Exploration}
UMAPPlot(scRNAseqDataCEA39, label = FALSE, pt.size = 1.25, label.size = 4, group.by = "ident",
         repel = TRUE)

# endothelial cells
FeaturePlot(scRNAseqDataCEA39, features = c("CD34"), cols =  c("#ECECEC", "#DB003F"))
FeaturePlot(scRNAseqDataCEA39, features = c("EDN1"), cols =  c("#ECECEC", "#DB003F"))
FeaturePlot(scRNAseqDataCEA39, features = c("EDNRA", "EDNRB"), cols =  c("#ECECEC", "#DB003F"))
FeaturePlot(scRNAseqDataCEA39, features = c("CDH5", "PECAM1"), cols =  c("#ECECEC", "#DB003F"))
FeaturePlot(scRNAseqDataCEA39, features = c("ACKR1"), cols =  c("#ECECEC", "#DB003F"))

# SMC
FeaturePlot(scRNAseqDataCEA39, features = c("MYH11"), cols =  c("#ECECEC", "#DB003F"))
FeaturePlot(scRNAseqDataCEA39, features = c("LGALS3", "ACTA2"), cols =  c("#ECECEC", "#DB003F"))

# macrophages
FeaturePlot(scRNAseqDataCEA39, features = c("CD14", "CD68"), cols =  c("#ECECEC", "#DB003F"))
FeaturePlot(scRNAseqDataCEA39, features = c("CD36"), cols =  c("#ECECEC", "#DB003F"))

# t-cells
FeaturePlot(scRNAseqDataCEA39, features = c("CD3E"), cols =  c("#ECECEC", "#DB003F"))
FeaturePlot(scRNAseqDataCEA39, features = c("CD4"), cols =  c("#ECECEC", "#DB003F"))
# FeaturePlot(scRNAseqDataCEA39, features = c("CD8"), cols =  c("#ECECEC", "#DB003F"))

# b-cells
FeaturePlot(scRNAseqDataCEA39, features = c("CD79A"), cols =  c("#ECECEC", "#DB003F"))

# mast cells
FeaturePlot(scRNAseqDataCEA39, features = c("KIT"), cols =  c("#ECECEC", "#DB003F"))

# NK cells
FeaturePlot(scRNAseqDataCEA39, features = c("NCAM1"), cols =  c("#ECECEC", "#DB003F"))

```

## Targets of interest:

We check whether the targets genes were sequenced using our method. In case some genes are not available in our data we could filter them here.

```{r list target genes}
target_genes <- gene_list
target_genes
```


This code is just an example to filter the list from genes that are not in the data.

- _COL3A_ ==> not found
- _COL2A_ ==> not found

```{r Visualisation: preparation}

gene_list_rm <- c("COL3A", "COL2A") 

temp = target_genes[!target_genes %in% gene_list_rm]

target_genes_qc <- c(temp)

# gene_list_qc <- gene_list
# 
# for debug
# gene_list_qc_replace <- c("MRTFA")

# target_genes_qc <- target_genes
target_genes_qc

```


### Expression in cell communities

```{r Visualisation: Targets Dot Plots, message=FALSE, warning=FALSE}
library(RColorBrewer)

p1 <- DotPlot(scRNAseqDataCEA39, 
              features = target_genes_qc,
              cols = "RdBu") + 
  theme(axis.text.x = element_text(angle = 45, hjust=1, size = 8)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank()) # remove the y- and x-axis labels

p1

ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.png"), plot = last_plot())
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ps"), plot = last_plot())
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.pdf"), plot = last_plot())

rm(p1)
```

Let's group this dotplot based on some grouping.

If we want we can also filter the genes that are not found in the data.

```{r}
# Assuming 'gene_list_df' is your data.frame with the relevant columns loaded
gene_list_df_filtered <- subset(gene_list_df, Comments != "not found" | is.na(Comments))
gene_list_df_filtered
```

### Expression in cell communities grouped

```{r}
# Separate genes by significance
gene_list_df_filtered_up <- gene_list_df_filtered$GeneSymbol[gene_list_df_filtered$Group == "UP"]
gene_list_df_filtered_down <- gene_list_df_filtered$GeneSymbol[gene_list_df_filtered$Group == "DOWN"]

# Combine the genes, first DOWN then UP (or vice versa based on preference)
ordered_genes_qc <- c(gene_list_df_filtered_up, gene_list_df_filtered_down)
```


```{r}
library(RColorBrewer)
# Create DotPlot ordered
p1_group <- DotPlot(scRNAseqData, 
              features = ordered_genes_qc, 
              cols = "RdBu") + 
  theme(axis.text.x = element_text(angle = 45, hjust=1, size = 8)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank()) # remove the y- and x-axis labels

p1_group

# Save the plots
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.Grouped.png"), plot = p1_group)
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.Grouped.ps"), plot = p1_group)
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.Grouped.pdf"), plot = p1_group)

```

#### Plotting gender

We should also plot these data stratified by gender smooth muscle cells.

##### Create subsets

```{r}
unique(scRNAseqDataCEA39$Gender)
```

```{r}
# Subset Seurat object for Males
male_cells <- WhichCells(scRNAseqDataCEA39, expression = Gender == "male")
scRNAseqDataCEA39_males <- subset(scRNAseqDataCEA39, cells = male_cells)

# Subset Seurat object for Females
female_cells <- WhichCells(scRNAseqDataCEA39, expression = Gender == "female")
scRNAseqDataCEA39_females <- subset(scRNAseqDataCEA39, cells = female_cells)
```

##### Smooth muscle cells

Get an overview of the various cell communities. 

```{r}
unique(scRNAseqDataCEA39@active.ident)
```

Select the cell community of interest - in this example 'smooth muscle cells' defined by `ACTA2+ SMC`.

```{r}
# Define the cell identities you want to include in the plot
selected_idents_cells <- c("ACTA2+ SMC")
```

```{r}
library(RColorBrewer)
library(cowplot)  # For combining plots

# Create DotPlot for males
p1_males_cells <- DotPlot(scRNAseqDataCEA39_males,
                       features = ordered_genes_qc,
                       idents = selected_idents_cells, 
                       
                       cols = "RdBu") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank()) + 
  ggtitle("Males")

# Create DotPlot for females
p2_females_cells <- DotPlot(scRNAseqDataCEA39_females,
                           features = ordered_genes_qc,
                           idents = selected_idents_cells,
                           
                           cols = "RdBu") + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8)) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank()) + 
  # theme(axis.title.x = element_blank(),
  #       axis.title.y = element_blank(),
  #       axis.text.y = element_blank(),  # Remove y-axis text
  #       axis.ticks.y = element_blank()) +  # Remove y-axis ticks 
  ggtitle("Females")

# Combine the two plots into one panel using cowplot
combined_plot_cells <- plot_grid(p1_males_cells, p2_females_cells, ncol = 2)

# Display the combined plot
print(combined_plot_cells)


# Save the plots
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.GenderMales.png"), plot = p1_males_cells)
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.GenderMales.ps"), plot = p1_males_cells)
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.GenderMales.pdf"), plot = p1_males_cells)

ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.GenderFemales.png"), plot = p2_females_cells)
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.GenderFemales.ps"), plot = p2_females_cells)
ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.GenderFemales.pdf"), plot = p2_females_cells)

# ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.Gender.png"), plot = combined_plot_cells)
# ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.Gender.ps"), plot = combined_plot_cells)
# ggsave(paste0(PLOT_loc, "/", Today, ".DotPlot.Targets.ColocGrouped.TCellonly.Gender.pdf"), plot = combined_plot_cells)

```

```{r Visualisation: Targets Feature Plots, message=FALSE, warning=FALSE}
FeaturePlot(scRNAseqDataCEA39, features = c(target_genes_qc),
            cols =  c("#ECECEC", "#DB003F", "#9A3480","#1290D9"),
            combine = TRUE)

ggsave(paste0(PLOT_loc, "/", Today, ".FeaturePlot.Targets.png"), plot = last_plot())
ggsave(paste0(PLOT_loc, "/", Today, ".FeaturePlot.Targets.ps"), plot = last_plot())

```

```{r Visualisation: Targets}
# VlnPlot(scRNAseqDataCEA39, features = "DUSP27")

# VlnPlot files
ifelse(!dir.exists(file.path(PLOT_loc, "/VlnPlot")), 
       dir.create(file.path(PLOT_loc, "/VlnPlot")), 
       FALSE)
VlnPlot_loc = paste0(PLOT_loc, "/VlnPlot")


for (GENE in target_genes_qc){
  print(paste0("Projecting the expression of ", GENE, "."))

  vp1 <-  VlnPlot(scRNAseqDataCEA39, features = GENE) + 
    xlab("cell communities") + 
    ylab(bquote("normalized expression")) +
    theme(axis.title.x = element_text(color = "#000000", size = 14, face = "bold"), 
            axis.title.y = element_text(color = "#000000", size = 14, face = "bold"), 
            legend.position = "none")
    ggsave(paste0(VlnPlot_loc, "/", Today, ".VlnPlot.",GENE,".png"), plot = last_plot())
    ggsave(paste0(VlnPlot_loc, "/", Today, ".VlnPlot.",GENE,".ps"), plot = last_plot())
    ggsave(paste0(VlnPlot_loc, "/", Today, ".VlnPlot.",GENE,".pdf"), plot = last_plot())
  
  # print(vp1)
  
}

```

### Differential expression between cell communities

Here we project genes to only the broad cell communities:

-   macrophages
-   endothelial cells
-   smooth muscle cells
-   T-cells
-   B-cells
-   Mast cells
-   NK-cells
-   Mixed cells

#### Macrophages

```{r}
unique(scRNAseqDataCEA39@active.ident)
```

Comparison between the macrophages cell communities (*CD14/CD68*<sup>+</sup>),
and all other communities.

```{r Visualisation: Volcano MAC calculate}

MAC.markers <- FindMarkers(object = scRNAseqDataCEA39, 
                          ident.1 = c("CD68+CASP1+IL1B+SELL MInf", 
                                      "CD68+CD1C+ DC", 
                                      "CD68+CD4+ Mono",
                                      "CD68+IL18+TLR4+TREM2+ MRes",
                                      "CD68+ABCA1+OLR1+TREM2+ FC"), 
                          ident.2 = c(#"CD68+CASP1+IL1B+SELL MInf", 
                                      #"CD68+CD1C+ DC", 
                                      #"CD68+CD4+ Mono",
                                      #"CD68+IL18+TLR4+TREM2+ MRes",
                                      #"CD68+ABCA1+OLR1+TREM2+ FC",
                                      "CD3+ TC I",
                                      "CD3+ TC II", 
                                      "CD3+ TC III", 
                                      "CD3+ TC IV", 
                                      "CD3+ TC V", 
                                      "CD3+ TC VI", 
                                      "FOXP3+ TC", 
                                      "CD34+ EC I", 
                                      "CD34+ EC II",
                                      "ACTA2+ SMC", 
                                      "CD3+CD56+ NK I",
                                      "CD3+CD56+ NK II", 
                                      "CD68+KIT+ MC",
                                      "CD79+ BCplasma", 
                                      "CD79A+ BCmem"))

DT::datatable(MAC.markers)
```

```{r Visualisation: Volcano MAC, message=FALSE, warning=FALSE}
MAC_Volcano_TargetsA = EnhancedVolcano(MAC.markers,
    lab = rownames(MAC.markers),
    x = "avg_log2FC",
    y = "p_val_adj",
    selectLab = target_genes_qc,
    axisLabSize = 12,
    xlab = "average fold-change",
    title = "Macrophage markers\n(Macrophage communities vs the rest)",
    titleLabSize = 14,
    pCutoff = 0.05/(nrow(MAC.markers)), # 20552 genes
    FCcutoff = 1.25,
    pointSize = 1.5,
    labSize = 3.0,
    legendLabels =c('NS','avg. fold-change','P',
      'P & avg. fold-change'),
    legendPosition = "right",
    legendLabSize = 10,
    legendIconSize = 3.0,
    drawConnectors = TRUE,
    widthConnectors = 0.2,
    colConnectors = "#595A5C",
    gridlines.major = FALSE,
    gridlines.minor = FALSE)
MAC_Volcano_TargetsA
ggsave(paste0(PLOT_loc, "/", Today, ".Volcano.MAC.DEG.Targets.pdf"), 
       plot = MAC_Volcano_TargetsA)
```

The target results are given below and written to a file.

```{r Results MAC}
library(tibble)
MAC.markers <- add_column(MAC.markers, Gene = row.names(MAC.markers), .before = 1)

temp <- MAC.markers[MAC.markers$Gene %in% target_genes_qc,]

DT::datatable(temp)
```

```{r Results MAC: writing}
fwrite(temp, file = paste0(OUT_loc, "/", Today, ".MAC.DEG.Targets.txt"),
       quote = FALSE,
       sep = "\t", 
       showProgress = FALSE, verbose = FALSE)
```

#### Smooth muscle cells

Comparison between the smooth muscle cell communities (*ACTA2*<sup>+</sup>), and
all other communities.

```{r Visualisation: Volcano SMC calculate}

SMC.markers <- FindMarkers(object = scRNAseqDataCEA39, 
                          ident.1 = c("ACTA2+ SMC"), 
                          ident.2 = c("CD68+CASP1+IL1B+SELL MInf", 
                                      "CD68+CD1C+ DC", 
                                      "CD68+CD4+ Mono",
                                      "CD68+IL18+TLR4+TREM2+ MRes",
                                      "CD68+ABCA1+OLR1+TREM2+ FC",
                                      "CD3+ TC I",
                                      "CD3+ TC II", 
                                      "CD3+ TC III", 
                                      "CD3+ TC IV", 
                                      "CD3+ TC V", 
                                      "CD3+ TC VI", 
                                      "FOXP3+ TC", 
                                      "CD34+ EC I", 
                                      "CD34+ EC II",
                                      #"ACTA2+ SMC", 
                                      "CD3+CD56+ NK I",
                                      "CD3+CD56+ NK II", 
                                      "CD68+KIT+ MC",
                                      "CD79+ BCplasma", 
                                      "CD79A+ BCmem"))

DT::datatable(SMC.markers)
```

```{r Visualisation: Volcano SMC, message=FALSE, warning=FALSE}
SMC_Volcano_TargetsA = EnhancedVolcano(SMC.markers,
    lab = rownames(SMC.markers),
    x = "avg_log2FC",
    y = "p_val_adj",
    selectLab = target_genes_qc,
    axisLabSize = 12,
    xlab = "average fold-change",
    title = "SMC markers\n(SMC communities vs the rest)",
    titleLabSize = 14,
    pCutoff = 0.05/(nrow(SMC.markers)), # 20552 genes
    FCcutoff = 1.25,
    pointSize = 1.5,
    labSize = 3.0,
    legendLabels =c('NS','avg. fold-change','P',
      'P & avg. fold-change'),
    legendPosition = "right",
    legendLabSize = 10,
    legendIconSize = 3.0,
    drawConnectors = TRUE,
    widthConnectors = 0.2,
    colConnectors = "#595A5C",
    gridlines.major = FALSE,
    gridlines.minor = FALSE)
SMC_Volcano_TargetsA
ggsave(paste0(PLOT_loc, "/", Today, ".Volcano.SMC.DEG.Targets.pdf"), 
       plot = SMC_Volcano_TargetsA)
```

The target results are given below and written to a file.

```{r Results SMC}
library(tibble)
SMC.markers <- add_column(SMC.markers, Gene = row.names(SMC.markers), .before = 1)

temp <- SMC.markers[SMC.markers$Gene %in% target_genes_qc,]

DT::datatable(temp)
```

```{r Results SMC: writing}
fwrite(temp, file = paste0(OUT_loc, "/", Today, ".SMC.DEG.Targets.txt"),
       quote = FALSE,
       sep = "\t", 
       showProgress = FALSE, verbose = FALSE)
```

#### Endothelial cells

Comparison between the endothelial cell communities (*CD34*<sup>+</sup>), and
all other communities.

```{r Visualisation: Volcano EC calculate}

EC.markers <- FindMarkers(object = scRNAseqDataCEA39, 
                          ident.1 = c("CD34+ EC I", 
                                      "CD34+ EC II"), 
                          ident.2 = c("CD68+CASP1+IL1B+SELL MInf", 
                                      "CD68+CD1C+ DC", 
                                      "CD68+CD4+ Mono",
                                      "CD68+IL18+TLR4+TREM2+ MRes",
                                      "CD68+ABCA1+OLR1+TREM2+ FC",
                                      "CD3+ TC I",
                                      "CD3+ TC II", 
                                      "CD3+ TC III", 
                                      "CD3+ TC IV", 
                                      "CD3+ TC V", 
                                      "CD3+ TC VI", 
                                      "FOXP3+ TC", 
                                      # "CD34+ EC I", 
                                      # "CD34+ EC II",
                                      "ACTA2+ SMC", 
                                      "CD3+CD56+ NK I",
                                      "CD3+CD56+ NK II", 
                                      "CD68+KIT+ MC",
                                      "CD79+ BCplasma", 
                                      "CD79A+ BCmem"))

DT::datatable(EC.markers)
```

```{r Visualisation: Volcano EC, message=FALSE, warning=FALSE}
EC_Volcano_TargetsA = EnhancedVolcano(EC.markers,
    lab = rownames(EC.markers),
    x = "avg_log2FC",
    y = "p_val_adj",
    selectLab = target_genes_qc,
    axisLabSize = 12,
    xlab = "average fold-change",
    title = "Endothelial cell markers\n(EC communities vs the rest)",
    titleLabSize = 14,
    pCutoff = 0.05/(nrow(EC.markers)), # 20552 genes
    FCcutoff = 1.25,
    pointSize = 1.5,
    labSize = 3.0,
    legendLabels =c('NS','avg. fold-change','P',
      'P & avg. fold-change'),
    legendPosition = "right",
    legendLabSize = 10,
    legendIconSize = 3.0,
    drawConnectors = TRUE,
    widthConnectors = 0.2,
    colConnectors = "#595A5C",
    gridlines.major = FALSE,
    gridlines.minor = FALSE)
EC_Volcano_TargetsA
ggsave(paste0(PLOT_loc, "/", Today, ".Volcano.EC.DEG.Targets.pdf"), 
       plot = EC_Volcano_TargetsA)
```

The target results are given below and written to a file.

```{r Results EC}
library(tibble)
EC.markers <- add_column(EC.markers, Gene = row.names(EC.markers), .before = 1)

temp <- EC.markers[EC.markers$Gene %in% target_genes_qc,]

DT::datatable(temp)
```

```{r Results EC: writing}
fwrite(temp, file = paste0(OUT_loc, "/", Today, ".EC.DEG.Targets.txt"),
       quote = FALSE,
       sep = "\t", 
       showProgress = FALSE, verbose = FALSE)
```

#### T-cells

Comparison between the T-cell communities (*CD3/CD4/CD8*<sup>+</sup>), and all
other communities.

```{r Visualisation: Volcano Tcell calculate}

TC.markers <- FindMarkers(object = scRNAseqDataCEA39, 
                          ident.1 = c("CD3+ TC I",
                                      "CD3+ TC II", 
                                      "CD3+ TC III", 
                                      "CD3+ TC IV", 
                                      "CD3+ TC V", 
                                      "CD3+ TC VI", 
                                      "FOXP3+ TC"), 
                          ident.2 = c("CD68+CASP1+IL1B+SELL MInf", 
                                      "CD68+CD1C+ DC", 
                                      "CD68+CD4+ Mono",
                                      "CD68+IL18+TLR4+TREM2+ MRes",
                                      "CD68+ABCA1+OLR1+TREM2+ FC",
                                      # "CD3+ TC I",
                                      # "CD3+ TC II", 
                                      # "CD3+ TC III", 
                                      # "CD3+ TC IV", 
                                      # "CD3+ TC V", 
                                      # "CD3+ TC VI", 
                                      # "FOXP3+ TC", 
                                      "CD34+ EC I", 
                                      "CD34+ EC II",
                                      "ACTA2+ SMC", 
                                      "CD3+CD56+ NK I",
                                      "CD3+CD56+ NK II", 
                                      "CD68+KIT+ MC",
                                      "CD79+ BCplasma", 
                                      "CD79A+ BCmem"))

DT::datatable(TC.markers)
```

```{r Visualisation: Volcano Tcell, message=FALSE, warning=FALSE}
TC_Volcano_TargetsA = EnhancedVolcano(TC.markers,
    lab = rownames(TC.markers),
    x = "avg_log2FC",
    y = "p_val_adj",
    selectLab = target_genes_qc,
    axisLabSize = 12,
    xlab = "average fold-change",
    title = "T-cell markers\n(T-cell communities vs the rest)",
    titleLabSize = 14,
    pCutoff = 0.05/nrow(TC.markers), # 20552 genes
    FCcutoff = 1.25,
    pointSize = 1.5,
    labSize = 3.0,
    legendLabels =c('NS','avg. fold-change','P',
      'P & avg. fold-change'),
    legendPosition = "right",
    legendLabSize = 10,
    legendIconSize = 3.0,
    drawConnectors = TRUE,
    widthConnectors = 0.2,
    colConnectors = "#595A5C",
    gridlines.major = FALSE,
    gridlines.minor = FALSE)
TC_Volcano_TargetsA
ggsave(paste0(PLOT_loc, "/", Today, ".Volcano.TC.DEG.Targets.pdf"), 
       plot = TC_Volcano_TargetsA)
```

The target results are given below and written to a file.

```{r Results TC}
library(tibble)
TC.markers <- add_column(TC.markers, Gene = row.names(TC.markers), .before = 1)

temp <- TC.markers[TC.markers$Gene %in% target_genes_qc,]

DT::datatable(temp)
```

```{r Results TC: writing}
fwrite(temp, file = paste0(OUT_loc, "/", Today, ".TC.DEG.Targets.txt"),
       quote = FALSE,
       sep = "\t", 
       showProgress = FALSE, verbose = FALSE)
```

#### B-cells

Comparison between the B-cell communities (*CD79A*<sup>+</sup>), and all other
communities.

```{r Visualisation: Volcano Bcell calculate}

BC.markers <- FindMarkers(object = scRNAseqDataCEA39, 
                          ident.1 = c("CD79+ BCplasma", 
                                      "CD79A+ BCmem"), 
                          ident.2 = c("CD68+CASP1+IL1B+SELL MInf", 
                                      "CD68+CD1C+ DC", 
                                      "CD68+CD4+ Mono",
                                      "CD68+IL18+TLR4+TREM2+ MRes",
                                      "CD68+ABCA1+OLR1+TREM2+ FC",
                                      "CD3+ TC I",
                                      "CD3+ TC II", 
                                      "CD3+ TC III", 
                                      "CD3+ TC IV", 
                                      "CD3+ TC V", 
                                      "CD3+ TC VI", 
                                      "FOXP3+ TC", 
                                      "CD34+ EC I", 
                                      "CD34+ EC II",
                                      "ACTA2+ SMC", 
                                      "CD3+CD56+ NK I",
                                      "CD3+CD56+ NK II", 
                                      "CD68+KIT+ MC"
                                      # "CD79+ BCplasma", 
                                      # "CD79A+ BCmem"
                                      ))

DT::datatable(BC.markers)
```

```{r Visualisation: Volcano Bcell, message=FALSE, warning=FALSE}
BC_Volcano_TargetsA = EnhancedVolcano(BC.markers,
    lab = rownames(BC.markers),
    x = "avg_log2FC",
    y = "p_val_adj",
    selectLab = target_genes_qc,
    axisLabSize = 12,
    xlab = "average fold-change",
    title = "B-cell markers\n(B-cell communities vs the rest)",
    titleLabSize = 14,
    pCutoff = 0.05/nrow(BC.markers), # 20552 genes
    FCcutoff = 1.25,
    pointSize = 1.5,
    labSize = 3.0,
    legendLabels =c('NS','avg. fold-change','P',
      'P & avg. fold-change'),
    legendPosition = "right",
    legendLabSize = 10,
    legendIconSize = 3.0,
    drawConnectors = TRUE,
    widthConnectors = 0.2,
    colConnectors = "#595A5C",
    gridlines.major = FALSE,
    gridlines.minor = FALSE)
BC_Volcano_TargetsA
ggsave(paste0(PLOT_loc, "/", Today, ".Volcano.BC.DEG.Targets.pdf"), 
       plot = BC_Volcano_TargetsA)
```

The target results are given below and written to a file.

```{r Results BC}
library(tibble)
BC.markers <- add_column(BC.markers, Gene = row.names(BC.markers), .before = 1)

temp <- BC.markers[BC.markers$Gene %in% target_genes_qc,]

DT::datatable(temp)
```

```{r Results BC: writing}
fwrite(temp, file = paste0(OUT_loc, "/", Today, ".BC.DEG.Targets.txt"),
       quote = FALSE,
       sep = "\t", 
       showProgress = FALSE, verbose = FALSE)
```

#### Mast cells

Comparison between the mast cell communities (*KIT*<sup>+</sup>), and all other
communities.

```{r Visualisation: Volcano Mast calculate}

MC.markers <- FindMarkers(object = scRNAseqDataCEA39, 
                          ident.1 = c("CD68+KIT+ MC"), 
                          ident.2 = c("CD68+CASP1+IL1B+SELL MInf", 
                                      "CD68+CD1C+ DC", 
                                      "CD68+CD4+ Mono",
                                      "CD68+IL18+TLR4+TREM2+ MRes",
                                      "CD68+ABCA1+OLR1+TREM2+ FC",
                                      "CD3+ TC I",
                                      "CD3+ TC II", 
                                      "CD3+ TC III", 
                                      "CD3+ TC IV", 
                                      "CD3+ TC V", 
                                      "CD3+ TC VI", 
                                      "FOXP3+ TC", 
                                      "CD34+ EC I", 
                                      "CD34+ EC II",
                                      "ACTA2+ SMC", 
                                      "CD3+CD56+ NK I",
                                      "CD3+CD56+ NK II", 
                                      # "CD68+KIT+ MC",
                                      "CD79+ BCplasma", 
                                      "CD79A+ BCmem"))

DT::datatable(MC.markers)
```

```{r Visualisation: Volcano Mast, message=FALSE, warning=FALSE}
MC_Volcano_TargetsA = EnhancedVolcano(MC.markers,
    lab = rownames(MC.markers),
    x = "avg_log2FC",
    y = "p_val_adj",
    selectLab = target_genes_qc,
    axisLabSize = 12,
    xlab = "average fold-change",
    title = "Mast cell markers\n(Mast cell communities vs the rest)",
    titleLabSize = 14,
    pCutoff = 0.05/nrow(MC.markers), # 20552 genes
    FCcutoff = 1.25,
    pointSize = 1.5,
    labSize = 3.0,
    legendLabels =c('NS','avg. fold-change','P',
      'P & avg. fold-change'),
    legendPosition = "right",
    legendLabSize = 10,
    legendIconSize = 3.0,
    drawConnectors = TRUE,
    widthConnectors = 0.2,
    colConnectors = "#595A5C",
    gridlines.major = FALSE,
    gridlines.minor = FALSE)
MC_Volcano_TargetsA
ggsave(paste0(PLOT_loc, "/", Today, ".Volcano.MC.DEG.Targets.pdf"), 
       plot = MC_Volcano_TargetsA)
```

The target results are given below and written to a file.

```{r Results MC}
library(tibble)
MC.markers <- add_column(MC.markers, Gene = row.names(MC.markers), .before = 1)

temp <- MC.markers[MC.markers$Gene %in% target_genes_qc,]

DT::datatable(temp)
```

```{r Results MC: writing}
fwrite(temp, file = paste0(OUT_loc, "/", Today, ".MC.DEG.Targets.txt"),
       quote = FALSE,
       sep = "\t", 
       showProgress = FALSE, verbose = FALSE)
```

#### NK-cells

Comparison between the natural killer cell communities (*NCAM1*<sup>+</sup>),
and all other communities.

```{r Visualisation: Volcano NK calculate}

NK.markers <- FindMarkers(object = scRNAseqDataCEA39, 
                          ident.1 = c("CD3+CD56+ NK I",
                                      "CD3+CD56+ NK II"), 
                          ident.2 = c("CD68+CASP1+IL1B+SELL MInf", 
                                      "CD68+CD1C+ DC", 
                                      "CD68+CD4+ Mono",
                                      "CD68+IL18+TLR4+TREM2+ MRes",
                                      "CD68+ABCA1+OLR1+TREM2+ FC",
                                      "CD3+ TC I",
                                      "CD3+ TC II", 
                                      "CD3+ TC III", 
                                      "CD3+ TC IV", 
                                      "CD3+ TC V", 
                                      "CD3+ TC VI", 
                                      "FOXP3+ TC", 
                                      "CD34+ EC I", 
                                      "CD34+ EC II",
                                      "ACTA2+ SMC", 
                                      # "CD3+CD56+ NK I",
                                      # "CD3+CD56+ NK II", 
                                      "CD68+KIT+ MC",
                                      "CD79+ BCplasma", 
                                      "CD79A+ BCmem"))

DT::datatable(NK.markers)
```

```{r Visualisation: Volcano NK, message=FALSE, warning=FALSE}
NK_Volcano_TargetsA = EnhancedVolcano(NK.markers,
    lab = rownames(NK.markers),
    x = "avg_log2FC",
    y = "p_val_adj",
    selectLab = target_genes_qc,
    axisLabSize = 12,
    xlab = "average fold-change",
    title = "NK markers\n(NK-cell communities vs the rest)",
    titleLabSize = 14,
    pCutoff = 0.05/nrow(NK.markers), # 20552 genes
    FCcutoff = 1.25,
    pointSize = 1.5,
    labSize = 3.0,
    legendLabels =c('NS','avg. fold-change','P',
      'P & avg. fold-change'),
    legendPosition = "right",
    legendLabSize = 10,
    legendIconSize = 3.0,
    drawConnectors = TRUE,
    widthConnectors = 0.2,
    colConnectors = "#595A5C",
    gridlines.major = FALSE,
    gridlines.minor = FALSE)
NK_Volcano_TargetsA
ggsave(paste0(PLOT_loc, "/", Today, ".Volcano.NK.DEG.Targets.pdf"), 
       plot = NK_Volcano_TargetsA)
```

The target results are given below and written to a file.

```{r Results NK}
library(tibble)
NK.markers <- add_column(NK.markers, Gene = row.names(NK.markers), .before = 1)

temp <- NK.markers[NK.markers$Gene %in% target_genes_qc,]

DT::datatable(temp)
```

```{r Results NK: writing}
fwrite(temp, file = paste0(OUT_loc, "/", Today, ".NK.DEG.Targets.txt"),
       quote = FALSE,
       sep = "\t", 
       showProgress = FALSE, verbose = FALSE)
```



# Session information

--------------------------------------------------------------------------------

    Version:      v1.2.1
    Last update:  2024-11-26
    Written by:   Sander W. van der Laan (s.w.vanderlaan-2[at]umcutrecht.nl).
    Description:  Script to load single-cell RNA sequencing (scRNAseq) data, and perform quality control (QC), and initial mapping to cells.
    Minimum requirements: R version 3.5.2 (2018-12-20) -- 'Eggshell Igloo', macOS Mojave (10.14.2).

    **MoSCoW To-Do List**
    The things we Must, Should, Could, and Would have given the time we have.
    _M_

    _S_

    _C_

    _W_

    **Changes log**
    * v1.2.1 Some fixes to the subsetting of a cell community and the visualization by group of target genes. 
    * v1.2.0 Fixed an issue where the subset of scRNAseq was loosing cell-identities.
    * v1.1.1 Textual fixes.
    * v1.1.1 Fix writing baseline table.
    * v1.1.0 Update to study database.
    * v1.0.2 Fixes to the start of the notebook. Update to loading of the clinical data. Fix on the gene-filtering.
    * v1.0.1 Update to main AEDB (there is an error in the Age-variable in the new version). Fewer patients in scRNAseq (32 vs 39 with the newer dataset).
    * v1.0.0 Initial version.

--------------------------------------------------------------------------------

```{r eval = TRUE}
sessionInfo()
```

# Saving environment

```{r Saving}
rm(backup.scRNAseqData)
rm(scRNAseqData, scRNAseqDataCEA39)

save.image(paste0(PROJECT_loc, "/",Today,".",PROJECTNAME,".AESCRNA.results.RData"))

```

+-----------------------------------------------------------------------------------------------------------------------------------------+
| <sup>© 1979-2024 Sander W. van der Laan | s.w.vanderlaan[at]gmail[dot]com | [vanderlaanand.science](https://vanderlaanand.science).</sup> |
+-----------------------------------------------------------------------------------------------------------------------------------------+

