---
title: "Preparation of DNA methylation data -- AEMS450K1 and AEMS450K2"
subtitle: Accompanying 'AE_TEMPLATE'
author: '[Sander W. van der Laan, PhD](https://vanderlaanand.science) | s.w.vanderlaan[at]gmail[dot]com'
date: '`r Sys.Date()`'
output:
  html_notebook: 
    cache: yes
    code_folding: hide
    collapse: yes
    df_print: paged
    fig.align: center
    fig_caption: yes
    fig_height: 10
    fig_retina: 2
    fig_width: 12
    number_sections: yes
    highlight: tango
    theme: paper
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
mainfont: Helvetica
editor_options:
  chunk_output_type: inline
bibliography: references.bib
knit: worcs::cite_all
---

# General Setup
We will clean the environment, setup the locations, define colors, and create a datestamp.

_Clean the environment._
```{r echo = FALSE}
rm(list = ls())
```

_Set locations and working directories..._
```{r LocalSystem, echo = FALSE}
source("scripts/local.system.R")

```

_... a package-installation function ..._
```{r Function: installations}
source("scripts/functions.R")
```


_... and load those packages._
```{r loading_packages, message=FALSE, warning=FALSE}
source("scripts/pack04.packages.R")
```

_We will create a datestamp and define the Utrecht Science Park Colour Scheme_.
```{r Setting: Colors}
Today = format(as.Date(as.POSIXlt(Sys.time())), "%Y%m%d")
Today.Report = format(as.Date(as.POSIXlt(Sys.time())), "%A, %B %d, %Y")
source("scripts/colors.R")
```

```{r global_options, include = FALSE}
# further define some knitr-options.
knitr::opts_chunk$set(fig.width = 12, fig.height = 8, fig.path = 'Figures/', 
                      wwarning = TRUE, # show warnings during codebook generation
  message = TRUE, # show messages during codebook generation
  error = TRUE, # do not interrupt codebook generation in case of errors,
                # usually better for debugging
  echo = TRUE,  # show R code
                      eval = TRUE)
ggplot2::theme_set(ggplot2::theme_minimal())
pander::panderOptions("table.split.table", Inf)
```

# Introduction

Here we will prepare the DNA methylation data (AEMS450K1 + AEMS450K2) and combine it with the clinical data and the AERNA data.

## Athero-Express Methylation Study

We study genome-wide _DNA methylation_ (DNAm) in carotid atherosclerotic plaques using data from the [Athero-Express Biobank Study](http://www.atheroexpress.nl). Previously we published an _Epigenome-Wide Association Study_ (EWAS) [on tobacco smoking](https://www.ahajournals.org/doi/10.1161/CIRCGEN.117.002030) based on plaque-derived DNAm as measured using the [Illumina Human Methylation 450K BeadArray](https://www.ncbi.nlm.nih.gov/pubmed/22126295). 

Three datasets are available:

- Athero-Express Methylation Study 450K1 (AEMS450K1), in plaques (n=485) and bkood (n=93)
- Athero-Express Methylation Study 450K2 (AEMS450K2), in plaques only (n=190)

Below we will load the data, filter out problematic probes, and prepare the data for downstream analyses. Here we tabulate the total CpGs passing QC and the number of samples in each dataset.

**Prior to filtering**

| Dataset             	| CpGs   	| Samples 	|
|---------------------	|--------	|---------	|
| AEMS450K1, plaques: 	| 483731 	| 485     	|
| AEMS450K1, blood:   	| 483731 	| 93      	|
| AEMS450K2, plaques: 	| 484249 	| 190     	|


**After to filtering**

| Dataset             	| CpGs   	| Samples 	|
|---------------------	|--------	|---------	|
| AEMS450K1, plaques: 	| 276373 	| 485     	|
| AEMS450K1, blood:   	| 276373 	| 93      	|
| AEMS450K2, plaques: 	| 276450 	| 190     	|


We used the [DNAmArray libary](https://github.com/molepi/DNAmArray) and associated [Workflow](https://molepi.github.io/DNAmArray_workflow/index.html) for quality control and normalization of the DNAm data[^1] developed by the group of prof. dr. Bastiaan Heijmans. 

Recently, there was an update of the list of problematic probes[^2]. Three groups of probes should generally be filtered out from Infinium microarray analyses: 

1) probes with internal SNPs close to the 3΄ end of the probe (*Group 1*);
2) probes with non-unique mapping to the bisulfite-converted genome (*Group 2*); and 
3) probes with off-target hybridization due to partial overlap with non-unique elements (*Group 3*).

More information can be found here: 

- https://rpubs.com/zhouwanding/417953
- https://zwdzwd.github.io/InfiniumAnnotation

There is also another list, which is partly overlapping with Zhou's, created by Naeem et al.[^3]

[^1]: Sinke L., van Iterson M., Cats D., BIOS Consortium, Slieker R., & Heijmans B.T. (2019) "DNAmArray: Streamlined workflow for the quality control, normalization, and analysis of Illumina methylation array data" [http://doi.org/10.5281/zenodo.3355292](http://doi.org/10.5281/zenodo.3355292).

[^2]: Zhou W., Laird P.W. and Shen H., "Comprehensive characterization, annotation and innovative use of Infinium DNA Methylation BeadChip probes.", [Nucleic Acids Research, Volume 45, Issue 4, 28 February 2017, Page e22](https://doi.org/10.1093/nar/gkw967).

[^3]: Naeem H., Wong N.C., Chatterton Z. _et al._ "Reducing the risk of false discovery enabling identification of biologically significant genome-wide methylation status using the HumanMethylation450 array." [BMC Genomics 15, 51 (2014).](https://doi.org/10.1186/1471-2164-15-51).


# Load data
We will load the data:

- Athero-Express clinical data.
- the AERNA data.
- the AEMS450K1 and AEMS450K2 data.

## Clinical Data
Loading Athero-Express Biobank Study clinical and biobank data, as well as the SampleList of genetic data.

```{r LoadAEDB}
cat("Loading Athero-Express Biobank Study Database...\n")

require(haven)

AEDB.CEA <- readRDS(file = paste0(OUT_loc, "/",Today,".",TRAIT_OF_INTEREST,".AEDB.CEA.RDS"))
# AEDB.CEA <- readRDS(file = paste0(OUT_loc, "/20240602.",TRAIT_OF_INTEREST,".AEDB.CEA.RDS"))
AEDB.CEA[1:10, 1:10]
dim(AEDB.CEA)

AEDB.full <- readRDS(file = paste0(OUT_loc, "/",Today,".",TRAIT_OF_INTEREST,".AEDB.FULL.RDS"))
# AEDB.full <- readRDS(file = paste0(OUT_loc, "/20240602.",TRAIT_OF_INTEREST,".AEDB.FULL.RDS"))
AEDB.full[1:10, 1:10]
dim(AEDB.full)

cat("\n* get Athero-Express Genomics Study keys...\n")
# b37 KeyTable
AEGS123.b37.sampleList.keytable <- fread(paste0(AEGSQC_loc, "/QC/SELECTIONS/20200419.QC.AEGS123.sampleList.keytable.txt"))
# b38 KeyTable
AEGS123.sampleList.keytable <- fread(paste0(TOPMED_loc, "/OUTPUT/20240604.aegscombo_tm_r3_f10_b38_pca_key_table_qc.txt"))
# fix this selection variable - latest version - to match the rest of the script
names(AEGS123.sampleList.keytable)[names(AEGS123.sampleList.keytable) == "QC_final_2023_combined_selection_v2"] <- "QC_final_2023_combined_selection"

dim(AEGS123.sampleList.keytable)
# AEGS123.sampleList.keytable[1:10, 1:10]

```

## Bulk RNAseq Study

Here we load the latest dataset from our Athero-Express bulk RNA experiments, AERNAS1 and AERNAS2. 

These bulk RNAseq data are filtered and corrected:

- UMI corrected
- unmappable genes are excluded

```{r}
AERNAScomboSE <- readRDS(file = paste0(OUT_loc, "/", Today, ".AERNAScomboSE.CEA.1093pts.SE.after_qc.IC_academic.RDS"))
# AERNAScomboSE <- readRDS(file = paste0(OUT_loc, "/20240602.AERNAScomboSE.CEA.1093pts.SE.after_qc.IC_academic.RDS"))
AERNAScomboSEnorm <- readRDS(file = paste0(OUT_loc, "/", Today, ".AERNAScomboSEnorm.CEA.1093pts.SE.after_qc.IC_academic.RDS"))
# AERNAScomboSEnorm <- readRDS(file = paste0(OUT_loc, "/20240602.AERNAScomboSEnorm.CEA.1093pts.SE.after_qc.IC_academic.RDS"))
AERNAScomboSEvst <- readRDS(file = paste0(OUT_loc, "/", Today, ".AERNAScomboSEvst.CEA.1093pts.SE.after_qc.IC_academic.RDS"))
# AERNAScomboSEvst <- readRDS(file = paste0(OUT_loc, "/20240602.AERNAScomboSEvst.CEA.1093pts.SE.after_qc.IC_academic.RDS"))
```

```{r}
# AERNASE <- AERNAScomboSE
# AERNASE <- AERNAScomboSEnorm
AERNASE <- AERNAScomboSEvst
# AERNASE$STUDY_NUMBER

AERNASE$STUDY_NUMBER <- gsub("ae", '', AERNASE$STUDY_NUMBER)
rownames(AERNASE@colData) <- gsub("ae", '', rownames(AERNASE@colData))
# AERNASE$STUDY_NUMBER
# rownames(AERNASE@colData)
```


## AEMS450K1 and AEMS450K2

Loading the *Athero-Express Methylation Study* datasets (*AEMS450K1* and *AEMS450K2*).

```{r AEMS450K: Loading data}
cat("\n  - loading B/Mvalues of plaque samples in AEMS450K1 ...")
load(paste0(AEMS450K1_loc,"/20180625.aems450k1.BvaluesQCIMPSE.plaque.RData"))
load(paste0(AEMS450K1_loc,"/20180625.aems450k1.MvaluesQCIMPSE.plaque.RData"))

cat("\n  - loading B/Mvalues of *blood* samples in AEMS450K1 ...")
load(paste0(AEMS450K1_loc,"/20180625.aems450k1.BvaluesQCIMPSE.blood.RData"))
load(paste0(AEMS450K1_loc,"/20180625.aems450k1.MvaluesQCIMPSE.blood.RData"))

cat("\n  - loading B/Mvalues of plaque samples in AEMS450K2 ...")
load(paste0(AEMS450K2_loc,"/20180625.aems450k2.BvaluesQCIMPSE.plaque.RData"))
load(paste0(AEMS450K2_loc,"/20180625.aems450k2.MvaluesQCIMPSE.plaque.RData"))
```

### Masking files

We load the various masking-files.

#### Zhou et al.

```{r Get MaskingZhou}
# hm450.hg19.manifest.180909 <- readRDS(file = url("https://zwdzwd.s3.amazonaws.com/InfiniumAnnotation/20180909/HM450/HM450.hg19.manifest.rds"),
#         "rb") 
# hm450.hg19.manifest.180909 <- fread(input = "https://zhouserver.research.chop.edu/InfiniumAnnotation/20180909/HM450/HM450.hg19.manifest.tsv.gz")

hm450.hg19.manifest.180909 <- readRDS(file = url("https://zhouserver.research.chop.edu/InfiniumAnnotation/20180909/HM450/HM450.hg19.manifest.rds"),
        "rb")
head(hm450.hg19.manifest.180909)
```

**Column Legends**

- *seqnames, start and end* - the location of the target (1-based coordinates, 2 nucleotides for CpG probes, or 1 nucleotide for CpH and SNP probes). strand is left as "*" always. Some erroneous CpH probe coordinates mapping information in the manufacturer's manifest have been corrected. SNP probe coordinates are provided. 
- *strand* - This is consistent with "orientation" column for strand information of the actual probe. '+' is for all the up-probes positioned in smaller coordinates and '-' for all the down-probes positioned in greater coordinates with respect to the target CpGs.
- *address_A and address_B* - addresses of probe A and B on the chip designated by the original manifest. 
- *channel* - "Both" for type II probes and "Grn"/"Red" for type I probes. 
- *designType* - either "I" or "II". 
- *nextBase* - the actual extension base (on the probe strand) after bisulfite conversion ("A" or "C" or "T"). Unmapped probe has extension base labeled in the original manifest. 
- *nextBaseRef* - the extension base (on the hybridized DNA) before bisulfite conversion ("A", "C", "G" or "T"). Unmapped probe has "NA". 
- *probeType* - either "cg", "ch" or "rs". 
- *orientation* - either "up" or "down" specifying whether the probe is positioned upstream (in smaller coordinates) or downstream (in greater coordinates) the target. 
Note that by design, probes positioned upstream (in smaller coordinates) are always on the Watson strand and probes positioned downstream (in greater coordinates) are always on the Crick strand.
- *probeCpGcnt* - the number of additional CpGs in the probe (not counting the interrogated CpG). 
- *context35* - the number of CpG in the [-35bp, +35bp] window. 
- *probeStart and probeEnd* - the mapped start and end position of the probe, it is always 50bp long. 
- *ProbeSeq_A and ProbeSeq_B* - the probe sequence for allele A and B.
- *gene* - ";"-separated list of gene annotations (unique and alphabetically sorted). Gene models follows GENCODE version 22 (hg38). 
- *gene_HGNC* - ";"-separated list of gene annotations (unique and alphabetically sorted). Genes are checked using HGNChelper for compatibility with HGNC. Gene models follows GENCODE version 22 (hg38). 
- *chrm_A, beg_A, flag_A, mapQ_A, cigar_A, NM_A* - the mapping info for probe A excluding decoy chromsomes. mapQ=mapping quality score, 0-60, with 60 being the best.
- *chrm_B, beg_B, flag_B, mapQ_B, cigar_B, NM_B* - the mapping info for probe B excluding decoy chromosomes, like above.
- *wDecoy_chrm_A, wDecoy_beg_A, wDecoy_mapQ_A, wDecoy_cigar_A, wDecoy_NM_A* - the mapping info for probe A including decoy chromosomes.
- *wDecoy_chrm_B, wDecoy_beg_B, wDecoy_mapQ_B, wDecoy_cigar_B, wDecoy_NM_B* - the mapping info for probe B including decoy chromosomes.
- *posMatch* - whether the mapping matches the original manifest, it only applies to hg19 and will be NA under hg38. 

Several masking columns indicate the different levels of 'masking', *i.e.* filtering, that could be applied.

- *MASK.mapping* - whether the probe is masked for mapping reason. Probes retained should have high quality (>=40 on 0-60 scale) consistent (with designed MAPINFO) mapping (for both in the case of type I) without INDELs. 
- *MASK.typeINextBaseSwitch* - whether the probe has a SNP in the extension base that causes a color channel switch from the official annotation (described as color-channel-switching, or CCS SNP in the reference). These probes should be processed differently than designed (by summing up both color channels instead of just the annotated color channel). 
- *MASK.rmsk15* - whether the 15bp 3'-subsequence of the probe overlap with repeat masker, this MASK is NOT recommended. 
- *MASK.sub25.copy, MASK.sub30.copy, MASK.sub35.copy and MASK.sub40.copy* - whether the 25bp, 30bp, 35bp and 40bp 3'-subsequence of the probe is non-unique. 
- *MASK.snp5.common* - whether 5bp 3'-subsequence (including extension for typeII) overlap with any of the common SNPs from dbSNP (global MAF can be under 1%). 
- *MASK.snp5.GMAF1p* - whether 5bp 3'-subsequence (including extension for typeII) overlap with any of the SNPs with global MAF >1%. 
- *MASK.extBase* - probes masked for extension base inconsistent with specified color channel (type-I) or CpG (type-II) based on mapping. 
- *MASK.general* - recommended general purpose masking merged from "MASK.sub30.copy", "MASK.mapping", "MASK.extBase", "MASK.typeINextBaseSwitch" and "MASK.snp5.GMAF1p". 

#### Naeem et al.
In addition there is another paper by Naeem _et al._[^3] suggesting to be even more stringent. They propose to filter using the following criteria:

- determining probe uniqueness across genome
- does the probe map to repititive sequence elements?
- does the probe map to DNA harboring an INDEL?
- does the probe map to DNA containing a SNP?
- does the probe map to sequence with a SNP at the interrogated CpG?
- is the SNP in the probe OK in bisulfite space?

This should be done *prior* to performing methylation quantitative trait loci (mQTL) analyses, but also for differential methylation or epigenome-wide association studies (EWAS). 

```{r Get MaskingNaeem}
hm450.hg19.manifest.naeem <- read.csv(url("https://static-content.springer.com/esm/art%3A10.1186%2F1471-2164-15-51/MediaObjects/12864_2013_7006_MOESM2_ESM.csv"))
head(hm450.hg19.manifest.naeem)

hm450.hg19.manifest.naeem.list <- hm450.hg19.manifest.naeem[!is.na(hm450.hg19.manifest.naeem$Flag.discard.keep.) &
                                                     hm450.hg19.manifest.naeem$Flag.discard.keep. == "discard", ]
```


**Legend** 

1. *Probe* This column contains the probe ID as provided by Illumina.
2. *SNP+INDEL_count* This column gives the total number of known SNPs (dbSNP ver135) and INDELs overlapping the probe region. Multiple SNPs that mapped to same location counted as 1.
3. *Flag(discard/keep)* This column contains the word ‘discard’ or ‘keep’. The word “keep” indicates that the given probe is recommended for subsequent analysis, whereas the word “discard” means that the probe is recommended to be removed from the subsequent analysis.
4. *MultiMap* This column contains value of either 1 or 0. The value of “1” indicates that the corresponding probe hybridizes to multiple genomic loci whereas the value of “0” represents that the probe maps to unique genomic loci.
5. *Indels* This column provides the total number of INDELs which reside in the region hybridized by the probe.
6. *SNP-at-CpG* This column holds value of either 1 or 0. The value of “1” indicates that the given probe mapped to sequence with the SNP at the interrogated CpG whereas the value of "0" represents that the probe does not contain any SNP at the interrogated CpG site.
7. *Repeat* This column contains the value of either 1 or 0. The value of “1” illustrates that the probe spans a region in the genome containing repeat sequence elements whereas the value of “0” represents that the probe hybridizes to non-repeat regions.
8. *WGBS_HM450K_GT_0.3* This column contains the value of either 1 or 0. The value of 1 represents that, for a given probe, the absolute beta difference between whole-genome bisulfite sequencing (WGBS) and Illumina HumanMethylation450 (HM450K) bead array is greater than 0.3 while the value of 0 shows that the probe holds absolute beta difference (WGBS-HM450K) less than or equal to 0.3.
9. *ProbeType* This column contains the character(s) I or II. The character ‘I’ indicates that the probe is of Infinium I type where as a character of ‘II’ indicates that the given probe is of Infinium II type.
10. *BisOK* This column provides the total number of SNPs that were okay in bisulfite-treated genome.

#### GoNL

Through above lists, probes with (common) SNPs (MAF > 1%) are removed. However, there are also variants specific to the Dutch population discovered through [GoNL](https://rdrr.io/bioc/omicsPrint/man/hm450.manifest.pop.GoNL.html)[^4]. We load this list here.

[^4]: Van Iterson M., Cats D., Hop P., and Heijmans B.T. 2018. "OmicsPrint: Detection of Data Linkage Errors in Multiple Omics Studies." Edited by Oliver Stegle. [Bioinformatics 34 (12) (June): 2142–2143](https://academic.oup.com/bioinformatics/article/34/12/2142/4840579). DOI:10.1093/bioinformatics/bty062.


```{r Remove CpG-probes with SNPs}
library(DNAmArray)
library(omicsPrint)
data(hm450.manifest.pop.GoNL) ##From omicsPrint
#hm450.manifest.pop.GoNL
hm450.manifest.pop.GoNL <- hm450.manifest.pop.GoNL[!is.na(hm450.manifest.pop.GoNL$MASK.general.GoNL) &
                                                     hm450.manifest.pop.GoNL$MASK.general.GoNL == TRUE, ]
```

A GRanges object with 485577 ranges and 65 metadata columns:

- *MASK.general.pop* - Recommended general purpose masking merged from "MASK.sub30.copy", "MASK.mapping" (in either the hg38 or hg19 genome), "MASK.extBase", "MASK.typeINextBaseSwitch" and "MASK.snp5.pop" from the "hm450.manifest" file and the "hm450.manifest.pop" file (see source).For GoNL, "MASK.typeINextBaseSwitchandINDEL.GoNL" is used instead of "MASK.typeINextBaseSwitch"
- *MASK.snp5.pop* - Whether the 5bp 3'-subsequence (including extension for type II) overlap with a SNP with population-specific AF > 0.01
- *MASK.typeINextBaseSwitchandINDEL.GoNL* - SNPs (that cause a color-channel switch) and INDELS with AF > 0.01 in GoNL. In contrast, "MASK.typeINextBaseSwitch" column is based on all SNPs in 1000 genomes and dbSNP, regardless of population or allele frequency


### Filtering

#### Problematic probes

Remove problematic probes.

```{r}
# Create probeMaskingFix function
probeMaskingFix <- function (values, array = c("EPIC", "450K"), genome = c("hg19", 
    "hg38"), verbose = TRUE) 
{
    if (verbose == TRUE) {
        numValues <- nrow(values)
        cat(paste0("[probeFilterDNAmArray] Extracting probe filter for ",numValues," probes... \n"))
    }
    if (array == "EPIC" & genome == "hg19") {
      cat("[probeFilterDNAmArray] EPIC array on hg19... \n")
      maskProbes <- read_tsv(paste0(path.package("DNAmArray"), "/extdata/EPIC.hg19.manifest.txt.gz"))
    }
    if (array == "EPIC" & genome == "hg38") {
      cat("[probeFilterDNAmArray] EPIC array on hg38... \n")
      maskProbes <- read_tsv(paste0(path.package("DNAmArray"), "/extdata/EPIC.hg38.manifest.txt.gz"))
    }
    if (array == "450K" & genome == "hg19") {
      cat("[probeFilterDNAmArray] 450K array on hg19... \n")
      maskProbes <- read_tsv(paste0(path.package("DNAmArray"), "/extdata/HM450.hg19.manifest.txt.gz"))
    }
    if (array == "450K" & genome == "hg38") {
      cat("[probeFilterDNAmArray] 450K array on hg38... \n")
      maskProbes <- read_tsv(paste0(path.package("DNAmArray"), "/extdata/HM450.hg38.manifest.txt.gz"))
    }
    maskProbes <- maskProbes[maskProbes$MASK_general == TRUE, 
        ]$probeID
    if (verbose == TRUE) {
        numMask <- length(maskProbes)
        cat("[DNAmArray]", numMask, "probes considered for filtering... \n")
    }
    values <- values[!(rownames(values) %in% maskProbes), ]
    if (verbose == TRUE) {
        numGone <- numValues - nrow(values)
        cat("[DNAmArray]", numGone, " out of ", numValues, "probes removed from the dataset \n")
    }
    return(values)
}
```


```{r Remove ProblematicProbes B-values}
cat("Processing B-value datasets ...\n")
cat("* plaque\n")
dim(aems450k1.BvaluesQCIMPplaqueSE)
aems450k1.BvaluesQCIMPplaqueSE.new <- probeMaskingFix(aems450k1.BvaluesQCIMPplaqueSE,
                                                      array = "450K", genome = "hg19",
                                                      verbose = TRUE)
dim(aems450k1.BvaluesQCIMPplaqueSE.new)

cat("* blood\n")
dim(aems450k1.BvaluesQCIMPbloodSE)
aems450k1.BvaluesQCIMPbloodSE.new <- probeMaskingFix(aems450k1.BvaluesQCIMPbloodSE,
                                                     array = "450K", genome = "hg19",
                                                     verbose = TRUE)
dim(aems450k1.BvaluesQCIMPbloodSE.new)

cat("* plaque\n")
dim(aems450k2.BvaluesQCIMPplaqueSE)
aems450k2.BvaluesQCIMPplaqueSE.new <- probeMaskingFix(aems450k2.BvaluesQCIMPplaqueSE,
                                                      array = "450K", genome = "hg19", 
                                                      verbose = TRUE)
dim(aems450k2.BvaluesQCIMPplaqueSE.new)
```


```{r Remove ProblematicProbes M-values}
cat("Processing M-value datasets ...\n")
cat("* plaque\n")
dim(aems450k1.MvaluesQCIMPplaqueSE)
aems450k1.MvaluesQCIMPplaqueSE.new <- probeMaskingFix(aems450k1.MvaluesQCIMPplaqueSE, 
                                                      array = "450K", genome = "hg19", 
                                                      verbose = TRUE)
dim(aems450k1.MvaluesQCIMPplaqueSE.new)

cat("* blood\n")
dim(aems450k1.MvaluesQCIMPbloodSE)
aems450k1.MvaluesQCIMPbloodSE.new <- probeMaskingFix(aems450k1.MvaluesQCIMPbloodSE,
                                                     array = "450K", genome = "hg19",
                                                     verbose = TRUE)
dim(aems450k1.MvaluesQCIMPbloodSE.new)

cat("* plaque\n")
dim(aems450k2.MvaluesQCIMPplaqueSE)
aems450k2.MvaluesQCIMPplaqueSE.new <- probeMaskingFix(aems450k2.MvaluesQCIMPplaqueSE, 
                                                      array = "450K", genome = "hg19", 
                                                      verbose = TRUE)
dim(aems450k2.MvaluesQCIMPplaqueSE.new)

```

#### Additional probes

Remove additional CpGs flagged for being problematic.

```{r Remove Additional CpGs AEMS450K1 and AEMS450K2 B-values}
cat("Processing B-value datasets ...\n")
cat("* plaque\n")
aems450k1.BvaluesQCIMPplaqueSE.new2 <- aems450k1.BvaluesQCIMPplaqueSE.new[!(names(aems450k1.BvaluesQCIMPplaqueSE.new) %in% hm450.hg19.manifest.naeem.list$probe),]
dim(aems450k1.BvaluesQCIMPplaqueSE.new2)

cat("* blood\n")
aems450k1.BvaluesQCIMPbloodSE.new2 <- aems450k1.BvaluesQCIMPbloodSE.new[!(names(aems450k1.BvaluesQCIMPbloodSE.new) %in% hm450.hg19.manifest.naeem.list$probe),]
dim(aems450k1.BvaluesQCIMPbloodSE.new2)

cat("* plaque\n")
aems450k2.BvaluesQCIMPplaqueSE.new2 <- aems450k2.BvaluesQCIMPplaqueSE.new[!(names(aems450k2.BvaluesQCIMPplaqueSE.new) %in% hm450.hg19.manifest.naeem.list$probe),]
dim(aems450k2.BvaluesQCIMPplaqueSE.new2)

rm(aems450k1.BvaluesQCIMPplaqueSE.new,
   aems450k1.BvaluesQCIMPbloodSE.new,
   aems450k2.BvaluesQCIMPplaqueSE.new)
```


```{r Remove Additional CpGs AEMS450K1 and AEMS450K2 M-values}
cat("Processing M-value datasets ...\n")
cat("* plaque\n")
aems450k1.MvaluesQCIMPplaqueSE.new2 <- aems450k1.MvaluesQCIMPplaqueSE.new[!(names(aems450k1.MvaluesQCIMPplaqueSE.new) %in% hm450.hg19.manifest.naeem.list$probe),]
dim(aems450k1.MvaluesQCIMPplaqueSE.new2)

cat("* blood\n")
aems450k1.MvaluesQCIMPbloodSE.new2 <- aems450k1.MvaluesQCIMPbloodSE.new[!(names(aems450k1.MvaluesQCIMPbloodSE.new) %in% hm450.hg19.manifest.naeem.list$probe),]
dim(aems450k1.MvaluesQCIMPbloodSE.new2)

cat("* plaque\n")
aems450k2.MvaluesQCIMPplaqueSE.new2 <- aems450k2.MvaluesQCIMPplaqueSE.new[!(names(aems450k2.MvaluesQCIMPplaqueSE.new) %in% hm450.hg19.manifest.naeem.list$probe),]
dim(aems450k2.MvaluesQCIMPplaqueSE.new2)

rm(aems450k1.MvaluesQCIMPbloodSE.new,
   aems450k1.MvaluesQCIMPplaqueSE.new,
   aems450k2.MvaluesQCIMPplaqueSE.new)

```


#### GoNL

Remove CpGs with GoNL SNPs AEMS450K1 and AEMS450K2.

```{r Remove CpGs with GoNL SNPs AEMS450K1 and AEMS450K2 B-values}
cat("Processing B-value datasets ...\n")
cat("* plaque\n")
aems450k1.BvaluesQCIMPplaqueSECLEAN <- aems450k1.BvaluesQCIMPplaqueSE.new2[!(names(aems450k1.BvaluesQCIMPplaqueSE.new2) %in% names(hm450.manifest.pop.GoNL)),]
dim(aems450k1.BvaluesQCIMPplaqueSECLEAN)

cat("* blood\n")
aems450k1.BvaluesQCIMPbloodSECLEAN <- aems450k1.BvaluesQCIMPbloodSE.new2[!(names(aems450k1.BvaluesQCIMPbloodSE.new2) %in% names(hm450.manifest.pop.GoNL)),]
dim(aems450k1.BvaluesQCIMPbloodSECLEAN)

cat("* plaque\n")
aems450k2.BvaluesQCIMPplaqueSECLEAN <- aems450k2.BvaluesQCIMPplaqueSE.new2[!(names(aems450k2.BvaluesQCIMPplaqueSE.new2) %in% names(hm450.manifest.pop.GoNL)),]
dim(aems450k2.BvaluesQCIMPplaqueSECLEAN)

rm(aems450k1.BvaluesQCIMPplaqueSE.new2,
   aems450k1.BvaluesQCIMPbloodSE.new2,
   aems450k2.BvaluesQCIMPplaqueSE.new2)
```


```{r Remove CpGs with GoNL SNPs AEMS450K1 and AEMS450K2 M-values}
cat("Processing M-value datasets ...\n")
cat("* plaque\n")
aems450k1.MvaluesQCIMPplaqueSECLEAN <- aems450k1.MvaluesQCIMPplaqueSE.new2[!(names(aems450k1.MvaluesQCIMPplaqueSE.new2) %in% names(hm450.manifest.pop.GoNL)),]
dim(aems450k1.MvaluesQCIMPplaqueSECLEAN)

 cat("* blood\n")
aems450k1.MvaluesQCIMPbloodSECLEAN <- aems450k1.MvaluesQCIMPbloodSE.new2[!(names(aems450k1.MvaluesQCIMPbloodSE.new2) %in% names(hm450.manifest.pop.GoNL)),]
dim(aems450k1.MvaluesQCIMPbloodSECLEAN)

cat("* plaque\n")
aems450k2.MvaluesQCIMPplaqueSECLEAN <- aems450k2.MvaluesQCIMPplaqueSE.new2[!(names(aems450k2.MvaluesQCIMPplaqueSE.new2) %in% names(hm450.manifest.pop.GoNL)),]
dim(aems450k2.MvaluesQCIMPplaqueSECLEAN)

rm(aems450k1.MvaluesQCIMPbloodSE.new2,
   aems450k1.MvaluesQCIMPplaqueSE.new2,
   aems450k2.MvaluesQCIMPplaqueSE.new2)

```


We started with:
```{r Probes & Samples Input}
dim(aems450k1.BvaluesQCIMPplaqueSE)  # 483,731 CpGs, 485 samples
dim(aems450k1.BvaluesQCIMPbloodSE)  # 483,731 CpGs, 93 samples
dim(aems450k2.BvaluesQCIMPplaqueSE)  # 484,249 CpGs, 190 samples
# rowRanges(aems450k1.MvaluesQCplaqueClean)

dim(aems450k1.MvaluesQCIMPplaqueSE)  # 483,731 CpGs, 485 samples
dim(aems450k1.MvaluesQCIMPbloodSE)  # 483,731 CpGs, 93 samples
dim(aems450k2.MvaluesQCIMPplaqueSE)  # 484,249 CpGs, 190 samples
# rowRanges(aems450k1.MvaluesQCplaqueClean)

```

...and we end up with:
```{r Probes & Samples Output}
dim(aems450k1.BvaluesQCIMPplaqueSECLEAN)  # 276,373 CpGs, 485 samples
dim(aems450k1.BvaluesQCIMPbloodSECLEAN)  # 276,373 CpGs, 93 samples
dim(aems450k2.BvaluesQCIMPplaqueSECLEAN)  # 276,450 CpGs, 190 samples
# rowRanges(aems450k2.MvaluesQCplaqueClean)

dim(aems450k1.MvaluesQCIMPplaqueSECLEAN)  # 276,373 CpGs, 485 samples
dim(aems450k1.MvaluesQCIMPbloodSECLEAN)  # 276,373 CpGs, 93 samples
dim(aems450k2.MvaluesQCIMPplaqueSECLEAN)  # 276,450 CpGs, 190 samples
# rowRanges(aems450k2.MvaluesQCplaqueClean)
```

### Annotations

#### Load Annotations
Annotating *AEMS450K* data.

```{r LoadAnnotations}
cat("\n * Annotating data with additional CpG information.")
cat("\n   - annotating...")
# plaque
# install.packages.auto("IlluminaHumanMethylation450kmanifest")
# install.packages.auto("IlluminaHumanMethylation450kanno.ilmn12.hg19")
library(IlluminaHumanMethylation450kmanifest)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
# ?IlluminaHumanMethylation450kmanifest
# data(IlluminaHumanMethylation450kmanifest)
# IlluminaHumanMethylation450kmanifest
# ?IlluminaHumanMethylation450kanno.ilmn12.hg19
data(IlluminaHumanMethylation450kanno.ilmn12.hg19)
IlluminaHumanMethylation450kanno.ilmn12.hg19
# data(Locations)
# data(Manifest)
# data(SNPs.137CommonSingle)
data(Islands.UCSC)
# data(Other)
# Locations
# Manifest
# SNPs.137CommonSingle
# Islands.UCSC
# Other

Islands.UCSC.forannotate <- as.data.frame(Islands.UCSC)
Islands.UCSC.forannotate$CpG <- rownames(Islands.UCSC.forannotate)
Other.forannotate <- as.data.frame(Other)
```

We loaded the annotations, now we will re-annotate the data.

First the **AEMS450K1** data.

```{r Annotating AEMS450K1}
cat("\n  - annotating B/Mvalues of plaque and blood samples in AEMS450K1 ...\n")
# B-values
rowData(aems450k1.BvaluesQCIMPplaqueSECLEAN) <- merge(rowData(aems450k1.BvaluesQCIMPplaqueSECLEAN), Islands.UCSC.forannotate, by = "row.names")
rowData(aems450k1.BvaluesQCIMPplaqueSECLEAN)$Row.names <- NULL
rowData(aems450k1.BvaluesQCIMPplaqueSECLEAN) <- merge(rowData(aems450k1.BvaluesQCIMPplaqueSECLEAN), Other.forannotate, by = "row.names")
rowData(aems450k1.BvaluesQCIMPplaqueSECLEAN)$Row.names <- NULL

rowData(aems450k1.BvaluesQCIMPbloodSECLEAN) <- merge(rowData(aems450k1.BvaluesQCIMPbloodSECLEAN), Islands.UCSC.forannotate, by = "row.names")
rowData(aems450k1.BvaluesQCIMPbloodSECLEAN)$Row.names <- NULL
rowData(aems450k1.BvaluesQCIMPbloodSECLEAN) <- merge(rowData(aems450k1.BvaluesQCIMPbloodSECLEAN), Other.forannotate, by = "row.names")
rowData(aems450k1.BvaluesQCIMPbloodSECLEAN)$Row.names <- NULL

# M-values
rowData(aems450k1.MvaluesQCIMPplaqueSECLEAN) <- merge(rowData(aems450k1.MvaluesQCIMPplaqueSECLEAN), Islands.UCSC.forannotate, by = "row.names")
rowData(aems450k1.MvaluesQCIMPplaqueSECLEAN)$Row.names <- NULL
rowData(aems450k1.MvaluesQCIMPplaqueSECLEAN) <- merge(rowData(aems450k1.MvaluesQCIMPplaqueSECLEAN), Other.forannotate, by = "row.names")
rowData(aems450k1.MvaluesQCIMPplaqueSECLEAN)$Row.names <- NULL

rowData(aems450k1.MvaluesQCIMPbloodSECLEAN) <- merge(rowData(aems450k1.MvaluesQCIMPbloodSECLEAN), Islands.UCSC.forannotate, by = "row.names")
rowData(aems450k1.MvaluesQCIMPbloodSECLEAN)$Row.names <- NULL
rowData(aems450k1.MvaluesQCIMPbloodSECLEAN) <- merge(rowData(aems450k1.MvaluesQCIMPbloodSECLEAN), Other.forannotate, by = "row.names")
rowData(aems450k1.MvaluesQCIMPbloodSECLEAN)$Row.names <- NULL
```

First the **AEMS450K2** data.

```{r Annotating AEMS450K2}
cat("\n  - annotating B/Mvalues of plaque samples in AEMS450K2 ...\n")
# B-values
rowData(aems450k2.BvaluesQCIMPplaqueSECLEAN) <- merge(rowData(aems450k2.BvaluesQCIMPplaqueSECLEAN), Islands.UCSC.forannotate, by = "row.names")
rowData(aems450k2.BvaluesQCIMPplaqueSECLEAN)$Row.names <- NULL
rowData(aems450k2.BvaluesQCIMPplaqueSECLEAN) <- merge(rowData(aems450k2.BvaluesQCIMPplaqueSECLEAN), Other.forannotate, by = "row.names")
rowData(aems450k2.BvaluesQCIMPplaqueSECLEAN)$Row.names <- NULL

# M-values
rowData(aems450k2.MvaluesQCIMPplaqueSECLEAN) <- merge(rowData(aems450k2.MvaluesQCIMPplaqueSECLEAN), Islands.UCSC.forannotate, by = "row.names")
rowData(aems450k2.MvaluesQCIMPplaqueSECLEAN)$Row.names <- NULL
rowData(aems450k2.MvaluesQCIMPplaqueSECLEAN) <- merge(rowData(aems450k2.MvaluesQCIMPplaqueSECLEAN), Other.forannotate, by = "row.names")
rowData(aems450k2.MvaluesQCIMPplaqueSECLEAN)$Row.names <- NULL

rm(Islands.UCSC.forannotate, Other.forannotate, Islands.UCSC)
```


```{r SubsetAnnotations}
dim(aems450k1.MvaluesQCIMPplaqueSECLEAN)
dim(aems450k2.MvaluesQCIMPplaqueSECLEAN)

infobcp <- cpgInfo(rownames(rowData(aems450k1.MvaluesQCIMPplaqueSE)), TxDb = "TxDb.Hsapiens.UCSC.hg19.knownGene")

infobcp$CpG <- rownames(infobcp)

cat("\n * Getting a table with annotations ...")
DT::datatable(head(infobcp))

```



# MultiAssayExperiment

Combining RNAseq with AEMS450K1 in one `MultiAssayExperiment` dataset[^5]. First, we will make a list of overlapping samples.

[^5]: Reference: http://waldronlab.io/MultiAssayExperiment/articles/MultiAssayExperiment.html

## SampleMaps
```{r MultiAssayExperiment: sampleMaps}
library(MultiAssayExperiment)
cat("* Creating sample lists of the AEMS450K datasets ...\n")
temp <- as.data.frame(colData(aems450k1.MvaluesQCIMPplaqueSECLEAN))
temp.cols <- colnames(temp)[c(922:946)]
aems450k1_sampleList <- subset(temp, select = c("STUDY_NUMBER", "AEMS450K", "AEMS450K_SampleType", "SAMPLE_TYPE_AES450K",
                                         temp.cols))

temp <- as.data.frame(colData(aems450k1.MvaluesQCIMPbloodSECLEAN))
temp.cols <- colnames(temp)[c(922:946)]
aems450k1blood_sampleList <- subset(temp, select = c("STUDY_NUMBER", "AEMS450K", "AEMS450K_SampleType", "SAMPLE_TYPE_AES450K",
                                         temp.cols))

temp <- as.data.frame(colData(aems450k2.MvaluesQCIMPplaqueSECLEAN))
temp.cols <- colnames(temp)[c(922:946)]
aems450k2_sampleList <- subset(temp, select = c("STUDY_NUMBER", "AEMS450K", "AEMS450K_SampleType", "SAMPLE_TYPE_AES450K",
                                         temp.cols))
rm(temp, temp.cols)

cat("* Creating sample lists of the AERNA dataset ...\n")
temp <- as.data.frame(colData(AERNASE))
aerna_sampleList <- subset(temp, select = c("STUDY_NUMBER", "SampleType", "RNAseqType"))
rm(temp)

cat("* Creating sampleMaps of the AEMS450K and AERNA datasets ...\n")
aems450K1_map <- data.frame(primary = aems450k1_sampleList$STUDY_NUMBER,
                           colname = aems450k1_sampleList$ID,
                           stringsAsFactors = FALSE)
aems450K1blood_map <- data.frame(primary = aems450k1blood_sampleList$STUDY_NUMBER,
                           colname = aems450k1blood_sampleList$ID,
                           stringsAsFactors = FALSE)
aems450K2_map <- data.frame(primary = aems450k2_sampleList$STUDY_NUMBER,
                           colname = aems450k2_sampleList$ID,
                           stringsAsFactors = FALSE)

aerna_map <- data.frame(primary = AERNASE$STUDY_NUMBER,
                        colname = AERNASE$STUDY_NUMBER,
                        stringsAsFactors = FALSE)

cat("* Creating lists and a dataframe of sampleMaps ...\n")
listmap <- list(aems450K1_map, aems450K1blood_map, aems450K2_map, aerna_map)
names(listmap) <- c("AEMS450K1", "AEMS450K1_blood", "AEMS450K2", "AERNA")
listmap

dfmap <- MultiAssayExperiment::listToMap(listmap)
dfmap

```

## Assays

Collecting the various assay data in an `object list`. We will create two new objects for the methylation data, to drop the `QC` from the names.

```{r MultiAssayExperiment: data}
cat("Processing plaque data from AEMS450K1 ...\n")
AEMS450K1.SE.assay <- assay(aems450k1.MvaluesQCIMPplaqueSECLEAN)
AEMS450K1.SE.rowData <- rowData(aems450k1.MvaluesQCIMPplaqueSECLEAN)
AEMS450K1.SE.metadata <- metadata(aems450k1.MvaluesQCIMPplaqueSECLEAN)

AEMS450K1.SE <- SummarizedExperiment(assays = AEMS450K1.SE.assay, 
                                     rowData = AEMS450K1.SE.rowData,
                                     colData = aems450k1_sampleList,
                                     metadata = AEMS450K1.SE.metadata)

cat("\nProcessing blood data from AEMS450K1 ...\n")
AEMS450K1.blood.SE.assay <- assay(aems450k1.MvaluesQCIMPbloodSECLEAN)
AEMS450K1.blood.SE.rowData <- rowData(aems450k1.MvaluesQCIMPbloodSECLEAN)
AEMS450K1.blood.SE.metadata <- metadata(aems450k1.MvaluesQCIMPbloodSECLEAN)

AEMS450K1.blood.SE <- SummarizedExperiment(assays = AEMS450K1.blood.SE.assay, 
                                     rowData = AEMS450K1.blood.SE.rowData,
                                     colData = aems450k1blood_sampleList,
                                     metadata = AEMS450K1.blood.SE.metadata)

cat("\nProcessing plaque data from AEMS450K ...\n")
AEMS450K2.SE.assay <- assay(aems450k2.MvaluesQCIMPplaqueSECLEAN)
AEMS450K2.SE.rowData <- rowData(aems450k2.MvaluesQCIMPplaqueSECLEAN)
AEMS450K2.SE.metadata <- metadata(aems450k2.MvaluesQCIMPplaqueSECLEAN)

AEMS450K2.SE <- SummarizedExperiment(assays = AEMS450K2.SE.assay, 
                                     rowData = AEMS450K2.SE.rowData,
                                     colData = aems450k2_sampleList,
                                     metadata = AEMS450K2.SE.metadata)

cat("\nList objects ...\n")
objlist <- list("AEMS450K1" = AEMS450K1.SE, 
                "AEMS450K1_blood" = AEMS450K1.blood.SE, 
                "AEMS450K2" = AEMS450K2.SE,
                "AERNA" = AERNASE)

```

## Clinical data

Editing the clinical data to match the `MultiAssayExperiment` format.

```{r Create New AEDB}
library(labelled)
variables_of_interest = c("Hospital", "ORyear", "informedconsent",
                          "Artery_summary",
                          "Age", "Gender", 
                          "TC_finalCU", "LDL_finalCU", "HDL_finalCU", "TG_finalCU",
                          "TC_final", "LDL_final", "HDL_final", "TG_final", 
                          "hsCRP_plasma",
                          "systolic", "diastoli", "GFR_MDRD", "BMI", 
                          "KDOQI", "BMI_WHO",
                          "SmokerStatus", "AlcoholUse",
                          "DiabetesStatus", 
                          "Hypertension.selfreport", "Hypertension.selfreportdrug", "Hypertension.composite", "Hypertension.drugs", 
                          "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD", 
                          "Stroke_Dx", "sympt", "Symptoms.5G", "AsymptSympt", "AsymptSympt2G",
                          "Symptoms.Update2G", "Symptoms.Update3G",
                          "restenos", "stenose", 
                          "CAD_history", "PAOD", "Peripheral.interv", 
                          "EP_composite", "EP_composite_time", "EP_major", "EP_major_time",
                          "MAC_rankNorm", "SMC_rankNorm", "Macrophages.bin", "SMC.bin",
                          "Neutrophils_rankNorm", "MastCells_rankNorm",
                          "IPH.bin", "VesselDensity_rankNorm",
                          "Calc.bin", "Collagen.bin", 
                          "Fat.bin_10", "Fat.bin_40", 
                          "OverallPlaquePhenotype", "Plaque_Vulnerability_Index",
                          "PCSK9_plasma", "PCSK9_plasma_rankNorm")

AEDB_all <- as.data.frame(subset(AEDB.CEA, select = c("STUDY_NUMBER", "UPID", "informedconsent", 
                                                      variables_of_interest)))

rownames(AEDB_all) <- AEDB_all$STUDY_NUMBER

```

## Creation

Finally ready to make our `MultiAssayExperiment` dataset.
```{r MultiAssayExperiment: creation}
cat("*Prepping and checking out the potential for a MultiAssayExperiment dataset ...\n")
try(prepMultiAssay(ExperimentList = objlist, colData = AEDB_all, sampleMap = dfmap)$experiments,
    outFile = stdout())

cat("\n* Creating the MultiAssayExperiment ...\n")
AE_MultiAssay <- MultiAssayExperiment(experiments = objlist,
                                      colData = AEDB_all,
                                      sampleMap = dfmap)
AE_MultiAssay

# rownames(AE_MultiAssay)
# 
# colnames(AE_MultiAssay)
# 
# assays(AE_MultiAssay)
# 
# str(AE_MultiAssay)

rm(objlist)

```

## Overlaps

Now that we have a `MultiAssayExperiment`, we can inspect the overlap between datasets. 
```{r MultiAssayExperiment: overlap, message=FALSE, warning=FALSE}
cat("*Visualizing the overlap ...\n")
# Reference: https://cran.r-project.org/web/packages/UpSetR/vignettes/attribute.plots.html
library(UpSetR)
###	1	  yellow			    #FBB820 (251,184,32)				      =>	1		or 1.0>INFO
###	2	  gold			      #F59D10 (245,157,16)				      =>	2		
###	3	  salmon			    #E55738 (229,87,56)				      =>	3		or 0.05<MAF<0.2 or 0.4<INFO<0.6
###	4	  darkpink		    #DB003F ((219,0,63)				      =>	4		
###	5	  lightpink		    #E35493 (227,84,147)				      =>	5		or 0.8<INFO<1.0
###	6	  pink			      #D5267B (213,38,123)				      =>	6		
###	7	  hardpink		    #CC0071 (204,0,113)				      =>	7		
###	8	  lightpurple	    #A8448A (168,68,138)				      =>	8		
###	9	  purple			    #9A3480 (154,52,128)				      =>	9		
###	10	lavendel		    #8D5B9A (141,91,154)				      =>	10		
###	11	bluepurple		  #705296 (112,82,150)				      =>	11		
###	12	purpleblue		  #686AA9 (104,106,169)			      =>	12		
###	13	lightpurpleblue	#6173AD (97,115,173/101,120,180)	=>	13		
###	14	seablue			    #4C81BF (76,129,191)				      =>	14		
###	15	skyblue			    #2F8BC9 (47,139,201)				      =>	15		
###	16	azurblue		    #1290D9 (18,144,217)				      =>	16		or 0.01<MAF<0.05 or 0.2<INFO<0.4
###	17	lightazurblue	  #1396D8 (19,150,216)				      =>	17		
###	18	greenblue		    #15A6C1 (21,166,193)				      =>	18		
###	19	seaweedgreen	  #5EB17F (94,177,127)				      =>	19		
###	20	yellowgreen		  #86B833 (134,184,51)				      =>	20		
###	21	lightmossgreen	#C5D220 (197,210,32)				      =>	21		
###	22	mossgreen		    #9FC228 (159,194,40)				      =>	22		or MAF>0.20 or 0.6<INFO<0.8
###	23	lightgreen	  	#78B113 (120,177,19)				      =>	23/X
###	24	green			      #49A01D (73,160,29)				      =>	24/Y
###	25	grey			      #595A5C (89,90,92)				        =>	25/XY	or MAF<0.01 or 0.0<INFO<0.2
###	26	lightgrey		    #A2A3A4	(162,163,164)			      =>	26/MT

upsetSamples(AE_MultiAssay, 
             group.by = "sets",
             matrix.color = "#595A5C",
             # main.bar.color = c("#1290D9", "#E55738", "#49A01D", "#D5267B"),
             sets.bar.color = c("#1290D9", "#49A01D", "#D5267B", "#E55738"),
             sets.x.label = "Athero-Express Omics Studies",
             number.angles = 45,
             )
             
             # main.bar.color = "#595A5C",
             # sets.bar.color = "#595A5C",
             # queries = list(list(query = intersects, params = list("AEMS450K1"), color = "#FBB820", active = FALSE),
             #                list(query = intersects, params = list("AEMS450K1_blood"), color = "#E55738", active = FALSE),
             #                list(query = intersects, params = list("AEMS450K2"), color = "#1290D9", active = FALSE),
             #                list(query = intersects, params = list("AERNA"), color = "#9FC228", active = FALSE),
             #                list(query = intersects, params = list("AEMS450K1", "AERNA"), color = "#FBB820", active = TRUE),
             #                list(query = intersects, params = list("AEMS450K1_blood", "AERNA"), color = "#E55738", active = TRUE),
             #                list(query = intersects, params = list("AEMS450K2", "AERNA"), color = "#1290D9", active = TRUE),
             #                list(query = intersects, params = list("AEMS450K1", "AEMS450K2", "AERNA"),  color = "#595A5C", active = TRUE),
             #                list(query = intersects, params = list("AEMS450K1", "AEMS450K2"), color = "#595A5C", active = TRUE)))

```


## Baseline Table

We first fix some issues regarding variables, making sure that factor-variables are factors and numeric variables are numeric. And also dropping unused levels.
```{r Baseline AE_MultiAssay: preparation}
cat("====================================================================================================\n")
cat("SELECTION THE SHIZZLE\n")

cat("- sanity checking PRIOR to selection\n")
library(data.table)
# to check some numbers
ae.gender <- ifelse(AE_MultiAssay$Gender == 0, "Female", "Male")
ae.hospital <- ifelse(AE_MultiAssay$Hospital == 1, "Antonius", "UMCU")
table(ae.gender, ae.hospital, dnn = c("Sex", "Hospital"))
ae.gender <- ifelse(AE_MultiAssay$Gender == 0, "Female", "Male")
table(ae.gender, AE_MultiAssay$Artery_summary, dnn = c("Sex", "Artery"))
table(ae.gender, AE_MultiAssay$informedconsent, dnn = c("Sex", "IC"))
rm(ae.gender, ae.hospital)

# I change numeric and factors manually because, well, I wouldn't know how to fix it otherwise
# to have this 'tibble' work with 'tableone'... :-)
AE_MultiAssay$ORyear <- as.numeric(AE_MultiAssay$ORyear)
AE_MultiAssay$Age <- as.numeric(AE_MultiAssay$Age)
AE_MultiAssay$TC_finalCU <- as.numeric(AE_MultiAssay$TC_finalCU)
AE_MultiAssay$LDL_finalCU <- as.numeric(AE_MultiAssay$LDL_finalCU)
AE_MultiAssay$HDL_finalCU <- as.numeric(AE_MultiAssay$HDL_finalCU)
AE_MultiAssay$TG_finalCU <- as.numeric(AE_MultiAssay$TG_finalCU)
AE_MultiAssay$TC_final <- as.numeric(AE_MultiAssay$TC_final)
AE_MultiAssay$LDL_final <- as.numeric(AE_MultiAssay$LDL_final)
AE_MultiAssay$HDL_final <- as.numeric(AE_MultiAssay$HDL_final)
AE_MultiAssay$TG_final <- as.numeric(AE_MultiAssay$TG_final)
AE_MultiAssay$hsCRP_plasma <- as.numeric(AE_MultiAssay$hsCRP_plasma)
AE_MultiAssay$diastoli <- as.numeric(AE_MultiAssay$diastoli)
AE_MultiAssay$systolic <- as.numeric(AE_MultiAssay$systolic)
AE_MultiAssay$GFR_MDRD <- as.numeric(AE_MultiAssay$GFR_MDRD)
AE_MultiAssay$BMI <- as.numeric(AE_MultiAssay$BMI)

AE_MultiAssay$EP_composite_time <- as.numeric(AE_MultiAssay$EP_composite_time)
AE_MultiAssay$EP_major_time <- as.numeric(AE_MultiAssay$EP_major_time)

AE_MultiAssay$MAC_rankNorm <- as.numeric(AE_MultiAssay$MAC_rankNorm)
AE_MultiAssay$SMC_rankNorm <- as.numeric(AE_MultiAssay$SMC_rankNorm)
AE_MultiAssay$Neutrophils_rankNorm <- as.numeric(AE_MultiAssay$Neutrophils_rankNorm)
AE_MultiAssay$MastCells_rankNorm <- as.numeric(AE_MultiAssay$MastCells_rankNorm)
AE_MultiAssay$VesselDensity_rankNorm <- as.numeric(AE_MultiAssay$VesselDensity_rankNorm)
AE_MultiAssay$PCSK9_plasma <- as.numeric(AE_MultiAssay$PCSK9_plasma)
AE_MultiAssay$PCSK9_plasma_rankNorm <- as.numeric(AE_MultiAssay$PCSK9_plasma_rankNorm)

require(labelled)
AE_MultiAssay$informedconsent <- to_factor(AE_MultiAssay$informedconsent)
AE_MultiAssay$Hospital <- to_factor(AE_MultiAssay$Hospital)
AE_MultiAssay$Gender <- to_factor(AE_MultiAssay$Gender)
AE_MultiAssay$Artery_summary <- to_factor(AE_MultiAssay$Artery_summary)
AE_MultiAssay$KDOQI <- to_factor(AE_MultiAssay$KDOQI)
AE_MultiAssay$BMI_WHO <- to_factor(AE_MultiAssay$BMI_WHO)

AE_MultiAssay$SmokerStatus <- to_factor(AE_MultiAssay$SmokerStatus)
AE_MultiAssay$AlcoholUse <- to_factor(AE_MultiAssay$AlcoholUse)
AE_MultiAssay$DiabetesStatus <- to_factor(AE_MultiAssay$DiabetesStatus)

AE_MultiAssay$Hypertension.selfreport <- to_factor(AE_MultiAssay$Hypertension.selfreport)
AE_MultiAssay$Hypertension.selfreportdrug <- to_factor(AE_MultiAssay$Hypertension.selfreportdrug)
AE_MultiAssay$Hypertension.composite <- to_factor(AE_MultiAssay$Hypertension.composite)
AE_MultiAssay$Hypertension.drugs <- to_factor(AE_MultiAssay$Hypertension.drugs)

AE_MultiAssay$Med.anticoagulants <- to_factor(AE_MultiAssay$Med.anticoagulants)
AE_MultiAssay$Med.all.antiplatelet <- to_factor(AE_MultiAssay$Med.all.antiplatelet)
AE_MultiAssay$Med.Statin.LLD <- to_factor(AE_MultiAssay$Med.Statin.LLD)

AE_MultiAssay$Stroke_Dx <- to_factor(AE_MultiAssay$Stroke_Dx)
AE_MultiAssay$sympt <- to_factor(AE_MultiAssay$sympt)
AE_MultiAssay$Symptoms.5G <- to_factor(AE_MultiAssay$Symptoms.5G)
AE_MultiAssay$AsymptSympt <- to_factor(AE_MultiAssay$AsymptSympt)
AE_MultiAssay$AsymptSympt2G <- to_factor(AE_MultiAssay$AsymptSympt2G)
AE_MultiAssay$Symptoms.Update2G <- to_factor(AE_MultiAssay$Symptoms.Update2G)
AE_MultiAssay$Symptoms.Update3G <- to_factor(AE_MultiAssay$Symptoms.Update3G)

AE_MultiAssay$restenos <- to_factor(AE_MultiAssay$restenos)
AE_MultiAssay$stenose <- to_factor(AE_MultiAssay$stenose)

AE_MultiAssay$CAD_history <- to_factor(AE_MultiAssay$CAD_history)
AE_MultiAssay$PAOD <- to_factor(AE_MultiAssay$PAOD)
AE_MultiAssay$Peripheral.interv <- to_factor(AE_MultiAssay$Peripheral.interv)

AE_MultiAssay$EP_composite <- to_factor(AE_MultiAssay$EP_composite)
AE_MultiAssay$EP_major <- to_factor(AE_MultiAssay$EP_major)

AE_MultiAssay$Macrophages.bin <- to_factor(AE_MultiAssay$Macrophages.bin)
AE_MultiAssay$SMC.bin <- to_factor(AE_MultiAssay$SMC.bin)
AE_MultiAssay$IPH.bin <- to_factor(AE_MultiAssay$IPH.bin)
AE_MultiAssay$Calc.bin <- to_factor(AE_MultiAssay$Calc.bin)
AE_MultiAssay$Collagen.bin <- to_factor(AE_MultiAssay$Collagen.bin)
AE_MultiAssay$Fat.bin_10 <- to_factor(AE_MultiAssay$Fat.bin_10)
AE_MultiAssay$Fat.bin_40 <- to_factor(AE_MultiAssay$Fat.bin_40)
AE_MultiAssay$OverallPlaquePhenotype <- to_factor(AE_MultiAssay$OverallPlaquePhenotype)
AE_MultiAssay$Plaque_Vulnerability_Index <- to_factor(AE_MultiAssay$Plaque_Vulnerability_Index)

# We also need to fix some variables

attach(as.data.frame(colData(AE_MultiAssay)))
colData(AE_MultiAssay)$KDOQI[KDOQI == "No data available/missing"] <- NA
colData(AE_MultiAssay)$BMI_WHO[BMI_WHO == "No data available/missing"] <- NA
colData(AE_MultiAssay)$SmokerStatus[SmokerStatus == "no data available/missing"] <- NA
colData(AE_MultiAssay)$AlcoholUse[AlcoholUse == "no data available/missing"] <- NA
colData(AE_MultiAssay)$DiabetesStatus[DiabetesStatus == "no data available/missing"] <- NA
colData(AE_MultiAssay)$Hypertension.selfreport[Hypertension.selfreport == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Hypertension.selfreportdrug[Hypertension.selfreportdrug == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Hypertension.composite[Hypertension.composite == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Hypertension.drugs[Hypertension.drugs == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Med.anticoagulants[Med.anticoagulants == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Med.all.antiplatelet[Med.all.antiplatelet == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Med.Statin.LLD[Med.Statin.LLD == "No data available/missing"] <- NA

colData(AE_MultiAssay)$Stroke_Dx[Stroke_Dx == "No data available/missing"] <- NA
colData(AE_MultiAssay)$sympt[sympt == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Symptoms.5G[Symptoms.5G == "No data available/missing"] <- NA
colData(AE_MultiAssay)$AsymptSympt[AsymptSympt == "No data available/missing"] <- NA
colData(AE_MultiAssay)$AsymptSympt2G[AsymptSympt2G == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Symptoms.Update2G[Symptoms.Update2G == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Symptoms.Update3G[Symptoms.Update3G == "No data available/missing"] <- NA

colData(AE_MultiAssay)$restenos[restenos == "No data available/missing"] <- NA
colData(AE_MultiAssay)$stenose[stenose == "No data available/missing"] <- NA

colData(AE_MultiAssay)$CAD_history[CAD_history == "No data available/missing"] <- NA
colData(AE_MultiAssay)$PAOD[PAOD == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Peripheral.interv[Peripheral.interv == "No data available/missing"] <- NA

colData(AE_MultiAssay)$Macrophages.bin[Macrophages.bin == "No data available/missing"] <- NA
colData(AE_MultiAssay)$SMC.bin[SMC.bin == "No data available/missing"] <- NA
colData(AE_MultiAssay)$IPH.bin[IPH.bin == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Calc.bin[Calc.bin == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Collagen.bin[Collagen.bin == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Fat.bin_10[Fat.bin_10 == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Fat.bin_40[Fat.bin_40 == "No data available/missing"] <- NA
colData(AE_MultiAssay)$OverallPlaquePhenotype[OverallPlaquePhenotype == "No data available/missing"] <- NA
colData(AE_MultiAssay)$Plaque_Vulnerability_Index[Plaque_Vulnerability_Index == "No data available/missing"] <- NA

colData(AE_MultiAssay)$EP_composite[EP_composite == "No data available."] <- NA
colData(AE_MultiAssay)$EP_major[EP_major == "No data available."] <- NA

gdata::drop.levels(colData(AE_MultiAssay))

detach(as.data.frame(colData(AE_MultiAssay)))

```

We select only CEA patients which have a valid informed consent.

```{r Baseline AE_MultiAssay: selection}
AE_MultiAssay.CEA <- subset(AE_MultiAssay,
                    (AE_MultiAssay$Artery_summary == "carotid (left & right)" | AE_MultiAssay$Artery_summary == "other carotid arteries (common, external)") & # we only want carotids
                     AE_MultiAssay$informedconsent != "missing" & # we are really strict in selecting based on 'informed consent'!
                     AE_MultiAssay$informedconsent != "no, died" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, no commerical business" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, no questionnaires, no medical info, no commercial business" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commerical business" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info, no commercial business" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, no health treatment" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, no questionnaires" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, health treatment when possible" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, no questionnaires, no health treatment, no medical info" & 
                     AE_MultiAssay$informedconsent != "yes, no tissue, no questionnaires, no health treatment, no commercial business" & 
                     AE_MultiAssay$informedconsent != "no, doesn't want to" & 
                     AE_MultiAssay$informedconsent != "no, unable to sign" & 
                     AE_MultiAssay$informedconsent != "no, no reaction" & 
                     AE_MultiAssay$informedconsent != "no, lost" & 
                     AE_MultiAssay$informedconsent != "no, too old" & 
                     AE_MultiAssay$informedconsent != "no (never asked for IC because there was no tissue)" & 
                     AE_MultiAssay$informedconsent != "no, endpoint" & 
                     AE_MultiAssay$informedconsent != "wil niets invullen, wel alles gebruiken" & 
                     AE_MultiAssay$informedconsent != "nooit geincludeerd" & 
                     AE_MultiAssay$informedconsent != "yes, no DNA")
# AE_MultiAssay.CEA[1:10, 1:10]
cat("Before filtering based on informed consent and artery type.\n")
AE_MultiAssay

cat("\n After filtering.\n")
AE_MultiAssay.CEA

```

Now we're ready to create the baseline table, we categorize the `variables_of_interest`.

```{r}
variables_of_interest
```

```{r Baseline AE_MultiAssay: creation}
cat("===========================================================================================\n")
cat("CREATE BASELINE TABLE\n")

# Baseline table variables
basetable_vars = c(variables_of_interest)

basetable_bin = c("Hospital", "Gender", "informedconsent", "Artery_summary",
                  "KDOQI", "BMI_WHO",
                  "SmokerStatus", "AlcoholUse", 
                  "DiabetesStatus", 
                  "Hypertension.selfreport", "Hypertension.selfreportdrug",
                  "Hypertension.composite", "Hypertension.drugs",
                  "Med.anticoagulants", "Med.all.antiplatelet", "Med.Statin.LLD", 
                  "Stroke_Dx", "sympt", "Symptoms.5G", "AsymptSympt", "AsymptSympt2G",
                  "Symptoms.Update2G", "Symptoms.Update3G", 
                  "CAD_history", "PAOD", "Peripheral.interv",
                  "restenos", "stenose", 
                  "EP_composite", "EP_major",
                  "Macrophages.bin", "SMC.bin", "Calc.bin", "Collagen.bin", "IPH.bin", "Fat.bin_10", "Fat.bin_40",
                  "OverallPlaquePhenotype", "Plaque_Vulnerability_Index")
# basetable_bin

basetable_con = basetable_vars[!basetable_vars %in% basetable_bin]
# basetable_con
```

## Athero-Express Omics Studies: baseline characteristics

Showing the baseline table of the whole Athero-Express Biobank.

```{r Baseline AE_MultiAssay: Visualize}
# Create baseline tables
# http://rstudio-pubs-static.s3.amazonaws.com/13321_da314633db924dc78986a850813a50d5.html
AE_MultiAssay.CEA.tableOne = print(CreateTableOne(vars = basetable_vars, 
                                                  # factorVars = basetable_bin,
                                                  strata = "Gender",
                                                  data = as.data.frame(colData(AE_MultiAssay.CEA)), includeNA = TRUE), 
                                   nonnormal = c(), missing = TRUE,
                                   quote = FALSE, noSpaces = FALSE, showAllLevels = TRUE, explain = TRUE, 
                                   format = "pf", 
                                   contDigits = 3)[,1:6]

```


We can also save this for future reference.
```{r AE_MultiAssay: Baseline SampleSelection: write}
# Write basetable
require(openxlsx)

openxlsx::write.xlsx(file = paste0(BASELINE_loc, "/",Today,".",PROJECTNAME,".AE_MultiAssay.BaselineTable.xlsx"), 
                     as.data.frame(AE_MultiAssay.CEA.tableOne), 
                     asTable = TRUE,
                     rowNames = TRUE, 
                     colNames = TRUE, 
                     creator = "Sander W. van der Laan",
                     sheetName = "AE_MultiAssay_Baseline", overwrite = TRUE)

```

# Saving datasets

We will save the selection of sample labels, rowRanges - genomic position information - and assays of the `AE_MultiAssay.CEA` object.

First, we create a copy of the `AE_MultiAssay.CEA` object.

```{r AE_MultiAssay: preparing AE_MultiAssay as RDS}
# Save a single object to a file
temp = AE_MultiAssay.CEA
# 
colData(temp)$informedconsent <- NULL
# colData(temp)$TC_final <- NULL
# colData(temp)$LDL_final <- NULL
# colData(temp)$HDL_final <- NULL
# colData(temp)$TG_final <- NULL
# colData(temp)$systolic <- NULL
# colData(temp)$diastoli <- NULL
# colData(temp)$GFR_MDRD <- NULL
# colData(temp)$BMI <- NULL
# colData(temp)$KDOQI <- NULL
# colData(temp)$BMI_WHO <- NULL
# 
# colData(temp)$SmokerStatus <- NULL
# colData(temp)$DiabetesStatus <- NULL
# 
# colData(temp)$Hypertension.selfreport <- NULL
# colData(temp)$Hypertension.selfreportdrug <- NULL
# colData(temp)$Hypertension.composite <- NULL
# colData(temp)$Hypertension.drugs <- NULL
# 
# colData(temp)$Med.anticoagulants <- NULL
# colData(temp)$Med.all.antiplatelet <- NULL
# colData(temp)$Med.Statin.LLD <- NULL
# 
# colData(temp)$AsymptSympt <- NULL
# colData(temp)$restenos <- NULL
# 
AE_MAE.CEA = temp
# str(AE_MAE.CEA)
# AEMS450K1.assay <- MultiAssayExperiment::longFormat(AE_MAE.CEA[, , "AEMS450K1"])
rm(temp)
```

<!-- ## Get Assay -->

<!-- We'll get the assays.  -->
<!-- ```{r AE_MultiAssay: extract assays} -->
<!-- AEMS450K1.assay <- AE_MAE.CEA[, , "AEMS450K1"] -->
<!-- AEMS450K1.assay -->

<!-- AEMS450K1.blood.assay <- AE_MAE.CEA[, , "AEMS450K1_blood"] -->
<!-- AEMS450K1.blood.assay -->

<!-- AEMS450K2.assay <- AE_MAE.CEA[, , "AEMS450K2"] -->
<!-- AEMS450K2.assay -->

<!-- AERNA.assay <- AE_MAE.CEA[, , "AERNA"] -->
<!-- AERNA.assay -->

<!-- ``` -->

<!-- ## Save assays -->

<!-- We'll save the assays. -->

<!-- ```{r AE_MultiAssay: saving AE_MultiAssay as CSV} -->
<!-- # RNA -->
<!-- fwrite(as.data.frame(assay(AERNA.assay)), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.RNA.assay.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = TRUE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K1 -->
<!-- fwrite(as.data.frame(assay(AEMS450K1.assay)), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K1.assay.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = TRUE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K1 blood -->
<!-- fwrite(as.data.frame(assay(AEMS450K1.blood.assay)), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K1.blood.assay.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = TRUE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K2 -->
<!-- fwrite(as.data.frame(assay(AEMS450K2.assay)), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K2.assay.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = TRUE, -->
<!--         showProgress = F, verbose = F) -->

<!-- ``` -->

<!-- ## Saving labels -->

<!-- We'll save the labels too.  -->

<!-- ```{r AE_MultiAssay: saving labels} -->

<!-- # RNA -->
<!-- foo <- as.data.frame(colData(AERNA.assay)) -->
<!-- temp <- merge(foo, aerna_sampleList, by.x = "STUDY_NUMBER", by.y = "STUDY_NUMBER", -->
<!--               sort = FALSE, all.x = TRUE) -->
<!-- # DT::datatable(temp) -->

<!-- fwrite(temp, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.RNA.labels.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = FALSE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K1 -->
<!-- foo <- as.data.frame(colData(AEMS450K1.assay)) -->
<!-- temp <- merge(foo, aems450k1_sampleList, by.x = "STUDY_NUMBER", by.y = "STUDY_NUMBER", -->
<!--               sort = FALSE, all.x = TRUE) -->
<!-- # DT::datatable(temp) -->

<!-- fwrite(as.data.frame(colData(AEMS450K1.assay)), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K1.labels.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = FALSE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K1 blood -->
<!-- foo <- as.data.frame(colData(AEMS450K1.blood.assay)) -->
<!-- temp <- merge(foo, aems450k1blood_sampleList, by.x = "STUDY_NUMBER", by.y = "STUDY_NUMBER", -->
<!--               sort = FALSE, all.x = TRUE) -->
<!-- # DT::datatable(temp) -->

<!-- fwrite(as.data.frame(colData(AEMS450K1.blood.assay)), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K1.blood.labels.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = FALSE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K2 -->
<!-- foo <- as.data.frame(colData(AEMS450K2.assay)) -->
<!-- temp <- merge(foo, aems450k2_sampleList, by.x = "STUDY_NUMBER", by.y = "STUDY_NUMBER", -->
<!--               sort = FALSE, all.x = TRUE) -->
<!-- # DT::datatable(temp) -->

<!-- fwrite(as.data.frame(colData(AEMS450K2.assay)), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K2.labels.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = FALSE, -->
<!--         showProgress = F, verbose = F) -->
<!-- rm(foo, temp) -->
<!-- ``` -->

<!-- ## Saving annotations -->

<!-- We'll save the per-assay annotations as well.  -->

<!-- ```{r AE_MultiAssay: saving annotations} -->
<!-- # RNA -->
<!-- fwrite(as.data.frame(rowRanges(AERNASE)), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.RNA.annotations.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = TRUE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K1 -->
<!-- fwrite(as.data.frame(AEMS450K1.SE.rowData), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K1.annotations.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = TRUE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K1 -->
<!-- fwrite(as.data.frame(AEMS450K1.blood.SE.rowData), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K1.blood.annotations.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = TRUE, -->
<!--         showProgress = F, verbose = F) -->

<!-- # AEMS450K2 -->
<!-- fwrite(as.data.frame(AEMS450K2.SE.rowData), paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.CEA.AEMS450K2.annotations.csv.gz"), -->
<!--         sep = ",", quote = TRUE, -->
<!--         na = "", dec = ".", row.names = TRUE, -->
<!--         showProgress = F, verbose = F) -->

<!-- ``` -->


## Saving the MultiAssay experiment

Saving RDS of `MultiAssayExperiment`, and individual datasets.

```{r AE_MultiAssay: saving AE_MultiAssay as RDS}
# Save as RDS
saveRDS(AE_MultiAssay.CEA, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.MvaluesQCIMPSECLEAN.AERNASEvst.CEA.allClinData.rds"))
saveRDS(AE_MAE.CEA, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AE_MEA.MvaluesQCIMPSECLEAN.AERNASEvst.CEA.rds"))

```

We will also save the cleaned datasets for the Athero-Express Methylation Study 1 and 2.

```{r}
saveRDS(aems450k1.MvaluesQCIMPplaqueSECLEAN, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K1.MvaluesQCIMPSECLEAN.plaque.rds"))
saveRDS(aems450k1.BvaluesQCIMPplaqueSECLEAN, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K1.BvaluesQCIMPSECLEAN.plaque.rds"))

saveRDS(aems450k1.MvaluesQCIMPbloodSECLEAN, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K1.MvaluesQCIMPSECLEAN.blood.rds"))
saveRDS(aems450k1.BvaluesQCIMPbloodSECLEAN, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K1.BvaluesQCIMPSECLEAN.blood.rds"))

saveRDS(aems450k2.MvaluesQCIMPplaqueSECLEAN, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K2.MvaluesQCIMPSECLEAN.plaque.rds"))
saveRDS(aems450k2.BvaluesQCIMPplaqueSECLEAN, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K2.BvaluesQCIMPSECLEAN.plaque.rds"))

```


# Information

## Session information 

------

    Version:     v1.0.1
    Last update: 2024-06-11
    Written by:  Sander W. van der Laan (s.w.vanderlaan-2[at]umcutrecht.nl).
    Description: Preparing the DNA methylation data from Athero-Express Methylation Study 1 and 2 for downstream analyses. 
    Minimum requirements: R version 3.5.2 (2018-12-20) -- 'Eggshell Igloo', macOS Mojave (10.14.2).
    
    Changes log
    * v1.0.1 Small fix regarding the genetic KeyTable to match the 2_SNP_analys.Rmd.
    * v1.0.0 Initial version. 

  
------

```{r eval = TRUE}
sessionInfo()
```


## Saving environment

Saving RDS of `SummarizedExperiment` datasets for AEMS450K1 and AEMS450K2.

```{r SE: saving SE as RDS}

# Save as RDS
saveRDS(AEMS450K1.SE, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K1.MvaluesQCIMPplaque.SEclean.allClinData.rds"))
saveRDS(AEMS450K1.blood.SE, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K1.MvaluesQCIMPblood.SEclean.allClinData.rds"))

saveRDS(AEMS450K2.SE, paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K2.MvaluesQCIMPplaque.SEclean.allClinData.rds"))

# AEMS450K1.SE <- readRDS(paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K1.SE.allClinData.rds"))
# AEMS450K2.SE <- readRDS(paste0(OUT_loc, "/",Today,".",PROJECTNAME,".AEMS450K2.SE.allClinData.rds"))

```


## Cleaning environment
```{r cleaning environment, message=FALSE, warning=FALSE}
rm(AE_MultiAssay)

rm(AEMAE.CEA, AEMAE.CEA.all)

rm(AEMS450K1.SE.rowData, AEMS450K2.SE.rowData,
   AEMS450K1.SE.assay, AEMS450K2.SE.assay,
   AEMS450K1.SE.metadata, AEMS450K2.SE.metadata,
   aems450k1.MvaluesQCIMPplaqueSE, aems450k2.MvaluesQCIMPplaqueSE,
   aems450k1.MvaluesQCIMPplaqueSECLEAN, aems450k2.MvaluesQCIMPplaqueSECLEAN,
   AEMS450K1.SE, AEMS450K2.SE)

rm(AERNA.SE.UMIcorr, AERNA.SE.UMIcorr.ProtCodeLincRNA)

rm(infobcp,
   hm450.hg19.manifest.180909, hm450.hg19.manifest.naeem, hm450.hg19.manifest.naeem.list, hm450.controls, hm450.manifest.pop.GoNL)

rm(AEMS450K1.assay, AEMS450K2.assay, AERNA.assay)

rm(IlluminaHumanMethylation450kanno.ilmn12.hg19)

rm(AEMS450K1.blood.SE, AEMS450K1.blood.assay,
   aems450k1.MvaluesQCIMPbloodSECLEAN, AEMS450K1.blood.SE.rowData,
   AEMS450K1.blood.SE.assay,
   aems450k1.MvaluesQCIMPbloodSE,
   aems450k1.MvaluesQCIMPbloodSE.new, aems450k1.MvaluesQCIMPbloodSE.new2,
   AEMS450K1.blood.SE.metadata,
   AERNASE, bulkRNA_counts_raw,
   AERNA_UMIcorrSE_assay,
   bulkRNA_meta_clin,
   gene_lists_new, gene_lists_old, id,
   intersect_gene_lists,
   setdif_gene_lists_NEWvsOLD, setdif_gene_lists_OLDvsNEW,
   setdif_samples_AEDBvsNEW, setdif_samples_NEWvsAEDBCEA,
   setdif_samples_NEWvsAEDB
   )

rm(AERNAScomboSE, AERNAScomboSEnorm, AERNAScomboSEvst)

rm(gene_dataframe_EnsDb)
```

```{r Saving}
save.image(paste0(PROJECT_loc, "/",Today,".",PROJECTNAME,".DNAm_AEMS450K.preparation.RData"))
```

+-----------------------------------------------------------------------------------------------------------------------------------------+
| <sup>© 1979-2024 Sander W. van der Laan | s.w.vanderlaan[at]gmail[dot]com | [vanderlaanand.science](https://vanderlaanand.science).</sup> |
+-----------------------------------------------------------------------------------------------------------------------------------------+



